---
title: "Script repository"
subtitle: "(Hench *et al.* supplement)"
author: "Kosmas Hench and Martin Helmkampf"
date: "`r Sys.Date()`"
documentclass: book
bibliography: [book.bib]
biblio-style: apalike
link-citations: yes
github-repo: k-hench/bookdown
description: "Scripts used to produce Figures and Supplementary Figures of 'Rapid radiation in a highly diverse marine environment' by Hench, Helmkampf, McMillan an Puebla"
---

# Intro

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

<!--<span style="color:#FF0000">**Disclaimer:** We are currently still in the process of updating the documentation after the last round of revisions. Some sections are thus still not up to date.</span>-->


[This repository](https://github.com/k-hench/hamlet_radiation) contains the complete workflow used in the paper "*Rapid radiation in a highly diverse marine environment*".
The individual chapters of this documentation follow the separate main steps of the workflow.
Each of the chapters thus refers to an individual prefix in the _git x.x_ references of the papers method section.
The individual steps partly depend on each other - especially git 1 - git 3 should be executed in order and before the other steps.

## Analysis

A documentation of the data preparation and the data analysis (git 1.x - 19.x) can be found at:

- git 1.x: [Genotyping](git-1-genotyping-i-snps-only.html)
- git 2.x: [Genotyping all base pairs](git-2-genotyping-ii-all-callable-sites.html)
- git 3.x: [Analysis (<i>F<sub>ST</sub></i> & GxP)](git-3-analysis-i-fst-gxp.html)
- git 4.x: [Analysis (<i>d<sub>XY</sub></i> & $\pi$)](git-4-analysis-ii-dxy-pi.html)
- git 5.x: [Analysis (topolgy weighting)](git-5-analysis-iii-topology-weighting.html)
- git 6.x: [Analysis ($\rho$)](git-6-analysis-iv-rho.html)
- git 7.x: [Analysis (PCA)](git-7-analysis-v-principal-component-analysis.html)
- git 8.x: [Analysis (demographic history)](git-8-analysis-vi-demographic-history.html)
- git 9.x: [Analysis (hybridization)](git-9-analysis-vii-hybridization.html)
- git 10.x: [Analysis (admixture)](git-10-analysis-viii-admixture.html)
- git 11.x: [Analysis (allele age)](git-11-analysis-ix-allele-age.html)
- git 12.x: [Analysis (<i>F<sub>ST</sub></i> permutation)](git-12-analysis-x-fst-permutation-test.html)
- git 13.x: [Analysis (whg phylogeny)](git-13-analysis-xi-whole-genome-phylogenies.html)
- git 14.x: [Analysis (outlier region phylogeny)](git-14-analysis-xii-outlier-region-phylogenies.html)
- git 15.x: [Analysis ($\pi$ with/without outlier regions)](git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html)
- git 16.x: [Analysis (IBD)](git-16-analysis-xiv-identity-by-descent.html)
- git 17.x: [Analysis (dstats)](git-17-analysis-xv-dstats.html)
- git 18.x: [Genotyping all base pairs (mtDNA and unplaced Contigs)](git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html)
- git 19.x: [Analysis (Serraninae phyologeny)](git-19-analysis-xvi-serraninae-phylogeny.html)

## Prerequesites

All scripts assume two variables to be set within the bash environment:

  - `$BASE_DIR` is assumed to point to the base folder of this repository
  - `$SFTWR` is a folder that contains all the software dependencies that are used within the scripts

The analysis is controlled using the workflow manager [`nextflow`](https://www.nextflow.io/) and uses slightly different configurations across the individual pipelines. The exact commands used to execute the analysis during the development of the publication are stored within the aliases set within `sh/nextflow_alias.sh`.

Furthermore, external dependencies need to be downloaded and deployed at the expected places (s. README.md at the `ressources` folder).

## Figures

The creation of the figures is bundled in a single script (git 20) which can be executed once all `nextflow` scripts have successfully run.

```sh
cd $BASE_DIR
bash sh/create_figures.sh
```

This is basically just a wrapper script that will run all scripts located under `$BASE_DIR/R/fig`.
Under this location, you will find one `R` script per figure (and suppl. figure).
So if you are only interested in a single figure - that is the place to start looking.

Furthermore, a more detailed documentation exists for all the figure scripts used for the manuscript:

[F1](figure-1.html), [F2](figure-2.html), [F3](figure-3.html), [F4](figure-4.html) [F5](figure-5.html) and [F6](figure-6.html)

as well as for all the supplementary figures:

[SF1](supplementary-figure-1.html), [SF2](supplementary-figure-2.html), [SF3](supplementary-figure-3.html),
[SF4](supplementary-figure-4.html), [SF5](supplementary-figure-5.html), [SF6](supplementary-figure-6.html),
[SF7](supplementary-figure-7.html), [SF8](supplementary-figure-8.html), [SF9](supplementary-figure-9.html),
[SF10](supplementary-figure-10.html), [SF11](supplementary-figure-11.html), [SF12](supplementary-figure-12.html),
[SF13](supplementary-figure-13.html), [SF14](supplementary-figure-14.html), [SF15](supplementary-figure-15.html),
[SF16](supplementary-figure-16.html), [SF17](supplementary-figure-17.html), [SF18](supplementary-figure-18.html),
[SF19](supplementary-figure-19.html), [SF20](supplementary-figure-20.html) and [SF21](supplementary-figure-21.html).

## R setup

There is an additional **R** package needed to run the plotting scripts for the figures ({[GenomicOriginsScripts](https://k-hench.github.io/GenomicOriginsScripts/)}).
This depends on several non-CRAN R-packages, so to be able to install the package successfully, the following packages will also need to be installed:

```r
# installing non-CRAN dependencies
install.packages("remotes")
remotes::install_bioc("rtracklayer")
remotes::install_github("YuLab-SMU/ggtree")
remotes::install_github("k-hench/hypogen")
remotes::install_github("k-hench/hypoimg")
# installing GenomicOriginsScripts
remotes::install_github("k-hench/GenomicOriginsScripts")
```

Once these non-CRAN packages are installed, it should be possible to re-create the used **R** environment using the {[renv](https://rstudio.github.io/renv/)} package.
After opening the RStudio project (`hamlet_radiation.Rproj`), call:

```r
# restoring R environment
install.packages("renv")
renv::restore()
```

Apart from the specific **R** packages that can be retrieved via {renv} from the `renv.lock` file, the used **R** setup at the at time of compilation is as follows:

```{r}
sessionInfo()
```

---
