<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>19 Figure 3 | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="19 Figure 3 | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="19 Figure 3 | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench and Martin Helmkampf" />


<meta name="date" content="2021-08-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="figure-2.html"/>
<link rel="next" href="figure-4.html"/>
<script src="libs/header-attrs-2.7.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#r-setup"><i class="fa fa-check"></i><b>1.4</b> R setup</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a>
<ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a>
<ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i> &amp; pi)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (topology weighting)</a>
<ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (Principal Component Analysis)</a>
<ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#details-of-analysis_pca.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_pca.nf</code></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#setup-2"><i class="fa fa-check"></i><b>8.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-4"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-5"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-6"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html"><i class="fa fa-check"></i><b>12</b> (git 11) Analysis IX (Allele Age)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#summary-10"><i class="fa fa-check"></i><b>12.1</b> Summary</a></li>
<li class="chapter" data-level="12.2" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#details-of-analysis_allele_age.nf"><i class="fa fa-check"></i><b>12.2</b> Details of <code>analysis_allele_age.nf</code></a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#setup-3"><i class="fa fa-check"></i><b>12.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html"><i class="fa fa-check"></i><b>13</b> (git 12) Analysis X (<i>F<sub>ST</sub></i> Permutation Test)</a>
<ul>
<li class="chapter" data-level="13.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#summary-11"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#details-of-analysis_fst_sign.nf"><i class="fa fa-check"></i><b>13.2</b> Details of <code>analysis_fst_sign.nf</code></a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#setup-4"><i class="fa fa-check"></i><b>13.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html"><i class="fa fa-check"></i><b>14</b> (git 13) Analysis XI (Whole Genome Phylogenies)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#summary-12"><i class="fa fa-check"></i><b>14.1</b> Summary</a></li>
<li class="chapter" data-level="14.2" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#details-of-analysis_phylo_whg.nf"><i class="fa fa-check"></i><b>14.2</b> Details of <code>analysis_phylo_whg.nf</code></a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#setup-5"><i class="fa fa-check"></i><b>14.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html"><i class="fa fa-check"></i><b>15</b> (git 14) Analysis XII (Outlier Region Phylogenies)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#summary-13"><i class="fa fa-check"></i><b>15.1</b> Summary</a></li>
<li class="chapter" data-level="15.2" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#details-of-analysis_phylo_regions.nf"><i class="fa fa-check"></i><b>15.2</b> Details of <code>analysis_phylo_regions.nf</code></a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#setup-6"><i class="fa fa-check"></i><b>15.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="git-15-data-visualization.html"><a href="git-15-data-visualization.html"><i class="fa fa-check"></i><b>16</b> (git 15) Data Visualization</a></li>
<li class="chapter" data-level="17" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>17</b> Figure 1</a>
<ul>
<li class="chapter" data-level="17.1" data-path="figure-1.html"><a href="figure-1.html#summary-14"><i class="fa fa-check"></i><b>17.1</b> Summary</a></li>
<li class="chapter" data-level="17.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>17.2</b> Details of <code>plot_F1.R</code></a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>17.2.1</b> Config</a></li>
<li class="chapter" data-level="17.2.2" data-path="figure-1.html"><a href="figure-1.html#fish-tree-of-life-subtree"><i class="fa fa-check"></i><b>17.2.2</b> Fish Tree Of Life Subtree</a></li>
<li class="chapter" data-level="17.2.3" data-path="figure-1.html"><a href="figure-1.html#genetic-differentiation-boxplots"><i class="fa fa-check"></i><b>17.2.3</b> Genetic differentiation boxplots</a></li>
<li class="chapter" data-level="17.2.4" data-path="figure-1.html"><a href="figure-1.html#principal-component-analyses"><i class="fa fa-check"></i><b>17.2.4</b> Principal Component Analyses</a></li>
<li class="chapter" data-level="17.2.5" data-path="figure-1.html"><a href="figure-1.html#legend-and-final-figure"><i class="fa fa-check"></i><b>17.2.5</b> Legend and Final Figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>18</b> Figure 2</a>
<ul>
<li class="chapter" data-level="18.1" data-path="figure-2.html"><a href="figure-2.html#summary-15"><i class="fa fa-check"></i><b>18.1</b> Summary</a></li>
<li class="chapter" data-level="18.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>18.2</b> Details of <code>plot_F2.R</code></a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>18.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>19</b> Figure 3</a>
<ul>
<li class="chapter" data-level="19.1" data-path="figure-3.html"><a href="figure-3.html#summary-16"><i class="fa fa-check"></i><b>19.1</b> Summary</a></li>
<li class="chapter" data-level="19.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>19.2</b> Details of <code>plot_F3.R</code></a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>19.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>20</b> Figure 4</a>
<ul>
<li class="chapter" data-level="20.1" data-path="figure-4.html"><a href="figure-4.html#summary-17"><i class="fa fa-check"></i><b>20.1</b> Summary</a></li>
<li class="chapter" data-level="20.2" data-path="figure-4.html"><a href="figure-4.html#details-of-plot_f4.r"><i class="fa fa-check"></i><b>20.2</b> Details of <code>plot_F4.R</code></a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="figure-4.html"><a href="figure-4.html#config-3"><i class="fa fa-check"></i><b>20.2.1</b> Config</a></li>
<li class="chapter" data-level="20.2.2" data-path="figure-4.html"><a href="figure-4.html#data-import"><i class="fa fa-check"></i><b>20.2.2</b> Data import</a></li>
<li class="chapter" data-level="20.2.3" data-path="figure-4.html"><a href="figure-4.html#plotting"><i class="fa fa-check"></i><b>20.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>21</b> Figure 5</a>
<ul>
<li class="chapter" data-level="21.1" data-path="figure-5.html"><a href="figure-5.html#summary-18"><i class="fa fa-check"></i><b>21.1</b> Summary</a></li>
<li class="chapter" data-level="21.2" data-path="figure-5.html"><a href="figure-5.html#details-of-plot_f5.r"><i class="fa fa-check"></i><b>21.2</b> Details of <code>plot_F5.R</code></a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="figure-5.html"><a href="figure-5.html#config-4"><i class="fa fa-check"></i><b>21.2.1</b> Config</a></li>
<li class="chapter" data-level="21.2.2" data-path="figure-5.html"><a href="figure-5.html#data-import-1"><i class="fa fa-check"></i><b>21.2.2</b> Data import</a></li>
<li class="chapter" data-level="21.2.3" data-path="figure-5.html"><a href="figure-5.html#plotting-1"><i class="fa fa-check"></i><b>21.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>22</b> Figure 6</a>
<ul>
<li class="chapter" data-level="22.1" data-path="figure-6.html"><a href="figure-6.html#summary-19"><i class="fa fa-check"></i><b>22.1</b> Summary</a></li>
<li class="chapter" data-level="22.2" data-path="figure-6.html"><a href="figure-6.html#details-of-plot_f6.r"><i class="fa fa-check"></i><b>22.2</b> Details of <code>plot_F6.R</code></a>
<ul>
<li class="chapter" data-level="22.2.1" data-path="figure-6.html"><a href="figure-6.html#config-5"><i class="fa fa-check"></i><b>22.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html"><i class="fa fa-check"></i><b>23</b> Supplementary Figure 1</a>
<ul>
<li class="chapter" data-level="23.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#summary-20"><i class="fa fa-check"></i><b>23.1</b> Summary</a></li>
<li class="chapter" data-level="23.2" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#details-of-plot_sf1.r"><i class="fa fa-check"></i><b>23.2</b> Details of <code>plot_SF1.R</code></a>
<ul>
<li class="chapter" data-level="23.2.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#config-6"><i class="fa fa-check"></i><b>23.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html"><i class="fa fa-check"></i><b>24</b> Supplementary Figure 2</a>
<ul>
<li class="chapter" data-level="24.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#summary-21"><i class="fa fa-check"></i><b>24.1</b> Summary</a></li>
<li class="chapter" data-level="24.2" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#details-of-plot_sf2.r"><i class="fa fa-check"></i><b>24.2</b> Details of <code>plot_SF2.R</code></a>
<ul>
<li class="chapter" data-level="24.2.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#config-7"><i class="fa fa-check"></i><b>24.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html"><i class="fa fa-check"></i><b>25</b> Supplementary Figure 3</a>
<ul>
<li class="chapter" data-level="25.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#summary-22"><i class="fa fa-check"></i><b>25.1</b> Summary</a></li>
<li class="chapter" data-level="25.2" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#details-of-plot_sf3.r"><i class="fa fa-check"></i><b>25.2</b> Details of <code>plot_SF3.R</code></a>
<ul>
<li class="chapter" data-level="25.2.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#config-8"><i class="fa fa-check"></i><b>25.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html"><i class="fa fa-check"></i><b>26</b> Supplementary Figure 4</a>
<ul>
<li class="chapter" data-level="26.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#summary-23"><i class="fa fa-check"></i><b>26.1</b> Summary</a></li>
<li class="chapter" data-level="26.2" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#details-of-plot_sf4.r"><i class="fa fa-check"></i><b>26.2</b> Details of <code>plot_SF4.R</code></a>
<ul>
<li class="chapter" data-level="26.2.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#config-9"><i class="fa fa-check"></i><b>26.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html"><i class="fa fa-check"></i><b>27</b> Supplementary Figure 5</a>
<ul>
<li class="chapter" data-level="27.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#summary-24"><i class="fa fa-check"></i><b>27.1</b> Summary</a></li>
<li class="chapter" data-level="27.2" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#details-of-plot_sf5.r"><i class="fa fa-check"></i><b>27.2</b> Details of <code>plot_SF5.R</code></a>
<ul>
<li class="chapter" data-level="27.2.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#config-10"><i class="fa fa-check"></i><b>27.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html"><i class="fa fa-check"></i><b>28</b> Supplementary Figure 6</a>
<ul>
<li class="chapter" data-level="28.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#summary-25"><i class="fa fa-check"></i><b>28.1</b> Summary</a></li>
<li class="chapter" data-level="28.2" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#details-of-plot_sf6.r"><i class="fa fa-check"></i><b>28.2</b> Details of <code>plot_SF6.R</code></a>
<ul>
<li class="chapter" data-level="28.2.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#config-11"><i class="fa fa-check"></i><b>28.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="29" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html"><i class="fa fa-check"></i><b>29</b> Supplementary Figure 7</a>
<ul>
<li class="chapter" data-level="29.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#summary-26"><i class="fa fa-check"></i><b>29.1</b> Summary</a></li>
<li class="chapter" data-level="29.2" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#details-of-plot_sf7.r"><i class="fa fa-check"></i><b>29.2</b> Details of <code>plot_SF7.R</code></a>
<ul>
<li class="chapter" data-level="29.2.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#config-12"><i class="fa fa-check"></i><b>29.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="30" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html"><i class="fa fa-check"></i><b>30</b> Supplementary Figure 8</a>
<ul>
<li class="chapter" data-level="30.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#summary-27"><i class="fa fa-check"></i><b>30.1</b> Summary</a></li>
<li class="chapter" data-level="30.2" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#details-of-plot_sf8.r"><i class="fa fa-check"></i><b>30.2</b> Details of <code>plot_SF8.R</code></a>
<ul>
<li class="chapter" data-level="30.2.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#config-13"><i class="fa fa-check"></i><b>30.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="31" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html"><i class="fa fa-check"></i><b>31</b> Supplementary Figure 9</a>
<ul>
<li class="chapter" data-level="31.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#summary-28"><i class="fa fa-check"></i><b>31.1</b> Summary</a></li>
<li class="chapter" data-level="31.2" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#details-of-plot_sf9.r"><i class="fa fa-check"></i><b>31.2</b> Details of <code>plot_SF9.R</code></a>
<ul>
<li class="chapter" data-level="31.2.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#config-14"><i class="fa fa-check"></i><b>31.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="32" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html"><i class="fa fa-check"></i><b>32</b> Supplementary Figure 10</a>
<ul>
<li class="chapter" data-level="32.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#summary-29"><i class="fa fa-check"></i><b>32.1</b> Summary</a></li>
<li class="chapter" data-level="32.2" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#details-of-plot_sf10.r"><i class="fa fa-check"></i><b>32.2</b> Details of <code>plot_SF10.R</code></a>
<ul>
<li class="chapter" data-level="32.2.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#config-15"><i class="fa fa-check"></i><b>32.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html"><i class="fa fa-check"></i><b>33</b> Supplementary Figure 11</a>
<ul>
<li class="chapter" data-level="33.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#summary-30"><i class="fa fa-check"></i><b>33.1</b> Summary</a></li>
<li class="chapter" data-level="33.2" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#details-of-plot_sf11.r"><i class="fa fa-check"></i><b>33.2</b> Details of <code>plot_SF11.R</code></a>
<ul>
<li class="chapter" data-level="33.2.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#config-16"><i class="fa fa-check"></i><b>33.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html"><i class="fa fa-check"></i><b>34</b> Supplementary Figure 12</a>
<ul>
<li class="chapter" data-level="34.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#summary-31"><i class="fa fa-check"></i><b>34.1</b> Summary</a></li>
<li class="chapter" data-level="34.2" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#details-of-plot_sf12.r"><i class="fa fa-check"></i><b>34.2</b> Details of <code>plot_SF12.R</code></a>
<ul>
<li class="chapter" data-level="34.2.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#config-17"><i class="fa fa-check"></i><b>34.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="35" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html"><i class="fa fa-check"></i><b>35</b> Supplementary Figure 13</a>
<ul>
<li class="chapter" data-level="35.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#summary-32"><i class="fa fa-check"></i><b>35.1</b> Summary</a></li>
<li class="chapter" data-level="35.2" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#details-of-plot_sf13.r"><i class="fa fa-check"></i><b>35.2</b> Details of <code>plot_SF13.R</code></a>
<ul>
<li class="chapter" data-level="35.2.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#config-18"><i class="fa fa-check"></i><b>35.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="36" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html"><i class="fa fa-check"></i><b>36</b> Supplementary Figure 14</a>
<ul>
<li class="chapter" data-level="36.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#summary-33"><i class="fa fa-check"></i><b>36.1</b> Summary</a></li>
<li class="chapter" data-level="36.2" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#details-of-plot_sf14.r"><i class="fa fa-check"></i><b>36.2</b> Details of <code>plot_SF14.R</code></a>
<ul>
<li class="chapter" data-level="36.2.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#config-19"><i class="fa fa-check"></i><b>36.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="37" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html"><i class="fa fa-check"></i><b>37</b> Supplementary Figure 15</a>
<ul>
<li class="chapter" data-level="37.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#summary-34"><i class="fa fa-check"></i><b>37.1</b> Summary</a></li>
<li class="chapter" data-level="37.2" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#details-of-plot_sf15.r"><i class="fa fa-check"></i><b>37.2</b> Details of <code>plot_SF15.R</code></a>
<ul>
<li class="chapter" data-level="37.2.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#config-20"><i class="fa fa-check"></i><b>37.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="38" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html"><i class="fa fa-check"></i><b>38</b> Supplementary Figure 16</a>
<ul>
<li class="chapter" data-level="38.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#summary-35"><i class="fa fa-check"></i><b>38.1</b> Summary</a></li>
<li class="chapter" data-level="38.2" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#details-of-plot_sf16.r"><i class="fa fa-check"></i><b>38.2</b> Details of <code>plot_SF16.R</code></a>
<ul>
<li class="chapter" data-level="38.2.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#config-21"><i class="fa fa-check"></i><b>38.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-3" class="section level1" number="19">
<h1><span class="header-section-number">19</span> Figure 3</h1>
<div id="summary-16" class="section level2" number="19.1">
<h2><span class="header-section-number">19.1</span> Summary</h2>
<p>This is the accessory documentation of Figure 3.
It should be possible to recreate the figure by running the <strong>R</strong> script <code>plot_F3.R</code>:</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb71-1"><a href="figure-3.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span></span>
<span id="cb71-2"><a href="figure-3.html#cb71-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-3"><a href="figure-3.html#cb71-3" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> <span class="at">--vanilla</span> R/fig/plot_F3.R <span class="dt">\</span></span>
<span id="cb71-4"><a href="figure-3.html#cb71-4" aria-hidden="true" tabindex="-1"></a>   2_analysis/fst/50k/ <span class="dt">\</span></span>
<span id="cb71-5"><a href="figure-3.html#cb71-5" aria-hidden="true" tabindex="-1"></a>   2_analysis/summaries/fst_globals.txt</span></code></pre></div>
</div>
<div id="details-of-plot_f3.r" class="section level2" number="19.2">
<h2><span class="header-section-number">19.2</span> Details of <code>plot_F3.R</code></h2>
<p>In the following, the individual steps of the R script are documented.
It is an executable R script that depends on the accessory <strong>R</strong> packages <a href="https://k-hench.github.io/GenomicOriginsScripts"><strong>GenomicOriginsScripts</strong></a> and on the <strong>R</strong> packages <a href="https://k-hench.github.io/hypoimg"><strong>hypoimg</strong></a>, <a href="https://k-hench.github.io/hypogen"><strong>hypogen</strong></a>, <a href="https://vroom.r-lib.org/"><strong>vroom</strong></a> and <a href="https://ggforce.data-imaginist.com/"><strong>ggforce</strong></a>.</p>
<div id="config-2" class="section level3" number="19.2.1">
<h3><span class="header-section-number">19.2.1</span> Config</h3>
<p>The scripts start with a header that contains copy &amp; paste templates to execute or debug the script:</p>
<p>The next section processes the input from the command line.
It stores the arguments in the vector <code>args</code>.
The R packages are loaded and the script name and the current working directory are stored inside variables (<code>script_name</code>, <code>plot_comment</code>).
This information will later be written into the meta data of the figure to help us tracing back the scripts that created the figures in the future.</p>
<p>Then we drop all the imported information besides the arguments following the script name and print the information to the terminal.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="figure-3.html#cb72-1" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly=</span><span class="cn">FALSE</span>)</span>
<span id="cb72-2"><a href="figure-3.html#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="co"># setup -----------------------</span></span>
<span id="cb72-3"><a href="figure-3.html#cb72-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicOriginsScripts)</span>
<span id="cb72-4"><a href="figure-3.html#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggforce)</span>
<span id="cb72-5"><a href="figure-3.html#cb72-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hypoimg)</span>
<span id="cb72-6"><a href="figure-3.html#cb72-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hypogen)</span>
<span id="cb72-7"><a href="figure-3.html#cb72-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vroom)</span>
<span id="cb72-8"><a href="figure-3.html#cb72-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-9"><a href="figure-3.html#cb72-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb72-10"><a href="figure-3.html#cb72-10" aria-hidden="true" tabindex="-1"></a>script_name <span class="ot">&lt;-</span> args[<span class="dv">5</span>] <span class="sc">%&gt;%</span></span>
<span id="cb72-11"><a href="figure-3.html#cb72-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_remove</span>(.,<span class="st">&#39;--file=&#39;</span>)</span>
<span id="cb72-12"><a href="figure-3.html#cb72-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-13"><a href="figure-3.html#cb72-13" aria-hidden="true" tabindex="-1"></a>plot_comment <span class="ot">&lt;-</span> script_name <span class="sc">%&gt;%</span></span>
<span id="cb72-14"><a href="figure-3.html#cb72-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_c</span>(<span class="st">&#39;mother-script = &#39;</span>,<span class="fu">getwd</span>(),<span class="st">&#39;/&#39;</span>,.)</span>
<span id="cb72-15"><a href="figure-3.html#cb72-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-16"><a href="figure-3.html#cb72-16" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">process_input</span>(script_name, args)</span></code></pre></div>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="figure-3.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ── Script: R/fig/plot_F3.R ────────────────────────────────────────────</span></span>
<span id="cb73-2"><a href="figure-3.html#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Parameters read:</span></span>
<span id="cb73-3"><a href="figure-3.html#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ★ 1: 2_analysis/fst/50k/</span></span>
<span id="cb73-4"><a href="figure-3.html#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ★ 2: 2_analysis/summaries/fst_globals.txt</span></span>
<span id="cb73-5"><a href="figure-3.html#cb73-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ────────────────────────────────────────── /current/working/directory ──</span></span></code></pre></div>
<p>The directory containing the sliding window <span class="math inline">\(F_{ST}\)</span> data and the and
the file with the genome wide average <span class="math inline">\(F_{ST}\)</span> for all the species
comparisons are received from the command line input.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="figure-3.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># config -----------------------</span></span>
<span id="cb74-2"><a href="figure-3.html#cb74-2" aria-hidden="true" tabindex="-1"></a>data_dir <span class="ot">&lt;-</span> <span class="fu">as.character</span>(args[<span class="dv">1</span>])</span>
<span id="cb74-3"><a href="figure-3.html#cb74-3" aria-hidden="true" tabindex="-1"></a>globals_file <span class="ot">&lt;-</span> <span class="fu">as.character</span>(args[<span class="dv">2</span>])</span></code></pre></div>
<p>Then, the data folder is scanned for windowed <span class="math inline">\(F_{ST}\)</span> data with an
window size of 50 kb.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="figure-3.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="co"># script -----------------------</span></span>
<span id="cb75-2"><a href="figure-3.html#cb75-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-3"><a href="figure-3.html#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="co"># locate data files</span></span>
<span id="cb75-4"><a href="figure-3.html#cb75-4" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">dir</span>(<span class="at">path =</span> data_dir, <span class="at">pattern =</span> <span class="st">&#39;.50k.windowed.weir.fst.gz&#39;</span>)</span></code></pre></div>
<p>Next, the genome wide average <span class="math inline">\(F_{ST}\)</span> data for each population pair is loaded.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="figure-3.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load genome wide average fst data</span></span>
<span id="cb76-2"><a href="figure-3.html#cb76-2" aria-hidden="true" tabindex="-1"></a>globals <span class="ot">&lt;-</span> vroom<span class="sc">::</span><span class="fu">vroom</span>(globals_file, <span class="at">delim =</span> <span class="st">&#39;</span><span class="sc">\t</span><span class="st">&#39;</span>,</span>
<span id="cb76-3"><a href="figure-3.html#cb76-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">&#39;loc&#39;</span>,<span class="st">&#39;run&#39;</span>,<span class="st">&#39;mean&#39;</span>,<span class="st">&#39;weighted&#39;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb76-4"><a href="figure-3.html#cb76-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">run =</span> <span class="fu">str_c</span>(loc,<span class="st">&#39;-&#39;</span>,run) <span class="sc">%&gt;%</span></span>
<span id="cb76-5"><a href="figure-3.html#cb76-5" aria-hidden="true" tabindex="-1"></a>           <span class="fu">reformat_run_name</span>()</span>
<span id="cb76-6"><a href="figure-3.html#cb76-6" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<p>The package <a href="https://k-hench.github.io/GenomicOriginsScripts"><strong>GenomicOriginsScripts</strong></a>
contains the function <code>get_fst_fixed</code> to import <span class="math inline">\(F_{ST}\)</span> data and compute the
number, average length and cumulative length of regions exceeding a given <span class="math inline">\(F_{ST}\)</span>
threshold.</p>
<p>Here, we prepare a table of a series of thresholds and all pair wise species comparisons
as a configuration table for the following import with <code>get_fst_fixed</code>.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="figure-3.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare data import settings within a data table (tibble)</span></span>
<span id="cb77-2"><a href="figure-3.html#cb77-2" aria-hidden="true" tabindex="-1"></a>import_table <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">file =</span> <span class="fu">str_c</span>(data_dir,files),</span>
<span id="cb77-3"><a href="figure-3.html#cb77-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">fst_threshold =</span> <span class="fu">c</span>(.<span class="dv">5</span>,.<span class="dv">4</span>,.<span class="dv">3</span>,.<span class="dv">2</span>,.<span class="dv">1</span>,</span>
<span id="cb77-4"><a href="figure-3.html#cb77-4" aria-hidden="true" tabindex="-1"></a>                                       .<span class="dv">05</span>,.<span class="dv">02</span>,.<span class="dv">01</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb77-5"><a href="figure-3.html#cb77-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cross_df</span>() <span class="sc">%&gt;%</span></span>
<span id="cb77-6"><a href="figure-3.html#cb77-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>( <span class="at">run =</span>  file <span class="sc">%&gt;%</span></span>
<span id="cb77-7"><a href="figure-3.html#cb77-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">str_remove</span>(<span class="st">&#39;^.*/&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb77-8"><a href="figure-3.html#cb77-8" aria-hidden="true" tabindex="-1"></a>            <span class="fu">str_sub</span>(., <span class="dv">1</span>, <span class="dv">11</span>) <span class="sc">%&gt;%</span></span>
<span id="cb77-9"><a href="figure-3.html#cb77-9" aria-hidden="true" tabindex="-1"></a>            <span class="fu">reformat_run_name</span>())</span></code></pre></div>
<p>Then we define a data import function (this used to live in {GenomicOriginsScript}, but was moved here due to namspace issue).</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="figure-3.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import dxy data and compute threshold stats</span></span>
<span id="cb78-2"><a href="figure-3.html#cb78-2" aria-hidden="true" tabindex="-1"></a>get_fst_fixed <span class="ot">&lt;-</span> <span class="cf">function</span>(file, run, fst_threshold,...){</span>
<span id="cb78-3"><a href="figure-3.html#cb78-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-4"><a href="figure-3.html#cb78-4" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> hypogen<span class="sc">::</span><span class="fu">hypo_import_windows</span>(file, ...) <span class="sc">%&gt;%</span></span>
<span id="cb78-5"><a href="figure-3.html#cb78-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">rank =</span> <span class="fu">rank</span>(WEIGHTED_FST, <span class="at">ties.method =</span> <span class="st">&quot;random&quot;</span>))<span class="sc">%&gt;%</span></span>
<span id="cb78-6"><a href="figure-3.html#cb78-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">thresh =</span> fst_threshold) <span class="sc">%&gt;%</span></span>
<span id="cb78-7"><a href="figure-3.html#cb78-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">outl =</span> (WEIGHTED_FST <span class="sc">&gt;</span> thresh) <span class="sc">%&gt;%</span> <span class="fu">as.numeric</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb78-8"><a href="figure-3.html#cb78-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(outl <span class="sc">==</span> <span class="dv">1</span> )</span>
<span id="cb78-9"><a href="figure-3.html#cb78-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-10"><a href="figure-3.html#cb78-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nrow</span>(data) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb78-11"><a href="figure-3.html#cb78-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">tibble</span>(<span class="at">run =</span> run, <span class="at">n =</span> <span class="dv">0</span>, <span class="at">avg_length =</span> <span class="cn">NA</span>, <span class="at">med_length =</span> <span class="cn">NA</span>, <span class="at">min_length =</span> <span class="cn">NA</span>, <span class="at">max_length =</span> <span class="cn">NA</span>,</span>
<span id="cb78-12"><a href="figure-3.html#cb78-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">sd_length =</span> <span class="cn">NA</span>, <span class="at">overal_length =</span> <span class="cn">NA</span>, <span class="at">threshold_value =</span> fst_threshold))</span>
<span id="cb78-13"><a href="figure-3.html#cb78-13" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb78-14"><a href="figure-3.html#cb78-14" aria-hidden="true" tabindex="-1"></a>    data <span class="sc">%&gt;%</span></span>
<span id="cb78-15"><a href="figure-3.html#cb78-15" aria-hidden="true" tabindex="-1"></a>      <span class="co"># next, we want to collapse overlapping windows</span></span>
<span id="cb78-16"><a href="figure-3.html#cb78-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(CHROM) <span class="sc">%&gt;%</span></span>
<span id="cb78-17"><a href="figure-3.html#cb78-17" aria-hidden="true" tabindex="-1"></a>      <span class="co"># we check for overlap and create &#39;region&#39; IDs</span></span>
<span id="cb78-18"><a href="figure-3.html#cb78-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">check =</span> <span class="dv">1</span><span class="sc">-</span>(<span class="fu">lag</span>(BIN_END,<span class="at">default =</span> <span class="dv">0</span>)<span class="sc">&gt;</span>BIN_START),</span>
<span id="cb78-19"><a href="figure-3.html#cb78-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ID =</span> <span class="fu">str_c</span>(CHROM,<span class="st">&#39;_&#39;</span>,<span class="fu">cumsum</span>(check))) <span class="sc">%&gt;%</span></span>
<span id="cb78-20"><a href="figure-3.html#cb78-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb78-21"><a href="figure-3.html#cb78-21" aria-hidden="true" tabindex="-1"></a>      <span class="co"># then we collapse the regions by ID</span></span>
<span id="cb78-22"><a href="figure-3.html#cb78-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(ID) <span class="sc">%&gt;%</span></span>
<span id="cb78-23"><a href="figure-3.html#cb78-23" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">run =</span> run[<span class="dv">1</span>],</span>
<span id="cb78-24"><a href="figure-3.html#cb78-24" aria-hidden="true" tabindex="-1"></a>                       <span class="at">run =</span> run[<span class="dv">1</span>],</span>
<span id="cb78-25"><a href="figure-3.html#cb78-25" aria-hidden="true" tabindex="-1"></a>                       <span class="at">treshold_value =</span> thresh[<span class="dv">1</span>],</span>
<span id="cb78-26"><a href="figure-3.html#cb78-26" aria-hidden="true" tabindex="-1"></a>                       <span class="at">CHROM =</span> CHROM[<span class="dv">1</span>],</span>
<span id="cb78-27"><a href="figure-3.html#cb78-27" aria-hidden="true" tabindex="-1"></a>                       <span class="at">BIN_START =</span> <span class="fu">min</span>(BIN_START),</span>
<span id="cb78-28"><a href="figure-3.html#cb78-28" aria-hidden="true" tabindex="-1"></a>                       <span class="at">BIN_END =</span> <span class="fu">max</span>(BIN_END)) <span class="sc">%&gt;%</span></span>
<span id="cb78-29"><a href="figure-3.html#cb78-29" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">PEAK_SIZE =</span> BIN_END<span class="sc">-</span>BIN_START) <span class="sc">%&gt;%</span></span>
<span id="cb78-30"><a href="figure-3.html#cb78-30" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarize</span>(<span class="at">run =</span> run[<span class="dv">1</span>],</span>
<span id="cb78-31"><a href="figure-3.html#cb78-31" aria-hidden="true" tabindex="-1"></a>                       <span class="at">run =</span> run[<span class="dv">1</span>],</span>
<span id="cb78-32"><a href="figure-3.html#cb78-32" aria-hidden="true" tabindex="-1"></a>                       <span class="at">n =</span> <span class="fu">length</span>(ID),</span>
<span id="cb78-33"><a href="figure-3.html#cb78-33" aria-hidden="true" tabindex="-1"></a>                       <span class="at">avg_length =</span> <span class="fu">mean</span>(PEAK_SIZE),</span>
<span id="cb78-34"><a href="figure-3.html#cb78-34" aria-hidden="true" tabindex="-1"></a>                       <span class="at">med_length =</span> <span class="fu">median</span>(PEAK_SIZE),</span>
<span id="cb78-35"><a href="figure-3.html#cb78-35" aria-hidden="true" tabindex="-1"></a>                       <span class="at">min_length =</span> <span class="fu">min</span>(PEAK_SIZE),</span>
<span id="cb78-36"><a href="figure-3.html#cb78-36" aria-hidden="true" tabindex="-1"></a>                       <span class="at">max_length =</span> <span class="fu">max</span>(PEAK_SIZE),</span>
<span id="cb78-37"><a href="figure-3.html#cb78-37" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sd_length =</span> <span class="fu">sd</span>(PEAK_SIZE),</span>
<span id="cb78-38"><a href="figure-3.html#cb78-38" aria-hidden="true" tabindex="-1"></a>                       <span class="at">overal_length =</span> <span class="fu">sum</span>(PEAK_SIZE),</span>
<span id="cb78-39"><a href="figure-3.html#cb78-39" aria-hidden="true" tabindex="-1"></a>                       <span class="at">threshold_value =</span> treshold_value[<span class="dv">1</span>])</span>
<span id="cb78-40"><a href="figure-3.html#cb78-40" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb78-41"><a href="figure-3.html#cb78-41" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Using the configuration table, the <span class="math inline">\(F_{ST}\)</span> data are loaded, and the
threshold-specific stats are computed.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="figure-3.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load data and compute statistics based on fixed fst treshold</span></span>
<span id="cb79-2"><a href="figure-3.html#cb79-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">pmap_dfr</span>(import_table, get_fst_fixed) <span class="sc">%&gt;%</span></span>
<span id="cb79-3"><a href="figure-3.html#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(globals) <span class="sc">%&gt;%</span></span>
<span id="cb79-4"><a href="figure-3.html#cb79-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">run =</span> <span class="fu">fct_reorder</span>(run, weighted))</span></code></pre></div>
<p>To simplify the figure, a subset of the original thresholds are selected
and some columns are renamed for clean figure labels.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="figure-3.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pre-format labels</span></span>
<span id="cb80-2"><a href="figure-3.html#cb80-2" aria-hidden="true" tabindex="-1"></a>data2 <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb80-3"><a href="figure-3.html#cb80-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(threshold_value,weighted,n,avg_length,overal_length) <span class="sc">%&gt;%</span></span>
<span id="cb80-4"><a href="figure-3.html#cb80-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">avg_length =</span> avg_length<span class="sc">/</span><span class="dv">1000</span>,</span>
<span id="cb80-5"><a href="figure-3.html#cb80-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">overal_length =</span> overal_length<span class="sc">/</span>(<span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb80-6"><a href="figure-3.html#cb80-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">atop(Number~of,Regions)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;n&#39;</span>,</span>
<span id="cb80-7"><a href="figure-3.html#cb80-7" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">atop(Average~Region,Length~(kb))</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;avg_length&#39;</span>,</span>
<span id="cb80-8"><a href="figure-3.html#cb80-8" aria-hidden="true" tabindex="-1"></a>         <span class="st">`</span><span class="at">atop(Cummulative~Region,Length~(Mb))</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&#39;overal_length&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb80-9"><a href="figure-3.html#cb80-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">&#39;variable&#39;</span>,<span class="at">values_to =</span> <span class="st">&#39;Value&#39;</span>,<span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb80-10"><a href="figure-3.html#cb80-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">threshold_value =</span> <span class="fu">str_c</span>(<span class="st">&#39;italic(F[ST])~threshold:~&#39;</span>,</span>
<span id="cb80-11"><a href="figure-3.html#cb80-11" aria-hidden="true" tabindex="-1"></a>                                 threshold_value),</span>
<span id="cb80-12"><a href="figure-3.html#cb80-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">variable =</span> <span class="fu">factor</span>(variable, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;atop(Number~of,Regions)&#39;</span>,</span>
<span id="cb80-13"><a href="figure-3.html#cb80-13" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&#39;atop(Average~Region,Length~(kb))&#39;</span>,</span>
<span id="cb80-14"><a href="figure-3.html#cb80-14" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">&#39;atop(Cummulative~Region,Length~(Mb))&#39;</span>)))</span></code></pre></div>
<p>At this point we can create the figure.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="figure-3.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set font size</span></span>
<span id="cb81-2"><a href="figure-3.html#cb81-2" aria-hidden="true" tabindex="-1"></a>base_line_clr <span class="ot">&lt;-</span> <span class="st">&quot;black&quot;</span></span>
<span id="cb81-3"><a href="figure-3.html#cb81-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-4"><a href="figure-3.html#cb81-4" aria-hidden="true" tabindex="-1"></a><span class="co"># compile plot</span></span>
<span id="cb81-5"><a href="figure-3.html#cb81-5" aria-hidden="true" tabindex="-1"></a>p_done <span class="ot">&lt;-</span> data2 <span class="sc">%&gt;%</span></span>
<span id="cb81-6"><a href="figure-3.html#cb81-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># select thresholds of interest</span></span>
<span id="cb81-7"><a href="figure-3.html#cb81-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(threshold_value <span class="sc">%in%</span> (<span class="fu">c</span>(<span class="fl">0.02</span>,.<span class="dv">1</span>,<span class="fl">0.2</span>, <span class="fl">0.3</span>, .<span class="dv">4</span>) <span class="sc">%&gt;%</span></span>
<span id="cb81-8"><a href="figure-3.html#cb81-8" aria-hidden="true" tabindex="-1"></a>                                   <span class="fu">str_c</span>(<span class="st">&quot;italic(F[ST])~threshold:~&quot;</span>,.)))) <span class="sc">%&gt;%</span></span>
<span id="cb81-9"><a href="figure-3.html#cb81-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> weighted, <span class="at">y =</span> Value<span class="co">#, fill = weighted</span></span>
<span id="cb81-10"><a href="figure-3.html#cb81-10" aria-hidden="true" tabindex="-1"></a>             )</span>
<span id="cb81-11"><a href="figure-3.html#cb81-11" aria-hidden="true" tabindex="-1"></a>         )<span class="sc">+</span></span>
<span id="cb81-12"><a href="figure-3.html#cb81-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add red line for genome extent in lowest row</span></span>
<span id="cb81-13"><a href="figure-3.html#cb81-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">data =</span> <span class="fu">tibble</span>(<span class="at">variable =</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="st">&#39;atop(Cummulative~Region,Length~(Mb))&#39;</span>,</span>
<span id="cb81-14"><a href="figure-3.html#cb81-14" aria-hidden="true" tabindex="-1"></a>                                               <span class="st">&#39;atop(Average~Region,Length~(kb))&#39;</span>,</span>
<span id="cb81-15"><a href="figure-3.html#cb81-15" aria-hidden="true" tabindex="-1"></a>                                               <span class="st">&#39;atop(Number~of,Regions)&#39;</span>),</span>
<span id="cb81-16"><a href="figure-3.html#cb81-16" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;atop(Number~of,Regions)&#39;</span>,</span>
<span id="cb81-17"><a href="figure-3.html#cb81-17" aria-hidden="true" tabindex="-1"></a>                                                        <span class="st">&#39;atop(Average~Region,Length~(kb))&#39;</span>,</span>
<span id="cb81-18"><a href="figure-3.html#cb81-18" aria-hidden="true" tabindex="-1"></a>                                                        <span class="st">&#39;atop(Cummulative~Region,Length~(Mb))&#39;</span>)),</span>
<span id="cb81-19"><a href="figure-3.html#cb81-19" aria-hidden="true" tabindex="-1"></a>                           <span class="at">y =</span> <span class="fu">c</span>(<span class="dv">559649677</span><span class="sc">/</span>(<span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>),<span class="cn">NA</span>,<span class="cn">NA</span>)),</span>
<span id="cb81-20"><a href="figure-3.html#cb81-20" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(<span class="at">yintercept =</span> y),</span>
<span id="cb81-21"><a href="figure-3.html#cb81-21" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="fu">rgb</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,.<span class="dv">25</span>))<span class="sc">+</span></span>
<span id="cb81-22"><a href="figure-3.html#cb81-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add data points</span></span>
<span id="cb81-23"><a href="figure-3.html#cb81-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> plot_size,</span>
<span id="cb81-24"><a href="figure-3.html#cb81-24" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> plot_clr<span class="co">#, shape = 21</span></span>
<span id="cb81-25"><a href="figure-3.html#cb81-25" aria-hidden="true" tabindex="-1"></a>             )<span class="sc">+</span></span>
<span id="cb81-26"><a href="figure-3.html#cb81-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># define plot stucture</span></span>
<span id="cb81-27"><a href="figure-3.html#cb81-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(variable<span class="sc">~</span>threshold_value,</span>
<span id="cb81-28"><a href="figure-3.html#cb81-28" aria-hidden="true" tabindex="-1"></a>             <span class="at">scale=</span><span class="st">&#39;free&#39;</span>,</span>
<span id="cb81-29"><a href="figure-3.html#cb81-29" aria-hidden="true" tabindex="-1"></a>             <span class="at">switch =</span> <span class="st">&#39;y&#39;</span>,</span>
<span id="cb81-30"><a href="figure-3.html#cb81-30" aria-hidden="true" tabindex="-1"></a>             <span class="at">labeller =</span> label_parsed)<span class="sc">+</span></span>
<span id="cb81-31"><a href="figure-3.html#cb81-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># configure scales</span></span>
<span id="cb81-32"><a href="figure-3.html#cb81-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_fill_gradientn(name = expression(weighted~italic(F[ST])),</span></span>
<span id="cb81-33"><a href="figure-3.html#cb81-33" aria-hidden="true" tabindex="-1"></a>  <span class="co">#                      colours = hypogen::hypo_clr_LGs[1:24] %&gt;% clr_lighten(factor = .3))+</span></span>
<span id="cb81-34"><a href="figure-3.html#cb81-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="fu">expression</span>(Whole<span class="sc">-</span>genome<span class="sc">~</span>differentiation<span class="sc">~</span>(weighted<span class="sc">~</span><span class="fu">italic</span>(F[ST]))),</span>
<span id="cb81-35"><a href="figure-3.html#cb81-35" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>,.<span class="dv">05</span>,.<span class="dv">1</span>),</span>
<span id="cb81-36"><a href="figure-3.html#cb81-36" aria-hidden="true" tabindex="-1"></a>                     <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span>.<span class="dv">00025</span>,.<span class="dv">10025</span>),</span>
<span id="cb81-37"><a href="figure-3.html#cb81-37" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;0&quot;</span>, <span class="st">&quot;0.05&quot;</span>, <span class="st">&quot;0.1&quot;</span>))<span class="sc">+</span></span>
<span id="cb81-38"><a href="figure-3.html#cb81-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># configure legend</span></span>
<span id="cb81-39"><a href="figure-3.html#cb81-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_colorbar</span>(<span class="at">barwidth =</span> <span class="fu">unit</span>(<span class="dv">150</span>, <span class="st">&quot;pt&quot;</span>),</span>
<span id="cb81-40"><a href="figure-3.html#cb81-40" aria-hidden="true" tabindex="-1"></a>                               <span class="at">label.position =</span> <span class="st">&quot;top&quot;</span>,</span>
<span id="cb81-41"><a href="figure-3.html#cb81-41" aria-hidden="true" tabindex="-1"></a>                               <span class="at">barheight =</span> <span class="fu">unit</span>(<span class="dv">5</span>,<span class="st">&quot;pt&quot;</span>)))<span class="sc">+</span></span>
<span id="cb81-42"><a href="figure-3.html#cb81-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># tweak plot apperance</span></span>
<span id="cb81-43"><a href="figure-3.html#cb81-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()<span class="sc">+</span></span>
<span id="cb81-44"><a href="figure-3.html#cb81-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> plot_text_size_small,</span>
<span id="cb81-45"><a href="figure-3.html#cb81-45" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">color =</span> <span class="fu">rgb</span>(.<span class="dv">6</span>,.<span class="dv">6</span>,.<span class="dv">6</span>)),</span>
<span id="cb81-46"><a href="figure-3.html#cb81-46" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb81-47"><a href="figure-3.html#cb81-47" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> .<span class="dv">5</span>, <span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb81-48"><a href="figure-3.html#cb81-48" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="dv">2</span>),</span>
<span id="cb81-49"><a href="figure-3.html#cb81-49" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="fu">rgb</span>(.<span class="dv">95</span>,.<span class="dv">95</span>,.<span class="dv">95</span>,.<span class="dv">5</span>),</span>
<span id="cb81-50"><a href="figure-3.html#cb81-50" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">color =</span> <span class="fu">rgb</span>(.<span class="dv">9</span>,.<span class="dv">9</span>,.<span class="dv">9</span>,.<span class="dv">5</span>),</span>
<span id="cb81-51"><a href="figure-3.html#cb81-51" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">size =</span> .<span class="dv">3</span>),</span>
<span id="cb81-52"><a href="figure-3.html#cb81-52" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb81-53"><a href="figure-3.html#cb81-53" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">size =</span> plot_lwd),</span>
<span id="cb81-54"><a href="figure-3.html#cb81-54" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb81-55"><a href="figure-3.html#cb81-55" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> plot_text_size),</span>
<span id="cb81-56"><a href="figure-3.html#cb81-56" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.direction =</span> <span class="st">&quot;horizontal&quot;</span>,</span>
<span id="cb81-57"><a href="figure-3.html#cb81-57" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.placement =</span> <span class="st">&#39;outside&#39;</span>,</span>
<span id="cb81-58"><a href="figure-3.html#cb81-58" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> plot_text_size),</span>
<span id="cb81-59"><a href="figure-3.html#cb81-59" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> plot_text_size),</span>
<span id="cb81-60"><a href="figure-3.html#cb81-60" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb81-61"><a href="figure-3.html#cb81-61" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.background =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p>Finally, we can export Figure 3.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="figure-3.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># export figure 3</span></span>
<span id="cb82-2"><a href="figure-3.html#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="fu">hypo_save</span>(<span class="at">filename =</span> <span class="st">&#39;figures/F3.pdf&#39;</span>,</span>
<span id="cb82-3"><a href="figure-3.html#cb82-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">plot =</span> p_done,</span>
<span id="cb82-4"><a href="figure-3.html#cb82-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">width =</span> .<span class="dv">52</span> <span class="sc">*</span> f_width,</span>
<span id="cb82-5"><a href="figure-3.html#cb82-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">height =</span> .<span class="dv">52</span> <span class="sc">*</span> f_width,</span>
<span id="cb82-6"><a href="figure-3.html#cb82-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">device =</span> cairo_pdf,</span>
<span id="cb82-7"><a href="figure-3.html#cb82-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">comment =</span> plot_comment,</span>
<span id="cb82-8"><a href="figure-3.html#cb82-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">bg =</span> <span class="st">&quot;transparent&quot;</span>)</span></code></pre></div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="figure-2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-4.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
