<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>17 Figure 1 | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="17 Figure 1 | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="17 Figure 1 | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench and Martin Helmkampf" />


<meta name="date" content="2021-04-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="git-15-data-visualization.html"/>
<link rel="next" href="figure-2.html"/>
<script src="libs/header-attrs-2.7.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#r-setup"><i class="fa fa-check"></i><b>1.4</b> R setup</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a>
<ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a>
<ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i> &amp; pi)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (topology weighting)</a>
<ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (Principal Component Analysis)</a>
<ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#details-of-analysis_pca.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_pca.nf</code></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#setup-2"><i class="fa fa-check"></i><b>8.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-4"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-5"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-6"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html"><i class="fa fa-check"></i><b>12</b> (git 11) Analysis IX (Allele Age)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#summary-10"><i class="fa fa-check"></i><b>12.1</b> Summary</a></li>
<li class="chapter" data-level="12.2" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#details-of-analysis_allele_age.nf"><i class="fa fa-check"></i><b>12.2</b> Details of <code>analysis_allele_age.nf</code></a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#setup-3"><i class="fa fa-check"></i><b>12.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html"><i class="fa fa-check"></i><b>13</b> (git 12) Analysis X (<i>F<sub>ST</sub></i> Permutation Test)</a>
<ul>
<li class="chapter" data-level="13.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#summary-11"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#details-of-analysis_fst_sign.nf"><i class="fa fa-check"></i><b>13.2</b> Details of <code>analysis_fst_sign.nf</code></a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#setup-4"><i class="fa fa-check"></i><b>13.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html"><i class="fa fa-check"></i><b>14</b> (git 13) Analysis XI (Whole Genome Phylogenies)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#summary-12"><i class="fa fa-check"></i><b>14.1</b> Summary</a></li>
<li class="chapter" data-level="14.2" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#details-of-analysis_phylo_whg.nf"><i class="fa fa-check"></i><b>14.2</b> Details of <code>analysis_phylo_whg.nf</code></a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#setup-5"><i class="fa fa-check"></i><b>14.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html"><i class="fa fa-check"></i><b>15</b> (git 14) Analysis XII (Outlier Region Phylogenies)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#summary-13"><i class="fa fa-check"></i><b>15.1</b> Summary</a></li>
<li class="chapter" data-level="15.2" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#details-of-analysis_phylo_regions.nf"><i class="fa fa-check"></i><b>15.2</b> Details of <code>analysis_phylo_regions.nf</code></a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#setup-6"><i class="fa fa-check"></i><b>15.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="git-15-data-visualization.html"><a href="git-15-data-visualization.html"><i class="fa fa-check"></i><b>16</b> (git 15) Data Visualization</a></li>
<li class="chapter" data-level="17" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>17</b> Figure 1</a>
<ul>
<li class="chapter" data-level="17.1" data-path="figure-1.html"><a href="figure-1.html#summary-14"><i class="fa fa-check"></i><b>17.1</b> Summary</a></li>
<li class="chapter" data-level="17.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>17.2</b> Details of <code>plot_F1.R</code></a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>17.2.1</b> Config</a></li>
<li class="chapter" data-level="17.2.2" data-path="figure-1.html"><a href="figure-1.html#fish-tree-of-life-subtree"><i class="fa fa-check"></i><b>17.2.2</b> Fish Tree Of Life Subtree</a></li>
<li class="chapter" data-level="17.2.3" data-path="figure-1.html"><a href="figure-1.html#genetic-differentiation-boxplots"><i class="fa fa-check"></i><b>17.2.3</b> Genetic differentiation boxplots</a></li>
<li class="chapter" data-level="17.2.4" data-path="figure-1.html"><a href="figure-1.html#principal-component-analyses"><i class="fa fa-check"></i><b>17.2.4</b> Principal Component Analyses</a></li>
<li class="chapter" data-level="17.2.5" data-path="figure-1.html"><a href="figure-1.html#legend-and-final-figure"><i class="fa fa-check"></i><b>17.2.5</b> Legend and Final Figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>18</b> Figure 2</a>
<ul>
<li class="chapter" data-level="18.1" data-path="figure-2.html"><a href="figure-2.html#summary-15"><i class="fa fa-check"></i><b>18.1</b> Summary</a></li>
<li class="chapter" data-level="18.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>18.2</b> Details of <code>plot_F2.R</code></a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>18.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>19</b> Figure 3</a>
<ul>
<li class="chapter" data-level="19.1" data-path="figure-3.html"><a href="figure-3.html#summary-16"><i class="fa fa-check"></i><b>19.1</b> Summary</a></li>
<li class="chapter" data-level="19.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>19.2</b> Details of <code>plot_F3.R</code></a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>19.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>20</b> Figure 4</a>
<ul>
<li class="chapter" data-level="20.1" data-path="figure-4.html"><a href="figure-4.html#summary-17"><i class="fa fa-check"></i><b>20.1</b> Summary</a></li>
<li class="chapter" data-level="20.2" data-path="figure-4.html"><a href="figure-4.html#details-of-plot_f4.r"><i class="fa fa-check"></i><b>20.2</b> Details of <code>plot_F4.R</code></a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="figure-4.html"><a href="figure-4.html#config-3"><i class="fa fa-check"></i><b>20.2.1</b> Config</a></li>
<li class="chapter" data-level="20.2.2" data-path="figure-4.html"><a href="figure-4.html#data-import"><i class="fa fa-check"></i><b>20.2.2</b> Data import</a></li>
<li class="chapter" data-level="20.2.3" data-path="figure-4.html"><a href="figure-4.html#plotting"><i class="fa fa-check"></i><b>20.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>21</b> Figure 5</a>
<ul>
<li class="chapter" data-level="21.1" data-path="figure-5.html"><a href="figure-5.html#summary-18"><i class="fa fa-check"></i><b>21.1</b> Summary</a></li>
<li class="chapter" data-level="21.2" data-path="figure-5.html"><a href="figure-5.html#details-of-plot_f5.r"><i class="fa fa-check"></i><b>21.2</b> Details of <code>plot_F5.R</code></a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="figure-5.html"><a href="figure-5.html#config-4"><i class="fa fa-check"></i><b>21.2.1</b> Config</a></li>
<li class="chapter" data-level="21.2.2" data-path="figure-5.html"><a href="figure-5.html#data-import-1"><i class="fa fa-check"></i><b>21.2.2</b> Data import</a></li>
<li class="chapter" data-level="21.2.3" data-path="figure-5.html"><a href="figure-5.html#plotting-1"><i class="fa fa-check"></i><b>21.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>22</b> Figure 6</a>
<ul>
<li class="chapter" data-level="22.1" data-path="figure-6.html"><a href="figure-6.html#summary-19"><i class="fa fa-check"></i><b>22.1</b> Summary</a></li>
<li class="chapter" data-level="22.2" data-path="figure-6.html"><a href="figure-6.html#details-of-plot_f6.r"><i class="fa fa-check"></i><b>22.2</b> Details of <code>plot_F6.R</code></a>
<ul>
<li class="chapter" data-level="22.2.1" data-path="figure-6.html"><a href="figure-6.html#config-5"><i class="fa fa-check"></i><b>22.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html"><i class="fa fa-check"></i><b>23</b> Supplementary Figure 1</a>
<ul>
<li class="chapter" data-level="23.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#summary-20"><i class="fa fa-check"></i><b>23.1</b> Summary</a></li>
<li class="chapter" data-level="23.2" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#details-of-plot_sf1.r"><i class="fa fa-check"></i><b>23.2</b> Details of <code>plot_SF1.R</code></a>
<ul>
<li class="chapter" data-level="23.2.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#config-6"><i class="fa fa-check"></i><b>23.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html"><i class="fa fa-check"></i><b>24</b> Supplementary Figure 2</a>
<ul>
<li class="chapter" data-level="24.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#summary-21"><i class="fa fa-check"></i><b>24.1</b> Summary</a></li>
<li class="chapter" data-level="24.2" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#details-of-plot_sf2.r"><i class="fa fa-check"></i><b>24.2</b> Details of <code>plot_SF2.R</code></a>
<ul>
<li class="chapter" data-level="24.2.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#config-7"><i class="fa fa-check"></i><b>24.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html"><i class="fa fa-check"></i><b>25</b> Supplementary Figure 3</a>
<ul>
<li class="chapter" data-level="25.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#summary-22"><i class="fa fa-check"></i><b>25.1</b> Summary</a></li>
<li class="chapter" data-level="25.2" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#details-of-plot_sf3.r"><i class="fa fa-check"></i><b>25.2</b> Details of <code>plot_SF3.R</code></a>
<ul>
<li class="chapter" data-level="25.2.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#config-8"><i class="fa fa-check"></i><b>25.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html"><i class="fa fa-check"></i><b>26</b> Supplementary Figure 4</a>
<ul>
<li class="chapter" data-level="26.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#summary-23"><i class="fa fa-check"></i><b>26.1</b> Summary</a></li>
<li class="chapter" data-level="26.2" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#details-of-plot_sf4.r"><i class="fa fa-check"></i><b>26.2</b> Details of <code>plot_SF4.R</code></a>
<ul>
<li class="chapter" data-level="26.2.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#config-9"><i class="fa fa-check"></i><b>26.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html"><i class="fa fa-check"></i><b>27</b> Supplementary Figure 5</a>
<ul>
<li class="chapter" data-level="27.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#summary-24"><i class="fa fa-check"></i><b>27.1</b> Summary</a></li>
<li class="chapter" data-level="27.2" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#details-of-plot_sf5.r"><i class="fa fa-check"></i><b>27.2</b> Details of <code>plot_SF5.R</code></a>
<ul>
<li class="chapter" data-level="27.2.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#config-10"><i class="fa fa-check"></i><b>27.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html"><i class="fa fa-check"></i><b>28</b> Supplementary Figure 6</a>
<ul>
<li class="chapter" data-level="28.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#summary-25"><i class="fa fa-check"></i><b>28.1</b> Summary</a></li>
<li class="chapter" data-level="28.2" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#details-of-plot_sf6.r"><i class="fa fa-check"></i><b>28.2</b> Details of <code>plot_SF6.R</code></a>
<ul>
<li class="chapter" data-level="28.2.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#config-11"><i class="fa fa-check"></i><b>28.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="29" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html"><i class="fa fa-check"></i><b>29</b> Supplementary Figure 7</a>
<ul>
<li class="chapter" data-level="29.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#summary-26"><i class="fa fa-check"></i><b>29.1</b> Summary</a></li>
<li class="chapter" data-level="29.2" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#details-of-plot_sf7.r"><i class="fa fa-check"></i><b>29.2</b> Details of <code>plot_SF7.R</code></a>
<ul>
<li class="chapter" data-level="29.2.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#config-12"><i class="fa fa-check"></i><b>29.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="30" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html"><i class="fa fa-check"></i><b>30</b> Supplementary Figure 8</a>
<ul>
<li class="chapter" data-level="30.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#summary-27"><i class="fa fa-check"></i><b>30.1</b> Summary</a></li>
<li class="chapter" data-level="30.2" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#details-of-plot_sf8.r"><i class="fa fa-check"></i><b>30.2</b> Details of <code>plot_SF8.R</code></a>
<ul>
<li class="chapter" data-level="30.2.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#config-13"><i class="fa fa-check"></i><b>30.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="31" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html"><i class="fa fa-check"></i><b>31</b> Supplementary Figure 9</a>
<ul>
<li class="chapter" data-level="31.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#summary-28"><i class="fa fa-check"></i><b>31.1</b> Summary</a></li>
<li class="chapter" data-level="31.2" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#details-of-plot_sf9.r"><i class="fa fa-check"></i><b>31.2</b> Details of <code>plot_SF9.R</code></a>
<ul>
<li class="chapter" data-level="31.2.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#config-14"><i class="fa fa-check"></i><b>31.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="32" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html"><i class="fa fa-check"></i><b>32</b> Supplementary Figure 10</a>
<ul>
<li class="chapter" data-level="32.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#summary-29"><i class="fa fa-check"></i><b>32.1</b> Summary</a></li>
<li class="chapter" data-level="32.2" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#details-of-plot_sf10.r"><i class="fa fa-check"></i><b>32.2</b> Details of <code>plot_SF10.R</code></a>
<ul>
<li class="chapter" data-level="32.2.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#config-15"><i class="fa fa-check"></i><b>32.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html"><i class="fa fa-check"></i><b>33</b> Supplementary Figure 11</a>
<ul>
<li class="chapter" data-level="33.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#summary-30"><i class="fa fa-check"></i><b>33.1</b> Summary</a></li>
<li class="chapter" data-level="33.2" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#details-of-plot_sf11.r"><i class="fa fa-check"></i><b>33.2</b> Details of <code>plot_SF11.R</code></a>
<ul>
<li class="chapter" data-level="33.2.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#config-16"><i class="fa fa-check"></i><b>33.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html"><i class="fa fa-check"></i><b>34</b> Supplementary Figure 12</a>
<ul>
<li class="chapter" data-level="34.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#summary-31"><i class="fa fa-check"></i><b>34.1</b> Summary</a></li>
<li class="chapter" data-level="34.2" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#details-of-plot_sf12.r"><i class="fa fa-check"></i><b>34.2</b> Details of <code>plot_SF12.R</code></a>
<ul>
<li class="chapter" data-level="34.2.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#config-17"><i class="fa fa-check"></i><b>34.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="35" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html"><i class="fa fa-check"></i><b>35</b> Supplementary Figure 13</a>
<ul>
<li class="chapter" data-level="35.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#summary-32"><i class="fa fa-check"></i><b>35.1</b> Summary</a></li>
<li class="chapter" data-level="35.2" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#details-of-plot_sf13.r"><i class="fa fa-check"></i><b>35.2</b> Details of <code>plot_SF13.R</code></a>
<ul>
<li class="chapter" data-level="35.2.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#config-18"><i class="fa fa-check"></i><b>35.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="36" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html"><i class="fa fa-check"></i><b>36</b> Supplementary Figure 14</a>
<ul>
<li class="chapter" data-level="36.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#summary-33"><i class="fa fa-check"></i><b>36.1</b> Summary</a></li>
<li class="chapter" data-level="36.2" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#details-of-plot_sf14.r"><i class="fa fa-check"></i><b>36.2</b> Details of <code>plot_SF14.R</code></a>
<ul>
<li class="chapter" data-level="36.2.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#config-19"><i class="fa fa-check"></i><b>36.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="37" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html"><i class="fa fa-check"></i><b>37</b> Supplementary Figure 15</a>
<ul>
<li class="chapter" data-level="37.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#summary-34"><i class="fa fa-check"></i><b>37.1</b> Summary</a></li>
<li class="chapter" data-level="37.2" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#details-of-plot_sf15.r"><i class="fa fa-check"></i><b>37.2</b> Details of <code>plot_SF15.R</code></a>
<ul>
<li class="chapter" data-level="37.2.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#config-20"><i class="fa fa-check"></i><b>37.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="38" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html"><i class="fa fa-check"></i><b>38</b> Supplementary Figure 16</a>
<ul>
<li class="chapter" data-level="38.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#summary-35"><i class="fa fa-check"></i><b>38.1</b> Summary</a></li>
<li class="chapter" data-level="38.2" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#details-of-plot_sf16.r"><i class="fa fa-check"></i><b>38.2</b> Details of <code>plot_SF16.R</code></a>
<ul>
<li class="chapter" data-level="38.2.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#config-21"><i class="fa fa-check"></i><b>38.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="figure-1" class="section level1" number="17">
<h1><span class="header-section-number">17</span> Figure 1</h1>
<div id="summary-14" class="section level2" number="17.1">
<h2><span class="header-section-number">17.1</span> Summary</h2>
<p>This is the accessory documentation of Figure 1.</p>
<blockquote>
<p>Note that this script depends on results that are created within the script <code>R/figs/plot_SF15.R</code> (specifically <code>2_analysis/summaries/fst_permutation_summary.ts</code>) and thus needs to be run after this script has finished.</p>
</blockquote>
<p>The Figure can be recreated by running the <strong>R</strong> script <code>plot_F1.R</code> from a (<code>bash</code> terminal):</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb23-1"><a href="figure-1.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span></span>
<span id="cb23-2"><a href="figure-1.html#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="figure-1.html#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> --vanilla R/fig/plot_F1.R 2_analysis/dxy/50k/ 2_analysis/fst/50k/</span></code></pre></div>
</div>
<div id="details-of-plot_f1.r" class="section level2" number="17.2">
<h2><span class="header-section-number">17.2</span> Details of <code>plot_F1.R</code></h2>
<p>In the following, the individual steps of the R script are documented.
It is an executable R script that depends on the accessory R package <a href="https://k-hench.github.io/GenomicOriginsScripts"><strong>GenomicOriginsScripts</strong></a>, <a href="https://cran.r-project.org/web/packages/BAMMtools/"><strong>BAMMtools</strong></a> and on the package <a href="https://k-hench.github.io/hypoimg"><strong>hypoimg</strong></a>.</p>
<div id="config" class="section level3" number="17.2.1">
<h3><span class="header-section-number">17.2.1</span> Config</h3>
<p>The scripts start with a header that contains copy &amp; paste templates to execute interactively or debug the script:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="figure-1.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/usr/bin/env Rscript</span></span>
<span id="cb24-2"><a href="figure-1.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb24-3"><a href="figure-1.html#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Context: this script depends on the input file 2_analysis/summaries/fst_permutation_summary.tsv</span></span>
<span id="cb24-4"><a href="figure-1.html#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co">#          which is created by R/figs/plot_SF15.R</span></span>
<span id="cb24-5"><a href="figure-1.html#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb24-6"><a href="figure-1.html#cb24-6" aria-hidden="true" tabindex="-1"></a><span class="co"># run from terminal:</span></span>
<span id="cb24-7"><a href="figure-1.html#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Rscript --vanilla R/fig/plot_F1.R \</span></span>
<span id="cb24-8"><a href="figure-1.html#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="co">#    2_analysis/fst/50k/ 2_analysis/summaries/fst_globals.txt 2_analysis/summaries/fst_permutation_summary.tsv</span></span>
<span id="cb24-9"><a href="figure-1.html#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="co"># ===============================================================</span></span>
<span id="cb24-10"><a href="figure-1.html#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="co"># This script produces Figure 1 of the study &quot;Ancestral variation, hybridization and modularity</span></span>
<span id="cb24-11"><a href="figure-1.html#cb24-11" aria-hidden="true" tabindex="-1"></a><span class="co"># fuel a marine radiation&quot; by Hench, Helmkampf, McMillan and Puebla</span></span>
<span id="cb24-12"><a href="figure-1.html#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="co"># ---------------------------------------------------------------</span></span>
<span id="cb24-13"><a href="figure-1.html#cb24-13" aria-hidden="true" tabindex="-1"></a><span class="co"># ===============================================================</span></span>
<span id="cb24-14"><a href="figure-1.html#cb24-14" aria-hidden="true" tabindex="-1"></a><span class="co"># args &lt;- c(&#39;2_analysis/fst/50k/&#39;, &#39;2_analysis/summaries/fst_globals.txt&#39;, &#39;2_analysis/summaries/fst_permutation_summary.tsv&#39;)</span></span>
<span id="cb24-15"><a href="figure-1.html#cb24-15" aria-hidden="true" tabindex="-1"></a><span class="co"># script_name &lt;- &quot;R/fig/plot_F1.R&quot;</span></span></code></pre></div>
<p>The next section processes the input from the command line.
It stores the arguments in the vector <code>args</code>.
The needed R packages are loaded and the script name and the current working directory are stored inside variables (<code>script_name</code>, <code>plot_comment</code>).
This information will later be written into the meta data of the figure to help us tracing back the scripts that created the figures in the future.</p>
<p>Then we drop all the imported information besides the arguments following the script name and print the information to the terminal.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="figure-1.html#cb25-1" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly=</span><span class="cn">FALSE</span>)</span>
<span id="cb25-2"><a href="figure-1.html#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co"># setup -----------------------</span></span>
<span id="cb25-3"><a href="figure-1.html#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicOriginsScripts)</span>
<span id="cb25-4"><a href="figure-1.html#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hypoimg)</span>
<span id="cb25-5"><a href="figure-1.html#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hypogen)</span>
<span id="cb25-6"><a href="figure-1.html#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb25-7"><a href="figure-1.html#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape)</span>
<span id="cb25-8"><a href="figure-1.html#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb25-9"><a href="figure-1.html#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidygraph)</span>
<span id="cb25-10"><a href="figure-1.html#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb25-11"><a href="figure-1.html#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="figure-1.html#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb25-13"><a href="figure-1.html#cb25-13" aria-hidden="true" tabindex="-1"></a>script_name <span class="ot">&lt;-</span> args[<span class="dv">5</span>] <span class="sc">%&gt;%</span></span>
<span id="cb25-14"><a href="figure-1.html#cb25-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_remove</span>(., <span class="st">&#39;--file=&#39;</span>)</span>
<span id="cb25-15"><a href="figure-1.html#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="figure-1.html#cb25-16" aria-hidden="true" tabindex="-1"></a>plot_comment <span class="ot">&lt;-</span> script_name <span class="sc">%&gt;%</span></span>
<span id="cb25-17"><a href="figure-1.html#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_c</span>(<span class="st">&#39;mother-script = &#39;</span>, <span class="fu">getwd</span>(), <span class="st">&#39;/&#39;</span>, .)</span>
<span id="cb25-18"><a href="figure-1.html#cb25-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-19"><a href="figure-1.html#cb25-19" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">process_input</span>(script_name, args)</span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="figure-1.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ── Script: R/fig/plot_F1.R ──────────────────────────────────────────────</span></span>
<span id="cb26-2"><a href="figure-1.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Parameters read:</span></span>
<span id="cb26-3"><a href="figure-1.html#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  ★ 1: 2_analysis/fst/50k/</span></span>
<span id="cb26-4"><a href="figure-1.html#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  ★ 2: 2_analysis/summaries/fst_globals.txt</span></span>
<span id="cb26-5"><a href="figure-1.html#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  ★ 3: 2_analysis/summaries/fst_permutation_summary.tsv</span></span>
<span id="cb26-6"><a href="figure-1.html#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ─────────────────────────────────────────── /current/working/directory ──</span></span></code></pre></div>
<p>The directories for the different data types are received and stored in respective variables.
Also, we set a few parameters for the plot layout:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="figure-1.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># config -----------------------</span></span>
<span id="cb27-2"><a href="figure-1.html#cb27-2" aria-hidden="true" tabindex="-1"></a>fst_dir <span class="ot">&lt;-</span> <span class="fu">as.character</span>(args[<span class="dv">1</span>])</span>
<span id="cb27-3"><a href="figure-1.html#cb27-3" aria-hidden="true" tabindex="-1"></a>fst_globals <span class="ot">&lt;-</span> <span class="fu">as.character</span>(args[<span class="dv">2</span>])</span>
<span id="cb27-4"><a href="figure-1.html#cb27-4" aria-hidden="true" tabindex="-1"></a>fst_permutation_file <span class="ot">&lt;-</span> <span class="fu">as.character</span>(args[<span class="dv">3</span>])</span>
<span id="cb27-5"><a href="figure-1.html#cb27-5" aria-hidden="true" tabindex="-1"></a>wdh <span class="ot">&lt;-</span> .<span class="dv">3</span>          <span class="co"># The width of the boxplots</span></span>
<span id="cb27-6"><a href="figure-1.html#cb27-6" aria-hidden="true" tabindex="-1"></a>scaler <span class="ot">&lt;-</span> <span class="dv">20</span>       <span class="co"># the ratio of the Fst and the dxy axis (legacy - not really needed anymore)</span></span>
<span id="cb27-7"><a href="figure-1.html#cb27-7" aria-hidden="true" tabindex="-1"></a>clr_sec <span class="ot">&lt;-</span> <span class="st">&#39;gray&#39;</span>  <span class="co"># the color of the secondary axis (dxy)</span></span></code></pre></div>
</div>
<div id="fish-tree-of-life-subtree" class="section level3" number="17.2.2">
<h3><span class="header-section-number">17.2.2</span> Fish Tree Of Life Subtree</h3>
<p>The Figure 1 is assembled from five panels (a-e) which are prepared independently and combined at the end of the R script.</p>
<p>The first panel (a) contains ar re-drawing of a of subset of the data from <span class="citation">(<a href="supplementary-figure-16.html#ref-rabosky18data" role="doc-biblioref">Rabosky et al. 2019</a>)</span> and also uses builds upon the <strong>R</strong> code from the script <code>scripts/main figures/Figure3.R</code> of their <a href="https://datadryad.org/stash/dataset/doi:10.5061/dryad.fc71cp4">Dryad repository</a>.</p>
<p>The code of <span class="citation">(<a href="supplementary-figure-16.html#ref-rabosky18data" role="doc-biblioref">Rabosky et al. 2019</a>)</span> relies on the packages {BAMMtools} and {geiger} which are loaded additionally to those loaded during the start-up (this is to keep the modified code as close to the original as possible).</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="figure-1.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># start script -------------------</span></span>
<span id="cb28-2"><a href="figure-1.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="co"># === fig 1. panel a: modified plugin from ressources/Rabosky_etal_2018/scripts/main figures/Figure3.R =============</span></span>
<span id="cb28-3"><a href="figure-1.html#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BAMMtools)</span>
<span id="cb28-4"><a href="figure-1.html#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(geiger)</span></code></pre></div>
<p>To be able to handle the tree-plot within a <strong>ggplot</strong> framework, we also need the package {ggplotify} to turn the plot into a grid object.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="figure-1.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplotify)</span></code></pre></div>
<p>Then, the path to the downloaded Dryad repository is specified and the original helper script <code>PlottingFunctions.R</code> from <span class="citation">(<a href="supplementary-figure-16.html#ref-rabosky18data" role="doc-biblioref">Rabosky et al. 2019</a>)</span> is sourced.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="figure-1.html#cb30-1" aria-hidden="true" tabindex="-1"></a>basepath <span class="ot">&lt;-</span> <span class="st">&#39;ressources/Rabosky_etal_2018/&#39;</span></span>
<span id="cb30-2"><a href="figure-1.html#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">paste0</span>(basepath, <span class="st">&quot;scripts/supporting_fxns/PlottingFunctions.R&quot;</span>))</span></code></pre></div>
<p>Next, the paths of Fish Tree Of Life (FToL) data are specified and the tree is loaded and prepared according to the original script by <span class="citation">(<a href="supplementary-figure-16.html#ref-rabosky18data" role="doc-biblioref">Rabosky et al. 2019</a>)</span>.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="figure-1.html#cb31-1" aria-hidden="true" tabindex="-1"></a>eventdata_vr <span class="ot">&lt;-</span> <span class="fu">paste0</span>(basepath, <span class="st">&quot;dataFiles/bamm_results/12k_tv1/event_data_thinned.csv&quot;</span>)</span>
<span id="cb31-2"><a href="figure-1.html#cb31-2" aria-hidden="true" tabindex="-1"></a>eventdata_cr <span class="ot">&lt;-</span> <span class="fu">paste0</span>(basepath, <span class="st">&quot;dataFiles/bamm_results/12k_tc1/event_data_thinned.csv&quot;</span>)</span>
<span id="cb31-3"><a href="figure-1.html#cb31-3" aria-hidden="true" tabindex="-1"></a>treefile <span class="ot">&lt;-</span> <span class="fu">paste0</span>(basepath, <span class="st">&quot;dataFiles/bamm_results/12k_tv1/bigfish_no_outgroup.tre&quot;</span>)</span>
<span id="cb31-4"><a href="figure-1.html#cb31-4" aria-hidden="true" tabindex="-1"></a>fspdata <span class="ot">&lt;-</span> <span class="fu">paste0</span>(basepath, <span class="st">&quot;dataFiles/rate_lat_stats_by_sp_fixed0.5.csv&quot;</span>)</span>
<span id="cb31-5"><a href="figure-1.html#cb31-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-6"><a href="figure-1.html#cb31-6" aria-hidden="true" tabindex="-1"></a>anadromous <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb31-7"><a href="figure-1.html#cb31-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-8"><a href="figure-1.html#cb31-8" aria-hidden="true" tabindex="-1"></a>vx <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(treefile)</span>
<span id="cb31-9"><a href="figure-1.html#cb31-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-10"><a href="figure-1.html#cb31-10" aria-hidden="true" tabindex="-1"></a><span class="co"># node rotations for plotting:</span></span>
<span id="cb31-11"><a href="figure-1.html#cb31-11" aria-hidden="true" tabindex="-1"></a>rset <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">15447</span>, <span class="dv">15708</span><span class="sc">:</span><span class="dv">15719</span>)</span>
<span id="cb31-12"><a href="figure-1.html#cb31-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(rset)){</span>
<span id="cb31-13"><a href="figure-1.html#cb31-13" aria-hidden="true" tabindex="-1"></a>  vx <span class="ot">&lt;-</span> <span class="fu">rotate</span>(vx, <span class="at">node =</span> rset[i])</span>
<span id="cb31-14"><a href="figure-1.html#cb31-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-15"><a href="figure-1.html#cb31-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-16"><a href="figure-1.html#cb31-16" aria-hidden="true" tabindex="-1"></a>vx <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="at">text =</span> <span class="fu">write.tree</span>(vx))</span>
<span id="cb31-17"><a href="figure-1.html#cb31-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-18"><a href="figure-1.html#cb31-18" aria-hidden="true" tabindex="-1"></a>spdata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(fspdata, <span class="at">stringsAsFactors =</span> F)</span>
<span id="cb31-19"><a href="figure-1.html#cb31-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(spdata) <span class="ot">&lt;-</span> spdata<span class="sc">$</span>sp</span>
<span id="cb31-20"><a href="figure-1.html#cb31-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-21"><a href="figure-1.html#cb31-21" aria-hidden="true" tabindex="-1"></a><span class="co"># ---------------------------</span></span>
<span id="cb31-22"><a href="figure-1.html#cb31-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-23"><a href="figure-1.html#cb31-23" aria-hidden="true" tabindex="-1"></a>latvals <span class="ot">&lt;-</span> <span class="fu">abs</span>(spdata<span class="sc">$</span>lat_centroid)</span>
<span id="cb31-24"><a href="figure-1.html#cb31-24" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(latvals) <span class="ot">&lt;-</span> spdata<span class="sc">$</span>sp</span>
<span id="cb31-25"><a href="figure-1.html#cb31-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-26"><a href="figure-1.html#cb31-26" aria-hidden="true" tabindex="-1"></a>inboth <span class="ot">&lt;-</span> <span class="fu">intersect</span>(spdata<span class="sc">$</span>sp, vx<span class="sc">$</span>tip.label)</span>
<span id="cb31-27"><a href="figure-1.html#cb31-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-28"><a href="figure-1.html#cb31-28" aria-hidden="true" tabindex="-1"></a><span class="co"># 1 species is in tree, but was dropped because is synonym to another species in tree according to fishbase (Gadus_ogac matches to Gadus_macrocephalus)</span></span>
<span id="cb31-29"><a href="figure-1.html#cb31-29" aria-hidden="true" tabindex="-1"></a>inboth <span class="ot">&lt;-</span> <span class="fu">intersect</span>(inboth, spdata<span class="sc">$</span>sp[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(spdata<span class="sc">$</span>tv.lambda))]) </span>
<span id="cb31-30"><a href="figure-1.html#cb31-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-31"><a href="figure-1.html#cb31-31" aria-hidden="true" tabindex="-1"></a>latvals <span class="ot">&lt;-</span> latvals[inboth]</span>
<span id="cb31-32"><a href="figure-1.html#cb31-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-33"><a href="figure-1.html#cb31-33" aria-hidden="true" tabindex="-1"></a>edvr <span class="ot">&lt;-</span> <span class="fu">getEventData</span>(vx, eventdata_vr, <span class="at">burnin=</span><span class="dv">0</span>)</span>
<span id="cb31-34"><a href="figure-1.html#cb31-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Reading event datafile:  ressources/Rabosky_etal_2018/dataFiles/bamm_results/12k_tv1/event_data_thinned.csv </span></span>
<span id="cb31-35"><a href="figure-1.html#cb31-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      ...........</span></span>
<span id="cb31-36"><a href="figure-1.html#cb31-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Read a total of 1000 samples from posterior</span></span>
<span id="cb31-37"><a href="figure-1.html#cb31-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb31-38"><a href="figure-1.html#cb31-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Discarded as burnin: GENERATIONS &lt;  1</span></span>
<span id="cb31-39"><a href="figure-1.html#cb31-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Analyzing  1000  samples from posterior</span></span>
<span id="cb31-40"><a href="figure-1.html#cb31-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb31-41"><a href="figure-1.html#cb31-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Setting recursive sequence on tree...</span></span>
<span id="cb31-42"><a href="figure-1.html#cb31-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb31-43"><a href="figure-1.html#cb31-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Done with recursive sequence</span></span></code></pre></div>
<p>At this point we diverge from the original script and define our own subset of interest, which covers all available species of the Serraninae subfamily.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="figure-1.html#cb32-1" aria-hidden="true" tabindex="-1"></a>serranids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Hypoplectrus_gemma&quot;</span>, <span class="st">&quot;Hypoplectrus_unicolor&quot;</span>, <span class="st">&quot;Hypoplectrus_gummigutta&quot;</span>,</span>
<span id="cb32-2"><a href="figure-1.html#cb32-2" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Hypoplectrus_chlorurus&quot;</span>, <span class="st">&quot;Hypoplectrus_aberrans&quot;</span>, <span class="st">&quot;Hypoplectrus_nigricans&quot;</span>,</span>
<span id="cb32-3"><a href="figure-1.html#cb32-3" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Hypoplectrus_guttavarius&quot;</span>, <span class="st">&quot;Hypoplectrus_indigo&quot;</span>, <span class="st">&quot;Hypoplectrus_puella&quot;</span>,</span>
<span id="cb32-4"><a href="figure-1.html#cb32-4" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Serranus_tortugarum&quot;</span>, <span class="st">&quot;Serranus_tabacarius&quot;</span>, <span class="st">&quot;Schultzea_beta&quot;</span>,</span>
<span id="cb32-5"><a href="figure-1.html#cb32-5" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Diplectrum_formosum&quot;</span>, <span class="st">&quot;Diplectrum_bivittatum&quot;</span>, <span class="st">&quot;Diplectrum_pacificum&quot;</span>,</span>
<span id="cb32-6"><a href="figure-1.html#cb32-6" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Diplectrum_maximum&quot;</span>, <span class="st">&quot;Serranus_notospilus&quot;</span>, <span class="st">&quot;Serranus_phoebe&quot;</span>,</span>
<span id="cb32-7"><a href="figure-1.html#cb32-7" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Serranus_psittacinus&quot;</span>, <span class="st">&quot;Serranus_baldwini&quot;</span>, <span class="st">&quot;Serranus_tigrinus&quot;</span>,</span>
<span id="cb32-8"><a href="figure-1.html#cb32-8" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Paralabrax_albomaculatus&quot;</span>, <span class="st">&quot;Paralabrax_dewegeri&quot;</span>, <span class="st">&quot;Paralabrax_callaensis&quot;</span>,</span>
<span id="cb32-9"><a href="figure-1.html#cb32-9" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Paralabrax_loro&quot;</span>, <span class="st">&quot;Paralabrax_auroguttatus&quot;</span>, <span class="st">&quot;Paralabrax_clathratus&quot;</span>,</span>
<span id="cb32-10"><a href="figure-1.html#cb32-10" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Paralabrax_humeralis&quot;</span>, <span class="st">&quot;Paralabrax_nebulifer&quot;</span>, <span class="st">&quot;Paralabrax_maculatofasciatus&quot;</span>,</span>
<span id="cb32-11"><a href="figure-1.html#cb32-11" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Zalanthias_kelloggi&quot;</span>, <span class="st">&quot;Serranus_cabrilla&quot;</span>, <span class="st">&quot;Serranus_atricauda&quot;</span>,</span>
<span id="cb32-12"><a href="figure-1.html#cb32-12" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Serranus_scriba&quot;</span>, <span class="st">&quot;Serranus_hepatus&quot;</span>, <span class="st">&quot;Serranus_accraensis&quot;</span>,</span>
<span id="cb32-13"><a href="figure-1.html#cb32-13" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Centropristis_striata&quot;</span>, <span class="st">&quot;Chelidoperca_occipitalis&quot;</span>, <span class="st">&quot;Chelidoperca_investigatoris&quot;</span>,</span>
<span id="cb32-14"><a href="figure-1.html#cb32-14" aria-hidden="true" tabindex="-1"></a>               <span class="st">&quot;Chelidoperca_pleurospilus&quot;</span>)</span></code></pre></div>
<p>Then, the data is subset based on the selection.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="figure-1.html#cb33-1" aria-hidden="true" tabindex="-1"></a>edvr_serr <span class="ot">&lt;-</span> edvr <span class="sc">%&gt;%</span> </span>
<span id="cb33-2"><a href="figure-1.html#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subtreeBAMM</span>(<span class="at">tips =</span> serranids)</span></code></pre></div>
<p>Next, the species labels are specified and formatted (for those species that have would have an ambiguous family abbreviation based on the first letter).</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="figure-1.html#cb34-1" aria-hidden="true" tabindex="-1"></a>label_two_chars <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">italic(S.~&#39;beta&#39;)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Sc.~&#39;beta&#39;)&quot;</span>,</span>
<span id="cb34-2"><a href="figure-1.html#cb34-2" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~notospilus)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~notospilus)&quot;</span>,</span>
<span id="cb34-3"><a href="figure-1.html#cb34-3" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~phoebe)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~phoebe)&quot;</span>,</span>
<span id="cb34-4"><a href="figure-1.html#cb34-4" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~psittacinus)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~psittacinus)&quot;</span>,</span>
<span id="cb34-5"><a href="figure-1.html#cb34-5" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~baldwini)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~baldwini)&quot;</span>,</span>
<span id="cb34-6"><a href="figure-1.html#cb34-6" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~tigrinus)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~tigrinus)&quot;</span>,</span>
<span id="cb34-7"><a href="figure-1.html#cb34-7" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~cabrilla)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~cabrilla)&quot;</span>,</span>
<span id="cb34-8"><a href="figure-1.html#cb34-8" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~atricauda)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~atricauda)&quot;</span>,</span>
<span id="cb34-9"><a href="figure-1.html#cb34-9" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~scriba)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~scriba)&quot;</span>,</span>
<span id="cb34-10"><a href="figure-1.html#cb34-10" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~hepatus)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~hepatus)&quot;</span>,</span>
<span id="cb34-11"><a href="figure-1.html#cb34-11" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~accraensis)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~accraensis)&quot;</span>,</span>
<span id="cb34-12"><a href="figure-1.html#cb34-12" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(C.~striata)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Cp.~striata)&quot;</span>,</span>
<span id="cb34-13"><a href="figure-1.html#cb34-13" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(C.~occipitalis)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Ch.~occipitalis)&quot;</span>,</span>
<span id="cb34-14"><a href="figure-1.html#cb34-14" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(C.~investigatoris)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Ch.~investigatoris)&quot;</span>,</span>
<span id="cb34-15"><a href="figure-1.html#cb34-15" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(C.~pleurospilus)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Ch.~pleurospilus)&quot;</span>,</span>
<span id="cb34-16"><a href="figure-1.html#cb34-16" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~tabacarius)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~tabacarius)&quot;</span>,</span>
<span id="cb34-17"><a href="figure-1.html#cb34-17" aria-hidden="true" tabindex="-1"></a>                     <span class="st">`</span><span class="at">italic(S.~tortugarum)</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;italic(Se.~tortugarum)&quot;</span>)</span></code></pre></div>
<p>Then, the labels of the data set are actually re-formatted.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="figure-1.html#cb35-1" aria-hidden="true" tabindex="-1"></a>edvr_serr_short <span class="ot">&lt;-</span> edvr_serr</span>
<span id="cb35-2"><a href="figure-1.html#cb35-2" aria-hidden="true" tabindex="-1"></a>edvr_serr_short<span class="sc">$</span>tip.label <span class="ot">&lt;-</span> edvr_serr<span class="sc">$</span>tip.label <span class="sc">%&gt;%</span> </span>
<span id="cb35-3"><a href="figure-1.html#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="at">pattern =</span> <span class="st">&quot;([A-Z])[a-z]*_([a-z]*)&quot;</span>, <span class="st">&quot;italic(</span><span class="sc">\\</span><span class="st">1.~</span><span class="sc">\\</span><span class="st">2)&quot;</span>)<span class="sc">%&gt;%</span> </span>
<span id="cb35-4"><a href="figure-1.html#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="at">pattern =</span> <span class="st">&quot;beta&quot;</span>, <span class="st">&quot;&#39;beta&#39;&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-5"><a href="figure-1.html#cb35-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ifelse</span>(. <span class="sc">%in%</span> <span class="fu">names</span>(label_two_chars), label_two_chars[.], .) <span class="sc">%&gt;%</span> </span>
<span id="cb35-6"><a href="figure-1.html#cb35-6" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">:::</span><span class="fu">parse_safe</span>()</span></code></pre></div>
<p>Now, the color scheme for the tree coloration is defined.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="figure-1.html#cb36-1" aria-hidden="true" tabindex="-1"></a>clr_tree <span class="ot">&lt;-</span> scico<span class="sc">::</span><span class="fu">scico</span>(<span class="dv">6</span>, <span class="at">palette =</span> <span class="st">&quot;berlin&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-2"><a href="figure-1.html#cb36-2" aria-hidden="true" tabindex="-1"></a>  prismatic<span class="sc">::</span><span class="fu">clr_desaturate</span>(<span class="at">shift =</span> .<span class="dv">4</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-3"><a href="figure-1.html#cb36-3" aria-hidden="true" tabindex="-1"></a>  prismatic<span class="sc">::</span><span class="fu">clr_darken</span>(<span class="at">shift =</span> .<span class="dv">2</span>)</span></code></pre></div>
<p>At this stage, the basic tree is drawn using a modified version of <code>BAMMtools::plot.bammdata()</code> (which allows for different colors for the tip labels).</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="figure-1.html#cb37-1" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">as.grob</span>(<span class="cf">function</span>(){</span>
<span id="cb37-2"><a href="figure-1.html#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb37-3"><a href="figure-1.html#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bammplot_k</span>(<span class="at">x =</span> edvr_serr_short,</span>
<span id="cb37-4"><a href="figure-1.html#cb37-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">labels =</span> T,</span>
<span id="cb37-5"><a href="figure-1.html#cb37-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">lwd =</span> .<span class="dv">8</span>, <span class="co">#legend = TRUE,</span></span>
<span id="cb37-6"><a href="figure-1.html#cb37-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">cex =</span> .<span class="dv">3</span>,</span>
<span id="cb37-7"><a href="figure-1.html#cb37-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">pal =</span> clr_tree,</span>
<span id="cb37-8"><a href="figure-1.html#cb37-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">labelcolor =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;black&quot;</span>, <span class="dv">9</span>),</span>
<span id="cb37-9"><a href="figure-1.html#cb37-9" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">rep</span>(<span class="st">&quot;darkgray&quot;</span>, <span class="dv">31</span>)))</span>
<span id="cb37-10"><a href="figure-1.html#cb37-10" aria-hidden="true" tabindex="-1"></a>  leg_shift_x <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb37-11"><a href="figure-1.html#cb37-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lines</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">25</span>) <span class="sc">+</span> leg_shift_x,</span>
<span id="cb37-12"><a href="figure-1.html#cb37-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">y =</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="fl">1.5</span>),</span>
<span id="cb37-13"><a href="figure-1.html#cb37-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> <span class="st">&quot;darkgray&quot;</span>)</span>
<span id="cb37-14"><a href="figure-1.html#cb37-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">text</span>(<span class="at">x =</span> <span class="fl">12.5</span> <span class="sc">+</span> leg_shift_x,</span>
<span id="cb37-15"><a href="figure-1.html#cb37-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> .<span class="dv">5</span>,</span>
<span id="cb37-16"><a href="figure-1.html#cb37-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">labels =</span> <span class="st">&quot;25 MYR&quot;</span>,</span>
<span id="cb37-17"><a href="figure-1.html#cb37-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">cex =</span> .<span class="dv">4</span>, </span>
<span id="cb37-18"><a href="figure-1.html#cb37-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;darkgray&quot;</span>)</span>
<span id="cb37-19"><a href="figure-1.html#cb37-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb37-20"><a href="figure-1.html#cb37-20" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p>Now that the basic tree is finished, we take care of the annotations starting with the color gradient that is going to highlight the Hypoplectrus family.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="figure-1.html#cb38-1" aria-hidden="true" tabindex="-1"></a>c1 <span class="ot">&lt;-</span> <span class="st">&quot;transparent&quot;</span></span>
<span id="cb38-2"><a href="figure-1.html#cb38-2" aria-hidden="true" tabindex="-1"></a>c2 <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, .<span class="dv">1</span>)</span>
<span id="cb38-3"><a href="figure-1.html#cb38-3" aria-hidden="true" tabindex="-1"></a>c3 <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, .<span class="dv">2</span>)</span>
<span id="cb38-4"><a href="figure-1.html#cb38-4" aria-hidden="true" tabindex="-1"></a>grad_mat <span class="ot">&lt;-</span> <span class="fu">c</span>(c1, c2, c3, c3) </span>
<span id="cb38-5"><a href="figure-1.html#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(grad_mat) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fu">length</span>(grad_mat))</span>
<span id="cb38-6"><a href="figure-1.html#cb38-6" aria-hidden="true" tabindex="-1"></a>grob_grad <span class="ot">&lt;-</span> <span class="fu">rasterGrob</span>(grad_mat,</span>
<span id="cb38-7"><a href="figure-1.html#cb38-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">width =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">&quot;npc&quot;</span>),</span>
<span id="cb38-8"><a href="figure-1.html#cb38-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">height =</span> <span class="fu">unit</span>(<span class="dv">1</span>, <span class="st">&quot;npc&quot;</span>), </span>
<span id="cb38-9"><a href="figure-1.html#cb38-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">interpolate =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="plot_F1_files/figure-html/unnamed-chunk-15-1.png" width="192" style="display: block; margin: auto;" /></p>
<p>Then, we create a black hamlet to add to the annotation.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="figure-1.html#cb39-1" aria-hidden="true" tabindex="-1"></a>blank_hamlet <span class="ot">&lt;-</span> hypoimg<span class="sc">::</span>hypo_outline <span class="sc">%&gt;%</span> </span>
<span id="cb39-2"><a href="figure-1.html#cb39-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>()<span class="sc">+</span></span>
<span id="cb39-3"><a href="figure-1.html#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_equal</span>()<span class="sc">+</span></span>
<span id="cb39-4"><a href="figure-1.html#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="fu">aes</span>(x, y), <span class="at">color =</span> <span class="fu">rgb</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, .<span class="dv">5</span>), <span class="at">fill =</span> <span class="fu">rgb</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, .<span class="dv">3</span>), <span class="at">size =</span> .<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb39-5"><a href="figure-1.html#cb39-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>()</span></code></pre></div>
<p><img src="plot_F1_files/figure-html/unnamed-chunk-17-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>At this point we have all pieces to assemble panel a of the figure.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="figure-1.html#cb40-1" aria-hidden="true" tabindex="-1"></a>p_tree <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb40-2"><a href="figure-1.html#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> <span class="fu">tibble</span>(<span class="at">v =</span> <span class="fu">c</span>(.<span class="dv">056</span>, <span class="fl">2.4</span>)),</span>
<span id="cb40-3"><a href="figure-1.html#cb40-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> .<span class="dv">5</span>, <span class="at">y =</span> .<span class="dv">5</span>, <span class="fu">aes</span>(<span class="at">color =</span> v), <span class="at">alpha =</span> <span class="dv">0</span>)<span class="sc">+</span></span>
<span id="cb40-4"><a href="figure-1.html#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradientn</span>(<span class="at">colours =</span> clr_tree, <span class="at">limits =</span> <span class="fu">c</span>(.<span class="dv">056</span>, <span class="fl">2.4</span>))<span class="sc">+</span></span>
<span id="cb40-5"><a href="figure-1.html#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_custom</span>(<span class="at">grob =</span> grob_grad,</span>
<span id="cb40-6"><a href="figure-1.html#cb40-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymin =</span> <span class="fl">0.01</span>, <span class="at">ymax =</span> .<span class="dv">2335</span>,</span>
<span id="cb40-7"><a href="figure-1.html#cb40-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xmin =</span> .<span class="dv">4</span>, <span class="at">xmax =</span> .<span class="dv">96</span>)<span class="sc">+</span></span>
<span id="cb40-8"><a href="figure-1.html#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_custom</span>(<span class="at">grob =</span> <span class="fu">ggplotGrob</span>(blank_hamlet),</span>
<span id="cb40-9"><a href="figure-1.html#cb40-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xmin =</span> <span class="fl">0.55</span>, <span class="at">xmax =</span> .<span class="dv">75</span>,</span>
<span id="cb40-10"><a href="figure-1.html#cb40-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymin =</span> <span class="fl">0.015</span>, <span class="at">ymax =</span> .<span class="dv">14</span>)<span class="sc">+</span></span>
<span id="cb40-11"><a href="figure-1.html#cb40-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_custom</span>(<span class="at">grob =</span> p1,</span>
<span id="cb40-12"><a href="figure-1.html#cb40-12" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xmin =</span> <span class="sc">-</span>.<span class="dv">16</span>,</span>
<span id="cb40-13"><a href="figure-1.html#cb40-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">xmax =</span> <span class="fl">1.05</span>,</span>
<span id="cb40-14"><a href="figure-1.html#cb40-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymin =</span> <span class="sc">-</span>.<span class="dv">22</span>,</span>
<span id="cb40-15"><a href="figure-1.html#cb40-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ymax =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb40-16"><a href="figure-1.html#cb40-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb40-17"><a href="figure-1.html#cb40-17" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb40-18"><a href="figure-1.html#cb40-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">expand =</span> <span class="dv">0</span>)<span class="sc">+</span></span>
<span id="cb40-19"><a href="figure-1.html#cb40-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">color =</span> <span class="fu">guide_colorbar</span>(<span class="at">title =</span> <span class="st">&quot;Speciation Rate&quot;</span>,</span>
<span id="cb40-20"><a href="figure-1.html#cb40-20" aria-hidden="true" tabindex="-1"></a>                                <span class="at">title.position =</span> <span class="st">&quot;top&quot;</span>,</span>
<span id="cb40-21"><a href="figure-1.html#cb40-21" aria-hidden="true" tabindex="-1"></a>                                <span class="at">direction =</span> <span class="st">&quot;horizontal&quot;</span>,</span>
<span id="cb40-22"><a href="figure-1.html#cb40-22" aria-hidden="true" tabindex="-1"></a>                                <span class="at">barheight =</span> <span class="fu">unit</span>(<span class="dv">3</span>, <span class="st">&quot;pt&quot;</span>),</span>
<span id="cb40-23"><a href="figure-1.html#cb40-23" aria-hidden="true" tabindex="-1"></a>                                <span class="at">barwidth =</span> <span class="fu">unit</span>(<span class="dv">57</span>, <span class="st">&quot;pt&quot;</span>),</span>
<span id="cb40-24"><a href="figure-1.html#cb40-24" aria-hidden="true" tabindex="-1"></a>                                <span class="at">ticks.colour =</span> <span class="st">&quot;white&quot;</span>)) <span class="sc">+</span></span>
<span id="cb40-25"><a href="figure-1.html#cb40-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>()<span class="sc">+</span></span>
<span id="cb40-26"><a href="figure-1.html#cb40-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(.<span class="dv">01</span>, .<span class="dv">08</span>),</span>
<span id="cb40-27"><a href="figure-1.html#cb40-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>))</span></code></pre></div>
<p><img src="plot_F1_files/figure-html/unnamed-chunk-19-1.png" width="432" style="display: block; margin: auto;" /></p>
</div>
<div id="genetic-differentiation-boxplots" class="section level3" number="17.2.3">
<h3><span class="header-section-number">17.2.3</span> Genetic differentiation boxplots</h3>
<p>For panel b, the data import starts with collecting the paths to all files containing the <span class="math inline">\(F_{ST}\)</span> data (<code>dir()</code>), then iterating the import function over all files (<code>map(summarize_fst)</code>) and finally combining the outputs into a single tibble (<code>bind_rows()</code>).</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="figure-1.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># =================================================================================================</span></span>
<span id="cb41-2"><a href="figure-1.html#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="co"># import Fst</span></span>
<span id="cb41-3"><a href="figure-1.html#cb41-3" aria-hidden="true" tabindex="-1"></a>fst_files <span class="ot">&lt;-</span> <span class="fu">dir</span>(fst_dir, <span class="at">pattern =</span> <span class="st">&#39;.50k.windowed.weir.fst.gz&#39;</span>)</span>
<span id="cb41-4"><a href="figure-1.html#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="figure-1.html#cb41-5" aria-hidden="true" tabindex="-1"></a>fst_data <span class="ot">&lt;-</span> <span class="fu">str_c</span>(fst_dir, fst_files) <span class="sc">%&gt;%</span> </span>
<span id="cb41-6"><a href="figure-1.html#cb41-6" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map</span>(summarize_fst) <span class="sc">%&gt;%</span></span>
<span id="cb41-7"><a href="figure-1.html#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>()</span></code></pre></div>
<p>We use the genome wide average <span class="math inline">\(F_{ST}\)</span> to rank the individual pair wise comparisons.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="figure-1.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># determine fst ranking</span></span>
<span id="cb42-2"><a href="figure-1.html#cb42-2" aria-hidden="true" tabindex="-1"></a>fst_order <span class="ot">&lt;-</span> fst_data <span class="sc">%&gt;%</span></span>
<span id="cb42-3"><a href="figure-1.html#cb42-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(run, <span class="st">`</span><span class="at">mean_weighted-fst</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-4"><a href="figure-1.html#cb42-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">run =</span> <span class="fu">fct_reorder</span>(run, <span class="st">`</span><span class="at">mean_weighted-fst</span><span class="st">`</span>))</span></code></pre></div>
<p>Then, we transform the <span class="math inline">\(F_{ST}\)</span> data and do quite a bit of data wrangling to prepare the placement of the boxplots.
(This is overly complicated by now, as this is mostly <em>legacy</em> code that was create for a previous version of the plot that combined re-scaled <span class="math inline">\(F_{ST}\)</span> and <span class="math inline">\(d_{XY}\)</span> boxplots.)</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="figure-1.html#cb43-1" aria-hidden="true" tabindex="-1"></a>fst_data_gather <span class="ot">&lt;-</span> fst_data <span class="sc">%&gt;%</span> </span>
<span id="cb43-2"><a href="figure-1.html#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(<span class="at">key =</span> <span class="st">&#39;stat&#39;</span>, <span class="at">value =</span> <span class="st">&#39;val&#39;</span>, <span class="sc">-</span>run) <span class="sc">%&gt;%</span></span>
<span id="cb43-3"><a href="figure-1.html#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># sumstat contains the values needed to plot the boxplots (quartiles, etc)</span></span>
<span id="cb43-4"><a href="figure-1.html#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(stat, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&#39;sumstat&#39;</span>, <span class="st">&#39;popstat&#39;</span>), <span class="at">sep =</span> <span class="st">&#39;_&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-5"><a href="figure-1.html#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># duplicate dxy values scaled to fst range</span></span>
<span id="cb43-6"><a href="figure-1.html#cb43-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">val_scaled =</span> <span class="fu">ifelse</span>(popstat <span class="sc">==</span> <span class="st">&#39;dxy&#39;</span>, val <span class="sc">*</span> scaler , val)) <span class="sc">%&gt;%</span></span>
<span id="cb43-7"><a href="figure-1.html#cb43-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(temp, val, val_scaled) <span class="sc">%&gt;%</span></span>
<span id="cb43-8"><a href="figure-1.html#cb43-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># separate th eoriginal values from the scales ons (scaled = secondary axis)</span></span>
<span id="cb43-9"><a href="figure-1.html#cb43-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">spread</span>(., <span class="at">key =</span> <span class="st">&#39;sumstat&#39;</span>, <span class="at">value =</span> <span class="st">&#39;temp&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-10"><a href="figure-1.html#cb43-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(mean, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&#39;mean&#39;</span>,<span class="st">&#39;mean_scaled&#39;</span>),<span class="at">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="at">convert =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-11"><a href="figure-1.html#cb43-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(median, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&#39;median&#39;</span>,<span class="st">&#39;median_scaled&#39;</span>), <span class="at">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="at">convert =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-12"><a href="figure-1.html#cb43-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(sd, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&#39;sd&#39;</span>,<span class="st">&#39;sd_scaled&#39;</span>),<span class="at">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="at">convert =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-13"><a href="figure-1.html#cb43-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(lower, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&#39;lower&#39;</span>,<span class="st">&#39;lower_scaled&#39;</span>), <span class="at">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="at">convert =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-14"><a href="figure-1.html#cb43-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(upper, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&#39;upper&#39;</span>,<span class="st">&#39;upper_scaled&#39;</span>), <span class="at">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="at">convert =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-15"><a href="figure-1.html#cb43-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(lowpoint, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&#39;lowpoint&#39;</span>,<span class="st">&#39;lowpoint_scaled&#39;</span>), <span class="at">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="at">convert =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-16"><a href="figure-1.html#cb43-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(highpoint, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&#39;highpoint&#39;</span>,<span class="st">&#39;highpoint_scaled&#39;</span>), <span class="at">sep =</span> <span class="st">&#39;_&#39;</span>, <span class="at">convert =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-17"><a href="figure-1.html#cb43-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># include &quot;dodge&quot;-positions for side-by-side plotting (secondary axis)</span></span>
<span id="cb43-18"><a href="figure-1.html#cb43-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">loc =</span> <span class="fu">str_sub</span>(run,<span class="dv">4</span>,<span class="dv">6</span>),</span>
<span id="cb43-19"><a href="figure-1.html#cb43-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">run =</span> <span class="fu">factor</span>(run, <span class="at">levels =</span> <span class="fu">levels</span>(fst_order<span class="sc">$</span>run)),</span>
<span id="cb43-20"><a href="figure-1.html#cb43-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="fu">as.numeric</span>(run) ,</span>
<span id="cb43-21"><a href="figure-1.html#cb43-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">x_dodge =</span> <span class="fu">ifelse</span>(popstat <span class="sc">==</span> <span class="st">&#39;dxy&#39;</span>, x <span class="sc">+</span> .<span class="dv">25</span>, x <span class="sc">-</span> .<span class="dv">25</span>),</span>
<span id="cb43-22"><a href="figure-1.html#cb43-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">x_start_dodge =</span> x_dodge <span class="sc">-</span> wdh<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb43-23"><a href="figure-1.html#cb43-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">x_end_dodge =</span> x_dodge <span class="sc">+</span> wdh<span class="sc">/</span><span class="dv">2</span>,</span>
<span id="cb43-24"><a href="figure-1.html#cb43-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">popstat_loc =</span> <span class="fu">str_c</span>(popstat,<span class="st">&#39;[&#39;</span>,loc,<span class="st">&#39;]&#39;</span>))</span></code></pre></div>
<p>Then we re-order the species pairs according to their average wheighted <span class="math inline">\(F_{ST}\)</span>.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="figure-1.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sort run by average genome wide Fst</span></span>
<span id="cb44-2"><a href="figure-1.html#cb44-2" aria-hidden="true" tabindex="-1"></a>run_ord <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">run =</span> <span class="fu">levels</span>(fst_data_gather<span class="sc">$</span>run),</span>
<span id="cb44-3"><a href="figure-1.html#cb44-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">run_ord =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">levels</span>(fst_data_gather<span class="sc">$</span>run)))</span></code></pre></div>
<p>As the last information that is needed for the <span class="math inline">\(F_{ST}\)</span> boxplots, we load the results from the permutation tests (this is the part that depends on <code>R/figs/plot_SF15.R</code>).</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="figure-1.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load fst permutation results</span></span>
<span id="cb45-2"><a href="figure-1.html#cb45-2" aria-hidden="true" tabindex="-1"></a>fst_sig_attach <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(fst_permutation_file) <span class="sc">%&gt;%</span> </span>
<span id="cb45-3"><a href="figure-1.html#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">loc =</span> <span class="fu">str_sub</span>(run, <span class="sc">-</span><span class="dv">3</span>, <span class="sc">-</span><span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-4"><a href="figure-1.html#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(loc) <span class="sc">%&gt;%</span> </span>
<span id="cb45-5"><a href="figure-1.html#cb45-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">loc_n =</span> <span class="dv">28</span>,<span class="co">#length(loc),</span></span>
<span id="cb45-6"><a href="figure-1.html#cb45-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">fdr_correction_factor =</span>  <span class="fu">sum</span>(<span class="dv">1</span> <span class="sc">/</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(loc)),</span>
<span id="cb45-7"><a href="figure-1.html#cb45-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">fdr_alpha =</span> .<span class="dv">05</span> <span class="sc">/</span> fdr_correction_factor,</span>
<span id="cb45-8"><a href="figure-1.html#cb45-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">is_sig =</span> p_perm <span class="sc">&gt;</span> fdr_alpha) <span class="sc">%&gt;%</span> </span>
<span id="cb45-9"><a href="figure-1.html#cb45-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code></pre></div>
<p>At this point, the data is ready for the boxplots.</p>
<p>But first, we are going to prepare the networks of pairwise comparisons that are used as annotation of that panel.</p>
<p>For this we create a tibble of the species pairs with their respective rank.
Then, we prepare a config table with one row per location, storing the parameters needed for the layout function for the networks.
We need to define the location, the number of species at the location, the short three letter ID of those species and a weight parameter that is shifting the comparison label on the link within the networks.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="figure-1.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create network annotation</span></span>
<span id="cb46-2"><a href="figure-1.html#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="co"># underlying structure for the network plots</span></span>
<span id="cb46-3"><a href="figure-1.html#cb46-3" aria-hidden="true" tabindex="-1"></a>networx <span class="ot">&lt;-</span> <span class="fu">tibble</span>( <span class="at">loc =</span> <span class="fu">c</span>(<span class="st">&#39;bel&#39;</span>,<span class="st">&#39;hon&#39;</span>, <span class="st">&#39;pan&#39;</span>),</span>
<span id="cb46-4"><a href="figure-1.html#cb46-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">n =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">6</span>, <span class="dv">3</span>),</span>
<span id="cb46-5"><a href="figure-1.html#cb46-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">label =</span> <span class="fu">list</span>(<span class="fu">str_c</span>(<span class="fu">c</span>(<span class="st">&#39;ind&#39;</span>,<span class="st">&#39;may&#39;</span>,<span class="st">&#39;nig&#39;</span>,<span class="st">&#39;pue&#39;</span>,<span class="st">&#39;uni&#39;</span>),<span class="st">&#39;bel&#39;</span>),</span>
<span id="cb46-6"><a href="figure-1.html#cb46-6" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">str_c</span>(<span class="fu">c</span>(<span class="st">&#39;abe&#39;</span>,<span class="st">&#39;gum&#39;</span>,<span class="st">&#39;nig&#39;</span>,<span class="st">&#39;pue&#39;</span>,<span class="st">&#39;ran&#39;</span>,<span class="st">&#39;uni&#39;</span>),<span class="st">&#39;hon&#39;</span>),</span>
<span id="cb46-7"><a href="figure-1.html#cb46-7" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">str_c</span>(<span class="fu">c</span>(<span class="st">&#39;nig&#39;</span>,<span class="st">&#39;pue&#39;</span>,<span class="st">&#39;uni&#39;</span>),<span class="st">&#39;pan&#39;</span>)),</span>
<span id="cb46-8"><a href="figure-1.html#cb46-8" aria-hidden="true" tabindex="-1"></a>                   <span class="at">weight =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="fl">1.45</span>,<span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb46-9"><a href="figure-1.html#cb46-9" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">pmap_dfr</span>(network_layout) <span class="sc">%&gt;%</span></span>
<span id="cb46-10"><a href="figure-1.html#cb46-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">edges =</span> <span class="fu">map</span>(edges, <span class="cf">function</span>(x){x <span class="sc">%&gt;%</span> <span class="fu">left_join</span>(fst_data_gather <span class="sc">%&gt;%</span> <span class="fu">filter</span>(popstat <span class="sc">==</span> <span class="st">&quot;weighted-fst&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(run, median, mean)) }))</span></code></pre></div>
<p>Next, we create one network plot per location…</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="figure-1.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the individual networks by location</span></span>
<span id="cb47-2"><a href="figure-1.html#cb47-2" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> networx <span class="sc">%&gt;%</span></span>
<span id="cb47-3"><a href="figure-1.html#cb47-3" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">pmap</span>(plot_network, <span class="at">node_lab_shift =</span> .<span class="dv">2</span>)</span></code></pre></div>
<p>…and assemble them into a single grid object.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="figure-1.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># combine the networks into a single grob</span></span>
<span id="cb48-2"><a href="figure-1.html#cb48-2" aria-hidden="true" tabindex="-1"></a>p_net <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb48-3"><a href="figure-1.html#cb48-3" aria-hidden="true" tabindex="-1"></a>  plot_list[[<span class="dv">1</span>]] <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>),</span>
<span id="cb48-4"><a href="figure-1.html#cb48-4" aria-hidden="true" tabindex="-1"></a>  plot_list[[<span class="dv">2</span>]] <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>),</span>
<span id="cb48-5"><a href="figure-1.html#cb48-5" aria-hidden="true" tabindex="-1"></a>  plot_list[[<span class="dv">3</span>]] <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>),</span>
<span id="cb48-6"><a href="figure-1.html#cb48-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">ncol =</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span></span>
<span id="cb48-7"><a href="figure-1.html#cb48-7" aria-hidden="true" tabindex="-1"></a>  cowplot<span class="sc">::</span><span class="fu">as_grob</span>()</span></code></pre></div>
<p><img src="plot_F1_files/figure-html/unnamed-chunk-28-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>At this point we can create panel b.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="figure-1.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># assemble panel b</span></span>
<span id="cb49-2"><a href="figure-1.html#cb49-2" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> fst_data_gather <span class="sc">%&gt;%</span></span>
<span id="cb49-3"><a href="figure-1.html#cb49-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(popstat <span class="sc">==</span> <span class="st">&quot;weighted-fst&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-4"><a href="figure-1.html#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(fst_sig_attach) <span class="sc">%&gt;%</span> </span>
<span id="cb49-5"><a href="figure-1.html#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">loc =</span> <span class="fu">str_sub</span>(run, <span class="sc">-</span><span class="dv">3</span>, <span class="sc">-</span><span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-6"><a href="figure-1.html#cb49-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">color =</span> loc)) <span class="sc">+</span> </span>
<span id="cb49-7"><a href="figure-1.html#cb49-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotation_custom</span>(p_net, <span class="at">ymin =</span> .<span class="dv">15</span>, <span class="at">xmax =</span> <span class="dv">25</span>)<span class="sc">+</span></span>
<span id="cb49-8"><a href="figure-1.html#cb49-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">xend =</span> x,</span>
<span id="cb49-9"><a href="figure-1.html#cb49-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> lowpoint, <span class="at">yend =</span> highpoint),</span>
<span id="cb49-10"><a href="figure-1.html#cb49-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">lwd =</span> plot_lwd)<span class="sc">+</span></span>
<span id="cb49-11"><a href="figure-1.html#cb49-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_rect</span>(<span class="fu">aes</span>(<span class="at">xmin =</span> x <span class="sc">-</span> wdh, <span class="at">xmax =</span> x <span class="sc">+</span> wdh,</span>
<span id="cb49-12"><a href="figure-1.html#cb49-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">ymin =</span> lower, <span class="at">ymax =</span> upper),</span>
<span id="cb49-13"><a href="figure-1.html#cb49-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">fill =</span> <span class="st">&#39;white&#39;</span>,</span>
<span id="cb49-14"><a href="figure-1.html#cb49-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">size =</span> plot_lwd)<span class="sc">+</span></span>
<span id="cb49-15"><a href="figure-1.html#cb49-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> x <span class="sc">-</span> wdh,</span>
<span id="cb49-16"><a href="figure-1.html#cb49-16" aria-hidden="true" tabindex="-1"></a>                   <span class="at">xend =</span> x <span class="sc">+</span> wdh,</span>
<span id="cb49-17"><a href="figure-1.html#cb49-17" aria-hidden="true" tabindex="-1"></a>                   <span class="at">y =</span> median,</span>
<span id="cb49-18"><a href="figure-1.html#cb49-18" aria-hidden="true" tabindex="-1"></a>                   <span class="at">yend =</span> median),</span>
<span id="cb49-19"><a href="figure-1.html#cb49-19" aria-hidden="true" tabindex="-1"></a>               <span class="at">lwd =</span> plot_lwd)<span class="sc">+</span></span>
<span id="cb49-20"><a href="figure-1.html#cb49-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> mean, <span class="at">shape =</span> is_sig, <span class="at">fill =</span> <span class="fu">after_scale</span>(color)),</span>
<span id="cb49-21"><a href="figure-1.html#cb49-21" aria-hidden="true" tabindex="-1"></a>             <span class="at">size =</span> .<span class="dv">8</span>)<span class="sc">+</span></span>
<span id="cb49-22"><a href="figure-1.html#cb49-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">name =</span> <span class="st">&quot;Pair of sympatric species&quot;</span>,</span>
<span id="cb49-23"><a href="figure-1.html#cb49-23" aria-hidden="true" tabindex="-1"></a>                     <span class="at">breaks =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">28</span>) <span class="sc">+</span></span>
<span id="cb49-24"><a href="figure-1.html#cb49-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="fu">expression</span>(<span class="fu">italic</span>(F[ST])))<span class="sc">+</span></span>
<span id="cb49-25"><a href="figure-1.html#cb49-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="fu">make_faint_clr</span>(<span class="st">&#39;bel&#39;</span>),</span>
<span id="cb49-26"><a href="figure-1.html#cb49-26" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">make_faint_clr</span>(<span class="st">&#39;hon&#39;</span>),</span>
<span id="cb49-27"><a href="figure-1.html#cb49-27" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">make_faint_clr</span>(<span class="st">&#39;pan&#39;</span>))[<span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">6</span>)])<span class="sc">+</span></span>
<span id="cb49-28"><a href="figure-1.html#cb49-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">TRUE</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">`</span><span class="at">FALSE</span><span class="st">`</span> <span class="ot">=</span> <span class="dv">21</span>)) <span class="sc">+</span></span>
<span id="cb49-29"><a href="figure-1.html#cb49-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">29</span>),</span>
<span id="cb49-30"><a href="figure-1.html#cb49-30" aria-hidden="true" tabindex="-1"></a>                  <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb49-31"><a href="figure-1.html#cb49-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()<span class="sc">+</span></span>
<span id="cb49-32"><a href="figure-1.html#cb49-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> plot_text_size),</span>
<span id="cb49-33"><a href="figure-1.html#cb49-33" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>,</span>
<span id="cb49-34"><a href="figure-1.html#cb49-34" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.placement =</span> <span class="st">&#39;outside&#39;</span>,</span>
<span id="cb49-35"><a href="figure-1.html#cb49-35" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb49-36"><a href="figure-1.html#cb49-36" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb49-37"><a href="figure-1.html#cb49-37" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb49-38"><a href="figure-1.html#cb49-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y.right =</span> <span class="fu">element_text</span>(<span class="at">color =</span> clr_sec),</span>
<span id="cb49-39"><a href="figure-1.html#cb49-39" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y.right =</span> <span class="fu">element_text</span>(<span class="at">color =</span> clr_sec))</span></code></pre></div>
<pre><code>#&gt; Warning in if (!expand) {: the condition has length &gt; 1 and only the first
#&gt; element will be used

#&gt; Warning in if (!expand) {: the condition has length &gt; 1 and only the first
#&gt; element will be used</code></pre>
<p><img src="plot_F1_files/figure-html/unnamed-chunk-30-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="principal-component-analyses" class="section level3" number="17.2.4">
<h3><span class="header-section-number">17.2.4</span> Principal Component Analyses</h3>
<p>Most of the plotting functions for the PCAs are provided in {GenomicOriginsScripts} package, so that the largest part of the work for the PCAs actually concerns the annotations.</p>
<p>Fist, we set an alternative color scale and define the size of the hamlet images.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="figure-1.html#cb51-1" aria-hidden="true" tabindex="-1"></a>clr_alt <span class="ot">&lt;-</span> clr</span>
<span id="cb51-2"><a href="figure-1.html#cb51-2" aria-hidden="true" tabindex="-1"></a>clr_alt[[<span class="st">&quot;uni&quot;</span>]] <span class="ot">&lt;-</span> <span class="st">&quot;lightgray&quot;</span></span>
<span id="cb51-3"><a href="figure-1.html#cb51-3" aria-hidden="true" tabindex="-1"></a>pca_fish_scale <span class="ot">&lt;-</span> <span class="fl">1.15</span></span></code></pre></div>
<p>The we create a configuration tibble that holds the desired positions of the hamlet annotations on the PCAs.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="figure-1.html#cb52-1" aria-hidden="true" tabindex="-1"></a>pca_fish_pos <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">pop =</span> GenomicOriginsScripts<span class="sc">::</span>pop_levels,</span>
<span id="cb52-2"><a href="figure-1.html#cb52-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">short =</span> <span class="fu">str_sub</span>(pop, <span class="dv">1</span>, <span class="dv">3</span>),</span>
<span id="cb52-3"><a href="figure-1.html#cb52-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">loc =</span> <span class="fu">str_sub</span>(pop, <span class="dv">4</span>, <span class="dv">6</span>),</span>
<span id="cb52-4"><a href="figure-1.html#cb52-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">width =</span> <span class="fu">c</span>(<span class="at">bel =</span> .<span class="dv">08</span>, <span class="at">hon =</span>.<span class="dv">08</span>, <span class="at">pan =</span> .<span class="dv">09</span>)[loc] <span class="sc">*</span> pca_fish_scale,</span>
<span id="cb52-5"><a href="figure-1.html#cb52-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">height =</span> <span class="fu">c</span>(<span class="at">bel =</span> .<span class="dv">08</span>, <span class="at">hon =</span>.<span class="dv">08</span>, <span class="at">pan =</span> .<span class="dv">09</span>)[loc] <span class="sc">*</span> pca_fish_scale) <span class="sc">%&gt;%</span> </span>
<span id="cb52-6"><a href="figure-1.html#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(loc) <span class="sc">%&gt;%</span></span>
<span id="cb52-7"><a href="figure-1.html#cb52-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="sc">-</span>.<span class="dv">18</span>, <span class="sc">-</span>.<span class="dv">01</span>, .<span class="dv">03</span>, <span class="sc">-</span>.<span class="dv">03</span>, .<span class="dv">075</span>,</span>
<span id="cb52-8"><a href="figure-1.html#cb52-8" aria-hidden="true" tabindex="-1"></a>               <span class="sc">-</span>.<span class="dv">04</span>, <span class="sc">-</span>.<span class="dv">2</span>, <span class="sc">-</span>.<span class="dv">02</span>, <span class="sc">-</span>.<span class="dv">01</span>, <span class="sc">-</span>.<span class="dv">02</span>, <span class="sc">-</span>.<span class="dv">01</span>,</span>
<span id="cb52-9"><a href="figure-1.html#cb52-9" aria-hidden="true" tabindex="-1"></a>               <span class="sc">-</span>.<span class="dv">15</span>, <span class="dv">0</span>, .<span class="dv">05</span>),</span>
<span id="cb52-10"><a href="figure-1.html#cb52-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="fu">c</span>(.<span class="dv">02</span>, .<span class="dv">27</span>, <span class="sc">-</span>.<span class="dv">13</span>, <span class="sc">-</span>.<span class="dv">03</span>, .<span class="dv">05</span>,</span>
<span id="cb52-11"><a href="figure-1.html#cb52-11" aria-hidden="true" tabindex="-1"></a>               <span class="sc">-</span>.<span class="dv">1</span>, .<span class="dv">05</span>, <span class="dv">0</span>, .<span class="dv">075</span>, <span class="sc">-</span>.<span class="dv">23</span>, .<span class="dv">2</span>,</span>
<span id="cb52-12"><a href="figure-1.html#cb52-12" aria-hidden="true" tabindex="-1"></a>               .<span class="dv">06</span>, <span class="sc">-</span>.<span class="dv">2</span>, .<span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb52-13"><a href="figure-1.html#cb52-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>pop) <span class="sc">%&gt;%</span></span>
<span id="cb52-14"><a href="figure-1.html#cb52-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(loc) <span class="sc">%&gt;%</span></span>
<span id="cb52-15"><a href="figure-1.html#cb52-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">nest</span>()</span></code></pre></div>
<p>Then we create the individual PCA plots using the function <code>GenomicOriginsScripts::pca_plot()</code>.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="figure-1.html#cb53-1" aria-hidden="true" tabindex="-1"></a>pcas <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;bel&quot;</span>, <span class="st">&quot;hon&quot;</span>, <span class="st">&quot;pan&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">map</span>(pca_plot)</span></code></pre></div>
<p><img src="plot_F1_files/figure-html/unnamed-chunk-34-1.png" width="288" style="display: block; margin: auto;" /></p>
</div>
<div id="legend-and-final-figure" class="section level3" number="17.2.5">
<h3><span class="header-section-number">17.2.5</span> Legend and Final Figure</h3>
<p>To create the figure legend we define a set of hamlet species (all sequenced species except the outgroups) with their relative horizontal spacing.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="figure-1.html#cb54-1" aria-hidden="true" tabindex="-1"></a>fish_tib <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">short =</span> <span class="fu">names</span>(clr)[<span class="sc">!</span><span class="fu">names</span>(clr) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;flo&quot;</span>, <span class="st">&quot;tab&quot;</span>, <span class="st">&quot;tor&quot;</span>)],</span>
<span id="cb54-2"><a href="figure-1.html#cb54-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">c</span>(<span class="fl">0.5</span>,  <span class="fl">3.5</span>,  <span class="dv">7</span>,  <span class="fl">9.7</span>, <span class="fl">12.25</span>, <span class="fl">15.25</span>, <span class="dv">18</span>, <span class="fl">21.5</span>))</span></code></pre></div>
<p>Then, we set the color-patch size and create a custom ggplot that is going to function as legend.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="figure-1.html#cb55-1" aria-hidden="true" tabindex="-1"></a>key_sz <span class="ot">&lt;-</span> .<span class="dv">75</span></span>
<span id="cb55-2"><a href="figure-1.html#cb55-2" aria-hidden="true" tabindex="-1"></a>p_leg <span class="ot">&lt;-</span> fish_tib <span class="sc">%&gt;%</span> </span>
<span id="cb55-3"><a href="figure-1.html#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb55-4"><a href="figure-1.html#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_equal</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span>.<span class="dv">05</span>, <span class="dv">24</span>), <span class="at">expand =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb55-5"><a href="figure-1.html#cb55-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> <span class="dv">0</span>,</span>
<span id="cb55-6"><a href="figure-1.html#cb55-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">fill =</span> short, </span>
<span id="cb55-7"><a href="figure-1.html#cb55-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">color =</span> <span class="fu">after_scale</span>(prismatic<span class="sc">::</span><span class="fu">clr_darken</span>(fill, .<span class="dv">25</span>))),</span>
<span id="cb55-8"><a href="figure-1.html#cb55-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">width =</span> key_sz, <span class="at">height =</span> key_sz, <span class="at">size =</span> .<span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb55-9"><a href="figure-1.html#cb55-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span> x <span class="sc">+</span> .<span class="dv">6</span>, <span class="at">y =</span> <span class="dv">0</span>,</span>
<span id="cb55-10"><a href="figure-1.html#cb55-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">label =</span> <span class="fu">str_c</span>(<span class="st">&quot;H. &quot;</span>, sp_names[short])), </span>
<span id="cb55-11"><a href="figure-1.html#cb55-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">fontface =</span> <span class="st">&quot;italic&quot;</span>, <span class="at">size =</span> plot_text_size <span class="sc">/</span> ggplot2<span class="sc">:::</span>.pt) <span class="sc">+</span></span>
<span id="cb55-12"><a href="figure-1.html#cb55-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pmap</span>(fish_tib, plot_fish_lwd, <span class="at">width =</span> <span class="dv">1</span>, <span class="at">height =</span> <span class="dv">1</span>, <span class="at">y =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb55-13"><a href="figure-1.html#cb55-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> clr, <span class="at">guide =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb55-14"><a href="figure-1.html#cb55-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>()</span></code></pre></div>
<p>After this, first the fice panels are combined…</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="figure-1.html#cb56-1" aria-hidden="true" tabindex="-1"></a>p_combined <span class="ot">&lt;-</span> ((<span class="fu">wrap_elements</span>(<span class="at">plot =</span> p_tree <span class="sc">+</span></span>
<span id="cb56-2"><a href="figure-1.html#cb56-2" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb56-3"><a href="figure-1.html#cb56-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> plot_text_size)),</span>
<span id="cb56-4"><a href="figure-1.html#cb56-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">clip =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb56-5"><a href="figure-1.html#cb56-5" aria-hidden="true" tabindex="-1"></a>                 p2 ) <span class="sc">/</span></span>
<span id="cb56-6"><a href="figure-1.html#cb56-6" aria-hidden="true" tabindex="-1"></a>                 (pcas <span class="sc">%&gt;%</span> <span class="fu">wrap_plots</span>()) <span class="sc">+</span></span>
<span id="cb56-7"><a href="figure-1.html#cb56-7" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">plot_layout</span>(<span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,.<span class="dv">75</span>)) <span class="sc">+</span></span>
<span id="cb56-8"><a href="figure-1.html#cb56-8" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">&#39;a&#39;</span>) <span class="sc">&amp;</span></span>
<span id="cb56-9"><a href="figure-1.html#cb56-9" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> plot_text_size),</span>
<span id="cb56-10"><a href="figure-1.html#cb56-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;transparent&quot;</span>,</span>
<span id="cb56-11"><a href="figure-1.html#cb56-11" aria-hidden="true" tabindex="-1"></a>                                                      <span class="at">color =</span> <span class="st">&quot;transparent&quot;</span>)))</span></code></pre></div>
<p>.. and then the legend is attached.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="figure-1.html#cb57-1" aria-hidden="true" tabindex="-1"></a>p_done <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p_combined, p_leg,</span>
<span id="cb57-2"><a href="figure-1.html#cb57-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>,.<span class="dv">06</span>))</span></code></pre></div>
<p><img src="plot_F1_files/figure-html/unnamed-chunk-39-1.png" width="648" style="display: block; margin: auto;" /></p>
<p>Finally, we can export Figure 1.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="figure-1.html#cb58-1" aria-hidden="true" tabindex="-1"></a>scl <span class="ot">&lt;-</span> .<span class="dv">75</span></span>
<span id="cb58-2"><a href="figure-1.html#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="fu">hypo_save</span>(p_done, <span class="at">filename =</span> <span class="st">&#39;figures/F1.pdf&#39;</span>,</span>
<span id="cb58-3"><a href="figure-1.html#cb58-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">width =</span> <span class="dv">9</span> <span class="sc">*</span> scl,</span>
<span id="cb58-4"><a href="figure-1.html#cb58-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">height =</span> <span class="dv">6</span> <span class="sc">*</span> scl,</span>
<span id="cb58-5"><a href="figure-1.html#cb58-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">device =</span> cairo_pdf,</span>
<span id="cb58-6"><a href="figure-1.html#cb58-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">bg =</span> <span class="st">&quot;transparent&quot;</span>,</span>
<span id="cb58-7"><a href="figure-1.html#cb58-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">comment =</span> plot_comment)</span></code></pre></div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="git-15-data-visualization.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="figure-2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
