<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>15 (git 14) Analysis XII (Outlier Region Phylogenies) | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="15 (git 14) Analysis XII (Outlier Region Phylogenies) | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="15 (git 14) Analysis XII (Outlier Region Phylogenies) | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench and Martin Helmkampf" />


<meta name="date" content="2021-08-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="git-13-analysis-xi-whole-genome-phylogenies.html"/>
<link rel="next" href="git-15-data-visualization.html"/>
<script src="libs/header-attrs-2.7.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#r-setup"><i class="fa fa-check"></i><b>1.4</b> R setup</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a>
<ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a>
<ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i> &amp; pi)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (topology weighting)</a>
<ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (Principal Component Analysis)</a>
<ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#details-of-analysis_pca.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_pca.nf</code></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#setup-2"><i class="fa fa-check"></i><b>8.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-4"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-5"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-6"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html"><i class="fa fa-check"></i><b>12</b> (git 11) Analysis IX (Allele Age)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#summary-10"><i class="fa fa-check"></i><b>12.1</b> Summary</a></li>
<li class="chapter" data-level="12.2" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#details-of-analysis_allele_age.nf"><i class="fa fa-check"></i><b>12.2</b> Details of <code>analysis_allele_age.nf</code></a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#setup-3"><i class="fa fa-check"></i><b>12.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html"><i class="fa fa-check"></i><b>13</b> (git 12) Analysis X (<i>F<sub>ST</sub></i> Permutation Test)</a>
<ul>
<li class="chapter" data-level="13.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#summary-11"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#details-of-analysis_fst_sign.nf"><i class="fa fa-check"></i><b>13.2</b> Details of <code>analysis_fst_sign.nf</code></a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#setup-4"><i class="fa fa-check"></i><b>13.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html"><i class="fa fa-check"></i><b>14</b> (git 13) Analysis XI (Whole Genome Phylogenies)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#summary-12"><i class="fa fa-check"></i><b>14.1</b> Summary</a></li>
<li class="chapter" data-level="14.2" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#details-of-analysis_phylo_whg.nf"><i class="fa fa-check"></i><b>14.2</b> Details of <code>analysis_phylo_whg.nf</code></a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#setup-5"><i class="fa fa-check"></i><b>14.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html"><i class="fa fa-check"></i><b>15</b> (git 14) Analysis XII (Outlier Region Phylogenies)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#summary-13"><i class="fa fa-check"></i><b>15.1</b> Summary</a></li>
<li class="chapter" data-level="15.2" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#details-of-analysis_phylo_regions.nf"><i class="fa fa-check"></i><b>15.2</b> Details of <code>analysis_phylo_regions.nf</code></a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#setup-6"><i class="fa fa-check"></i><b>15.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="git-15-data-visualization.html"><a href="git-15-data-visualization.html"><i class="fa fa-check"></i><b>16</b> (git 15) Data Visualization</a></li>
<li class="chapter" data-level="17" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>17</b> Figure 1</a>
<ul>
<li class="chapter" data-level="17.1" data-path="figure-1.html"><a href="figure-1.html#summary-14"><i class="fa fa-check"></i><b>17.1</b> Summary</a></li>
<li class="chapter" data-level="17.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>17.2</b> Details of <code>plot_F1.R</code></a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>17.2.1</b> Config</a></li>
<li class="chapter" data-level="17.2.2" data-path="figure-1.html"><a href="figure-1.html#fish-tree-of-life-subtree"><i class="fa fa-check"></i><b>17.2.2</b> Fish Tree Of Life Subtree</a></li>
<li class="chapter" data-level="17.2.3" data-path="figure-1.html"><a href="figure-1.html#genetic-differentiation-boxplots"><i class="fa fa-check"></i><b>17.2.3</b> Genetic differentiation boxplots</a></li>
<li class="chapter" data-level="17.2.4" data-path="figure-1.html"><a href="figure-1.html#principal-component-analyses"><i class="fa fa-check"></i><b>17.2.4</b> Principal Component Analyses</a></li>
<li class="chapter" data-level="17.2.5" data-path="figure-1.html"><a href="figure-1.html#legend-and-final-figure"><i class="fa fa-check"></i><b>17.2.5</b> Legend and Final Figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>18</b> Figure 2</a>
<ul>
<li class="chapter" data-level="18.1" data-path="figure-2.html"><a href="figure-2.html#summary-15"><i class="fa fa-check"></i><b>18.1</b> Summary</a></li>
<li class="chapter" data-level="18.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>18.2</b> Details of <code>plot_F2.R</code></a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>18.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>19</b> Figure 3</a>
<ul>
<li class="chapter" data-level="19.1" data-path="figure-3.html"><a href="figure-3.html#summary-16"><i class="fa fa-check"></i><b>19.1</b> Summary</a></li>
<li class="chapter" data-level="19.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>19.2</b> Details of <code>plot_F3.R</code></a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>19.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>20</b> Figure 4</a>
<ul>
<li class="chapter" data-level="20.1" data-path="figure-4.html"><a href="figure-4.html#summary-17"><i class="fa fa-check"></i><b>20.1</b> Summary</a></li>
<li class="chapter" data-level="20.2" data-path="figure-4.html"><a href="figure-4.html#details-of-plot_f4.r"><i class="fa fa-check"></i><b>20.2</b> Details of <code>plot_F4.R</code></a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="figure-4.html"><a href="figure-4.html#config-3"><i class="fa fa-check"></i><b>20.2.1</b> Config</a></li>
<li class="chapter" data-level="20.2.2" data-path="figure-4.html"><a href="figure-4.html#data-import"><i class="fa fa-check"></i><b>20.2.2</b> Data import</a></li>
<li class="chapter" data-level="20.2.3" data-path="figure-4.html"><a href="figure-4.html#plotting"><i class="fa fa-check"></i><b>20.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>21</b> Figure 5</a>
<ul>
<li class="chapter" data-level="21.1" data-path="figure-5.html"><a href="figure-5.html#summary-18"><i class="fa fa-check"></i><b>21.1</b> Summary</a></li>
<li class="chapter" data-level="21.2" data-path="figure-5.html"><a href="figure-5.html#details-of-plot_f5.r"><i class="fa fa-check"></i><b>21.2</b> Details of <code>plot_F5.R</code></a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="figure-5.html"><a href="figure-5.html#config-4"><i class="fa fa-check"></i><b>21.2.1</b> Config</a></li>
<li class="chapter" data-level="21.2.2" data-path="figure-5.html"><a href="figure-5.html#data-import-1"><i class="fa fa-check"></i><b>21.2.2</b> Data import</a></li>
<li class="chapter" data-level="21.2.3" data-path="figure-5.html"><a href="figure-5.html#plotting-1"><i class="fa fa-check"></i><b>21.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>22</b> Figure 6</a>
<ul>
<li class="chapter" data-level="22.1" data-path="figure-6.html"><a href="figure-6.html#summary-19"><i class="fa fa-check"></i><b>22.1</b> Summary</a></li>
<li class="chapter" data-level="22.2" data-path="figure-6.html"><a href="figure-6.html#details-of-plot_f6.r"><i class="fa fa-check"></i><b>22.2</b> Details of <code>plot_F6.R</code></a>
<ul>
<li class="chapter" data-level="22.2.1" data-path="figure-6.html"><a href="figure-6.html#config-5"><i class="fa fa-check"></i><b>22.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html"><i class="fa fa-check"></i><b>23</b> Supplementary Figure 1</a>
<ul>
<li class="chapter" data-level="23.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#summary-20"><i class="fa fa-check"></i><b>23.1</b> Summary</a></li>
<li class="chapter" data-level="23.2" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#details-of-plot_sf1.r"><i class="fa fa-check"></i><b>23.2</b> Details of <code>plot_SF1.R</code></a>
<ul>
<li class="chapter" data-level="23.2.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#config-6"><i class="fa fa-check"></i><b>23.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html"><i class="fa fa-check"></i><b>24</b> Supplementary Figure 2</a>
<ul>
<li class="chapter" data-level="24.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#summary-21"><i class="fa fa-check"></i><b>24.1</b> Summary</a></li>
<li class="chapter" data-level="24.2" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#details-of-plot_sf2.r"><i class="fa fa-check"></i><b>24.2</b> Details of <code>plot_SF2.R</code></a>
<ul>
<li class="chapter" data-level="24.2.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#config-7"><i class="fa fa-check"></i><b>24.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html"><i class="fa fa-check"></i><b>25</b> Supplementary Figure 3</a>
<ul>
<li class="chapter" data-level="25.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#summary-22"><i class="fa fa-check"></i><b>25.1</b> Summary</a></li>
<li class="chapter" data-level="25.2" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#details-of-plot_sf3.r"><i class="fa fa-check"></i><b>25.2</b> Details of <code>plot_SF3.R</code></a>
<ul>
<li class="chapter" data-level="25.2.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#config-8"><i class="fa fa-check"></i><b>25.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html"><i class="fa fa-check"></i><b>26</b> Supplementary Figure 4</a>
<ul>
<li class="chapter" data-level="26.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#summary-23"><i class="fa fa-check"></i><b>26.1</b> Summary</a></li>
<li class="chapter" data-level="26.2" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#details-of-plot_sf4.r"><i class="fa fa-check"></i><b>26.2</b> Details of <code>plot_SF4.R</code></a>
<ul>
<li class="chapter" data-level="26.2.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#config-9"><i class="fa fa-check"></i><b>26.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html"><i class="fa fa-check"></i><b>27</b> Supplementary Figure 5</a>
<ul>
<li class="chapter" data-level="27.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#summary-24"><i class="fa fa-check"></i><b>27.1</b> Summary</a></li>
<li class="chapter" data-level="27.2" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#details-of-plot_sf5.r"><i class="fa fa-check"></i><b>27.2</b> Details of <code>plot_SF5.R</code></a>
<ul>
<li class="chapter" data-level="27.2.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#config-10"><i class="fa fa-check"></i><b>27.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html"><i class="fa fa-check"></i><b>28</b> Supplementary Figure 6</a>
<ul>
<li class="chapter" data-level="28.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#summary-25"><i class="fa fa-check"></i><b>28.1</b> Summary</a></li>
<li class="chapter" data-level="28.2" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#details-of-plot_sf6.r"><i class="fa fa-check"></i><b>28.2</b> Details of <code>plot_SF6.R</code></a>
<ul>
<li class="chapter" data-level="28.2.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#config-11"><i class="fa fa-check"></i><b>28.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="29" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html"><i class="fa fa-check"></i><b>29</b> Supplementary Figure 7</a>
<ul>
<li class="chapter" data-level="29.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#summary-26"><i class="fa fa-check"></i><b>29.1</b> Summary</a></li>
<li class="chapter" data-level="29.2" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#details-of-plot_sf7.r"><i class="fa fa-check"></i><b>29.2</b> Details of <code>plot_SF7.R</code></a>
<ul>
<li class="chapter" data-level="29.2.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#config-12"><i class="fa fa-check"></i><b>29.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="30" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html"><i class="fa fa-check"></i><b>30</b> Supplementary Figure 8</a>
<ul>
<li class="chapter" data-level="30.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#summary-27"><i class="fa fa-check"></i><b>30.1</b> Summary</a></li>
<li class="chapter" data-level="30.2" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#details-of-plot_sf8.r"><i class="fa fa-check"></i><b>30.2</b> Details of <code>plot_SF8.R</code></a>
<ul>
<li class="chapter" data-level="30.2.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#config-13"><i class="fa fa-check"></i><b>30.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="31" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html"><i class="fa fa-check"></i><b>31</b> Supplementary Figure 9</a>
<ul>
<li class="chapter" data-level="31.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#summary-28"><i class="fa fa-check"></i><b>31.1</b> Summary</a></li>
<li class="chapter" data-level="31.2" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#details-of-plot_sf9.r"><i class="fa fa-check"></i><b>31.2</b> Details of <code>plot_SF9.R</code></a>
<ul>
<li class="chapter" data-level="31.2.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#config-14"><i class="fa fa-check"></i><b>31.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="32" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html"><i class="fa fa-check"></i><b>32</b> Supplementary Figure 10</a>
<ul>
<li class="chapter" data-level="32.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#summary-29"><i class="fa fa-check"></i><b>32.1</b> Summary</a></li>
<li class="chapter" data-level="32.2" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#details-of-plot_sf10.r"><i class="fa fa-check"></i><b>32.2</b> Details of <code>plot_SF10.R</code></a>
<ul>
<li class="chapter" data-level="32.2.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#config-15"><i class="fa fa-check"></i><b>32.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html"><i class="fa fa-check"></i><b>33</b> Supplementary Figure 11</a>
<ul>
<li class="chapter" data-level="33.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#summary-30"><i class="fa fa-check"></i><b>33.1</b> Summary</a></li>
<li class="chapter" data-level="33.2" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#details-of-plot_sf11.r"><i class="fa fa-check"></i><b>33.2</b> Details of <code>plot_SF11.R</code></a>
<ul>
<li class="chapter" data-level="33.2.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#config-16"><i class="fa fa-check"></i><b>33.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html"><i class="fa fa-check"></i><b>34</b> Supplementary Figure 12</a>
<ul>
<li class="chapter" data-level="34.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#summary-31"><i class="fa fa-check"></i><b>34.1</b> Summary</a></li>
<li class="chapter" data-level="34.2" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#details-of-plot_sf12.r"><i class="fa fa-check"></i><b>34.2</b> Details of <code>plot_SF12.R</code></a>
<ul>
<li class="chapter" data-level="34.2.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#config-17"><i class="fa fa-check"></i><b>34.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="35" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html"><i class="fa fa-check"></i><b>35</b> Supplementary Figure 13</a>
<ul>
<li class="chapter" data-level="35.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#summary-32"><i class="fa fa-check"></i><b>35.1</b> Summary</a></li>
<li class="chapter" data-level="35.2" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#details-of-plot_sf13.r"><i class="fa fa-check"></i><b>35.2</b> Details of <code>plot_SF13.R</code></a>
<ul>
<li class="chapter" data-level="35.2.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#config-18"><i class="fa fa-check"></i><b>35.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="36" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html"><i class="fa fa-check"></i><b>36</b> Supplementary Figure 14</a>
<ul>
<li class="chapter" data-level="36.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#summary-33"><i class="fa fa-check"></i><b>36.1</b> Summary</a></li>
<li class="chapter" data-level="36.2" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#details-of-plot_sf14.r"><i class="fa fa-check"></i><b>36.2</b> Details of <code>plot_SF14.R</code></a>
<ul>
<li class="chapter" data-level="36.2.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#config-19"><i class="fa fa-check"></i><b>36.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="37" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html"><i class="fa fa-check"></i><b>37</b> Supplementary Figure 15</a>
<ul>
<li class="chapter" data-level="37.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#summary-34"><i class="fa fa-check"></i><b>37.1</b> Summary</a></li>
<li class="chapter" data-level="37.2" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#details-of-plot_sf15.r"><i class="fa fa-check"></i><b>37.2</b> Details of <code>plot_SF15.R</code></a>
<ul>
<li class="chapter" data-level="37.2.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#config-20"><i class="fa fa-check"></i><b>37.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="38" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html"><i class="fa fa-check"></i><b>38</b> Supplementary Figure 16</a>
<ul>
<li class="chapter" data-level="38.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#summary-35"><i class="fa fa-check"></i><b>38.1</b> Summary</a></li>
<li class="chapter" data-level="38.2" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#details-of-plot_sf16.r"><i class="fa fa-check"></i><b>38.2</b> Details of <code>plot_SF16.R</code></a>
<ul>
<li class="chapter" data-level="38.2.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#config-21"><i class="fa fa-check"></i><b>38.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="git-14-analysis-xii-outlier-region-phylogenies" class="section level1" number="15">
<h1><span class="header-section-number">15</span> (git 14) Analysis XII (Outlier Region Phylogenies)</h1>
<p>This pipeline can be executed as follows:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb19-1"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span>/nf/14_analysis_phylo_regions</span>
<span id="cb19-2"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="ex">nextflow</span> run analysis_phylo_regions.nf</span></code></pre></div>
<div id="summary-13" class="section level2" number="15.1">
<h2><span class="header-section-number">15.1</span> Summary</h2>
<p>The phylogenies specific to particular differentiation outlier regions are reconstructed within the <a href="https://www.nextflow.io/"><strong>nextflow</strong></a> script <code>analysis_phylo_regions.nf</code> (located under <code>$BASE_DIR/nf/14_analysis_phylo_regions/</code>).
This includes both the <em>sample-level</em> as well as the <em>population-level</em> phylogenies.</p>
</div>
<div id="details-of-analysis_phylo_regions.nf" class="section level2" number="15.2">
<h2><span class="header-section-number">15.2</span> Details of <code>analysis_phylo_regions.nf</code></h2>
<blockquote>
<p>This part of the analysis was actually manged manually and not via <code>nextflow</code>.
We still report the analysis as a <code>.nf</code> script as we believe this is a cleaner and more concise report of the conducted analysis.</p>
</blockquote>
<div id="setup-6" class="section level3" number="15.2.1">
<h3><span class="header-section-number">15.2.1</span> Setup</h3>
<p>The nextflow script starts by opening the two specific linkage groups of the <em>all_bp</em> genotype data set and binding it to differentiation outlier IDs as well as to a reference table containing the genomic coordinates of all differentiation outlier regions.</p>
<div class="kclass">
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode">#<span class="hl opt">!/</span>usr<span class="hl opt">/</span>bin<span class="hl opt">/</span>env nextflow

<span class="hl slc">// ----------------------- DISCLAIMER ----------------------</span>
<span class="hl slc">// this pipeline was not actually run using nexflow,</span>
<span class="hl slc">// but managed manually</span>
<span class="hl slc">// ---------------------------------------------------------</span>

<span class="hl slc">// Region-specific phylogenies</span>
<span class="hl slc">// ---------------------------</span>

<span class="hl slc">// git 14.1</span>
<span class="hl slc">// bundle allBP files and outlier table</span>
<span class="hl kwa">Channel</span>
  .fromFilePairs<span class="hl opt">(</span><span class="hl str">&quot;../../1_genotyping/3_gatk_filtered/byLG/filterd.allBP.LG04.vcf.{gz,gz.tbi}&quot;</span><span class="hl opt">)</span>
  .concat<span class="hl opt">(</span><span class="hl kwa">Channel</span>.fromFilePairs<span class="hl opt">(</span><span class="hl str">&quot;../../1_genotyping/3_gatk_filtered/byLG/filterd.allBP.LG12.vcf.{gz,gz.tbi}&quot;</span><span class="hl opt">))</span>
  .concat<span class="hl opt">(</span><span class="hl kwa">Channel</span>.fromFilePairs<span class="hl opt">(</span><span class="hl str">&quot;../../1_genotyping/3_gatk_filtered/byLG/filterd.allBP.LG12.vcf.{gz,gz.tbi}&quot;</span><span class="hl opt">))</span>
  .merge<span class="hl opt">(</span><span class="hl kwa">Channel</span>.from<span class="hl opt">(</span><span class="hl str">&quot;LG04_1&quot;</span><span class="hl opt">,</span> <span class="hl str">&quot;LG12_3&quot;</span><span class="hl opt">,</span> <span class="hl str">&quot;LG12_4&quot;</span><span class="hl opt">))</span>
  .combine<span class="hl opt">(</span><span class="hl kwa">Channel</span>.fromPath<span class="hl opt">(</span><span class="hl str">&quot;../../ressources/focal_outlier.tsv&quot;</span><span class="hl opt">))</span>
  .set<span class="hl opt">{</span> vcf_lg_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Then, two different sample lists (both excluding hybrid samples, one with and one without Serranus outgroup samples) are loaded and bound to a sample mode identifier.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 14.2</span>
<span class="hl slc">// toggle sample modes (with / without Serranus outgroup)</span>
<span class="hl kwa">Channel</span>.fromPath<span class="hl opt">(</span><span class="hl str">&quot;../../ressources/samples_155.txt&quot;</span><span class="hl opt">)</span>
  .concat<span class="hl opt">(</span><span class="hl kwa">Channel</span>.fromPath<span class="hl opt">(</span><span class="hl str">&quot;../../ressources/samples_hybrids.txt&quot;</span><span class="hl opt">))</span>
  .merge<span class="hl opt">(</span><span class="hl kwa">Channel</span>.from<span class="hl opt">(</span><span class="hl str">&quot;155&quot;</span><span class="hl opt">,</span> <span class="hl str">&quot;hyS&quot;</span><span class="hl opt">))</span>
  .set<span class="hl opt">{</span> sample_mode_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Next, for each outlier region, the genotype data is subset to the respective outlier region.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 14.3</span>
<span class="hl slc">// subset genotypes to outlier region</span>
<span class="hl kwa">process</span> extract_regions <span class="hl opt">{</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> vcfIdx <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> outlierId <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> outlier_file <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> sample_file <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sample_mode <span class="hl opt">)</span> <span class="hl kwa">from</span> vcf_lg_ch.combine<span class="hl opt">(</span> sample_mode_ch <span class="hl opt">)</span>

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">.vcf&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> outlierId <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sample_mode <span class="hl opt">)</span> <span class="hl kwa">into</span> <span class="hl opt">(</span> vcf_raxml_ch<span class="hl opt">,</span> vcf_pomo_ch <span class="hl opt">)</span>

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   # Extract regions of interest from genotype data (allBP),</span>
<span class="hl str">   # remove hybrid / Serranus samples and indels; simplify headers</span>
<span class="hl str"></span>
<span class="hl str">   head -n 1</span> <span class="hl ipl">${outlier_file}</span> <span class="hl str">| cut -f 1-3 &gt; outlier.bed</span>
<span class="hl str">   grep</span> <span class="hl ipl">${outlierId} ${outlier_file}</span> <span class="hl str">| cut -f 1-3 &gt;&gt; outlier.bed</span>
<span class="hl str"></span>
<span class="hl str">   OUT_ALT=\$(echo</span> <span class="hl ipl">${outlierId}</span> <span class="hl str">| tr &#39;[:upper:]&#39; &#39;[:lower:]&#39; | sed &#39;s/_/./&#39;)</span>
<span class="hl str"></span>
<span class="hl str">   vcftools --gzvcf \</span>
<span class="hl str"></span>      <span class="hl ipl">${vcf[0]}</span> <span class="hl str">\</span>
<span class="hl str">     --bed outlier.bed \</span>
<span class="hl str">     --remove-indels \</span>
<span class="hl str">     --remove</span> <span class="hl ipl">${sample_file}</span> <span class="hl str">\</span>
<span class="hl str">     --recode \</span>
<span class="hl str">     --stdout | \</span>
<span class="hl str">     grep -v &#39;##&#39; &gt; \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">.vcf</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Then, for the <em>population-level</em> phylogenies, the genotypes are first converted to <code>fasta</code> format and then to a allele frequency format (<code>.cf</code>).
At that point, <code>iqtree2</code> is run to create the <em>population-level</em> phylogenies.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 14.4</span>
<span class="hl slc">// run iqtree under pomo model</span>
<span class="hl kwa">process</span> run_pomo <span class="hl opt">{</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../2_analysis/revPoMo/outlier_regions/&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span> 
    
    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> outlierId <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sample_mode <span class="hl opt">)</span> <span class="hl kwa">from</span> vcf_raxml_ch

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*_pop.cf.treefile&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> pomo_results_ch

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   OUT_ALT=\$(echo</span> <span class="hl ipl">${outlierId}</span> <span class="hl str">| tr &#39;[:upper:]&#39; &#39;[:lower:]&#39; | sed &#39;s/_/./&#39;)</span>
<span class="hl str"></span>
<span class="hl str">   # Convert to fasta format (Python scripts available at https://github.com/simonhmartin/genomics_general), picked up from 6.1.1 output</span>
<span class="hl str">   python \$SFTWR/genomics_general/VCF_processing/parseVCF.py -i</span> <span class="hl ipl">${vcf}</span> <span class="hl str">&gt; \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">.geno</span>
<span class="hl str"></span>
<span class="hl str">   python \$SFTWR/genomics_general/genoToSeq.py \</span>
<span class="hl str">       -g \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">.geno \</span>
<span class="hl str">       -s \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">.fas \</span>
<span class="hl str">       -f fasta \</span>
<span class="hl str">       --splitPhased</span>
<span class="hl str">   </span>
<span class="hl str">   # Reformat sample ids to provide population prefixes for cflib</span>
<span class="hl str">   sed -e &#39;s/-/_/g&#39; -e &#39;s/&gt;\(.*\)\([a-z]\{6\}\)_\([AB]\)/&gt;\2-\1_\3/g&#39; \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">.fas &gt; \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">_p.fas</span>
<span class="hl str"></span>
<span class="hl str">   # Convert to allele frequency format (cflib library available at https://github.com/pomo-dev/cflib)</span>
<span class="hl str">   \$SFTWR/cflib/FastaToCounts.py \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">_p.fas \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">_pop.cf</span>
<span class="hl str"></span>
<span class="hl str">   # IQTREE analysis under PoMo model</span>
<span class="hl str">   iqtree2 \</span>
<span class="hl str">       -nt 16 \</span>
<span class="hl str">       -s \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">_pop.cf \</span>
<span class="hl str">       -m HKY+F+P+N9+G4 \</span>
<span class="hl str">       -b 100</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>For the <em>sample-level</em> phylogenies, the genotypes are also converted to <code>fasta</code> format.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 14.5</span>
<span class="hl slc">// convert genotypes to fasta for raxml</span>
<span class="hl kwa">process</span> conversion_raxml <span class="hl opt">{</span>
    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> outlierId <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sample_mode <span class="hl opt">)</span> <span class="hl kwa">from</span> vcf_pomo_ch
    
    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> outlierId <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sample_mode <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*N.fas&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> outlier_regions_ch

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   OUT_ALT=\$(echo</span> <span class="hl ipl">${outlierId}</span> <span class="hl str">| tr &#39;[:upper:]&#39; &#39;[:lower:]&#39; | sed &#39;s/_/./&#39;)</span>
<span class="hl str"></span>
<span class="hl str">   # Replace unknown character states and asterisks (deletions as encoded by GATK) with &quot;N&quot;</span>
<span class="hl str">   vcf-to-tab &lt;</span> <span class="hl ipl">${vcf}</span> <span class="hl str">| sed -e &#39;s/</span><span class="hl esc">\\</span><span class="hl str">.</span><span class="hl esc">\\</span><span class="hl str">/</span><span class="hl esc">\\</span><span class="hl str">./N</span><span class="hl esc">\\</span><span class="hl str">/N/g&#39; -e &#39;s/[ACGTN</span><span class="hl esc">\\</span><span class="hl str">*]</span><span class="hl esc">\\</span><span class="hl str">/</span><span class="hl esc">\\</span><span class="hl str">*/N</span><span class="hl esc">\\</span><span class="hl str">/N/g&#39; &gt; \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">N.tab</span>
<span class="hl str"></span>
<span class="hl str">   # Convert to fasta format (Perl script available at https://github.com/JinfengChen/vcf-tab-to-fasta)</span>
<span class="hl str">   wget https://raw.githubusercontent.com/JinfengChen/vcf-tab-to-fasta/master/vcf_tab_to_fasta_alignment.pl</span>
<span class="hl str">   perl ~/apps/vcf-tab-to-fasta/vcf_tab_to_fasta_alignment.pl -i \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">N.tab &gt; \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">N.fas</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Then, <code>raxml</code> is run directly on the <code>fasta</code> files.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 14.6</span>
<span class="hl slc">// run raxml</span>
<span class="hl kwa">process</span> run_raxml <span class="hl opt">{</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../2_analysis/raxml/&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span> 
    
    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> outlierId <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sample_mode <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> fas <span class="hl opt">)</span> <span class="hl kwa">from</span> outlier_regions_ch

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*.raxml.support&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> outlier_results_ch

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   OUT_ALT=\$(echo</span> <span class="hl ipl">${outlierId}</span> <span class="hl str">| tr &#39;[:upper:]&#39; &#39;[:lower:]&#39; | sed &#39;s/_/./&#39;)</span>
<span class="hl str"></span>
<span class="hl str">   # Reconstruct phylogenies</span>
<span class="hl str">   raxml-NG --all \</span>
<span class="hl str">       --msa</span> <span class="hl ipl">${fas}</span> <span class="hl str">\</span>
<span class="hl str">       --model GTR+G \</span>
<span class="hl str">       --tree pars{10},rand{10} \</span>
<span class="hl str">       --bs-trees 100 \</span>
<span class="hl str">       --threads 24 \</span>
<span class="hl str">       --worker 8 \</span>
<span class="hl str">       --seed 123 \</span>
<span class="hl str">       --prefix \</span><span class="hl ipl">${OUT_ALT}</span><span class="hl str">_</span><span class="hl ipl">${sample_mode}</span><span class="hl str">N</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
</div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="git-13-analysis-xi-whole-genome-phylogenies.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="git-15-data-visualization.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
