<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>17 (git 16) Analysis XIV (Identity by Descent) | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Rapid radiation in a highly diverse marine environment’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="17 (git 16) Analysis XIV (Identity by Descent) | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Rapid radiation in a highly diverse marine environment’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="17 (git 16) Analysis XIV (Identity by Descent) | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Rapid radiation in a highly diverse marine environment’ by Hench, Helmkampf, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench and Martin Helmkampf" />


<meta name="date" content="2022-01-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"/>
<link rel="next" href="git-17-analysis-xv-dstats.html"/>
<script src="libs/header-attrs-2.7.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#r-setup"><i class="fa fa-check"></i><b>1.4</b> R setup</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a>
<ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a>
<ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy.html"><a href="git-4-analysis-ii-dxy.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i>)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy.html"><a href="git-4-analysis-ii-dxy.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy.html"><a href="git-4-analysis-ii-dxy.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy.html"><a href="git-4-analysis-ii-dxy.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (topology weighting)</a>
<ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (Principal Component Analysis)</a>
<ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#details-of-analysis_pca.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_pca.nf</code></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#setup-2"><i class="fa fa-check"></i><b>8.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-4"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-5"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-6"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html"><i class="fa fa-check"></i><b>12</b> (git 11) Analysis IX (Allele Age)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#summary-10"><i class="fa fa-check"></i><b>12.1</b> Summary</a></li>
<li class="chapter" data-level="12.2" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#details-of-analysis_allele_age.nf"><i class="fa fa-check"></i><b>12.2</b> Details of <code>analysis_allele_age.nf</code></a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#setup-3"><i class="fa fa-check"></i><b>12.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html"><i class="fa fa-check"></i><b>13</b> (git 12) Analysis X (<i>F<sub>ST</sub></i> Permutation Test)</a>
<ul>
<li class="chapter" data-level="13.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#summary-11"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#details-of-analysis_fst_sign.nf"><i class="fa fa-check"></i><b>13.2</b> Details of <code>analysis_fst_sign.nf</code></a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#setup-4"><i class="fa fa-check"></i><b>13.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html"><i class="fa fa-check"></i><b>14</b> (git 13) Analysis XI (Whole Genome Phylogenies)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#summary-12"><i class="fa fa-check"></i><b>14.1</b> Summary</a></li>
<li class="chapter" data-level="14.2" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#details-of-analysis_phylo_whg.nf"><i class="fa fa-check"></i><b>14.2</b> Details of <code>analysis_phylo_whg.nf</code></a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#setup-5"><i class="fa fa-check"></i><b>14.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html"><i class="fa fa-check"></i><b>15</b> (git 14) Analysis XII (Outlier Region Phylogenies)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#summary-13"><i class="fa fa-check"></i><b>15.1</b> Summary</a></li>
<li class="chapter" data-level="15.2" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#details-of-analysis_phylo_regions.nf"><i class="fa fa-check"></i><b>15.2</b> Details of <code>analysis_phylo_regions.nf</code></a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#setup-6"><i class="fa fa-check"></i><b>15.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"><a href="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"><i class="fa fa-check"></i><b>16</b> (git 15) Analysis XIII (diversity with and without outlier regions)</a>
<ul>
<li class="chapter" data-level="16.1" data-path="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"><a href="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html#summary-14"><i class="fa fa-check"></i><b>16.1</b> Summary</a></li>
<li class="chapter" data-level="16.2" data-path="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"><a href="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html#details-of-analysis_pi.nf"><i class="fa fa-check"></i><b>16.2</b> Details of <code>analysis_pi.nf</code></a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"><a href="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html#setup-7"><i class="fa fa-check"></i><b>16.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="git-16-analysis-xiv-identity-by-descent.html"><a href="git-16-analysis-xiv-identity-by-descent.html"><i class="fa fa-check"></i><b>17</b> (git 16) Analysis XIV (Identity by Descent)</a>
<ul>
<li class="chapter" data-level="17.1" data-path="git-16-analysis-xiv-identity-by-descent.html"><a href="git-16-analysis-xiv-identity-by-descent.html#summary-15"><i class="fa fa-check"></i><b>17.1</b> Summary</a></li>
<li class="chapter" data-level="17.2" data-path="git-16-analysis-xiv-identity-by-descent.html"><a href="git-16-analysis-xiv-identity-by-descent.html#details-of-analysis_ibd.nf"><i class="fa fa-check"></i><b>17.2</b> Details of <code>analysis_ibd.nf</code></a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="git-16-analysis-xiv-identity-by-descent.html"><a href="git-16-analysis-xiv-identity-by-descent.html#setup-8"><i class="fa fa-check"></i><b>17.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="git-17-analysis-xv-dstats.html"><a href="git-17-analysis-xv-dstats.html"><i class="fa fa-check"></i><b>18</b> (git 17) Analysis XV (dstats)</a>
<ul>
<li class="chapter" data-level="18.1" data-path="git-17-analysis-xv-dstats.html"><a href="git-17-analysis-xv-dstats.html#summary-16"><i class="fa fa-check"></i><b>18.1</b> Summary</a></li>
<li class="chapter" data-level="18.2" data-path="git-17-analysis-xv-dstats.html"><a href="git-17-analysis-xv-dstats.html#details-of-analysis_dstats.nf"><i class="fa fa-check"></i><b>18.2</b> Details of <code>analysis_dstats.nf</code></a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="git-17-analysis-xv-dstats.html"><a href="git-17-analysis-xv-dstats.html#setup-9"><i class="fa fa-check"></i><b>18.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html"><a href="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html"><i class="fa fa-check"></i><b>19</b> (git 18) Genotyping III (all callable sites for mtDNA and unplaced contigs)</a>
<ul>
<li class="chapter" data-level="19.1" data-path="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html"><a href="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html#summary-17"><i class="fa fa-check"></i><b>19.1</b> Summary</a></li>
<li class="chapter" data-level="19.2" data-path="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html"><a href="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html#details-of-genotyping_all_basepairs_mt.nf"><i class="fa fa-check"></i><b>19.2</b> Details of <code>genotyping_all_basepairs_mt.nf</code></a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html"><a href="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html#data-preparation-7"><i class="fa fa-check"></i><b>19.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="git-19-analysis-xvi-serraninae-phylogeny.html"><a href="git-19-analysis-xvi-serraninae-phylogeny.html"><i class="fa fa-check"></i><b>20</b> (git 19) Analysis XVI (Serraninae Phylogeny)</a>
<ul>
<li class="chapter" data-level="20.1" data-path="git-19-analysis-xvi-serraninae-phylogeny.html"><a href="git-19-analysis-xvi-serraninae-phylogeny.html#summary-18"><i class="fa fa-check"></i><b>20.1</b> Summary</a></li>
<li class="chapter" data-level="20.2" data-path="git-19-analysis-xvi-serraninae-phylogeny.html"><a href="git-19-analysis-xvi-serraninae-phylogeny.html#details-of-analysis_phylo_serraninae.nf"><i class="fa fa-check"></i><b>20.2</b> Details of <code>analysis_phylo_serraninae.nf</code></a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="git-19-analysis-xvi-serraninae-phylogeny.html"><a href="git-19-analysis-xvi-serraninae-phylogeny.html#setup-10"><i class="fa fa-check"></i><b>20.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="git-20-data-visualization.html"><a href="git-20-data-visualization.html"><i class="fa fa-check"></i><b>21</b> (git 20) Data Visualization</a></li>
<li class="chapter" data-level="22" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>22</b> Figure 1</a>
<ul>
<li class="chapter" data-level="22.1" data-path="figure-1.html"><a href="figure-1.html#summary-19"><i class="fa fa-check"></i><b>22.1</b> Summary</a></li>
<li class="chapter" data-level="22.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>22.2</b> Details of <code>plot_F1.R</code></a>
<ul>
<li class="chapter" data-level="22.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>22.2.1</b> Config</a></li>
<li class="chapter" data-level="22.2.2" data-path="figure-1.html"><a href="figure-1.html#actual-script-start"><i class="fa fa-check"></i><b>22.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>23</b> Figure 2</a>
<ul>
<li class="chapter" data-level="23.1" data-path="figure-2.html"><a href="figure-2.html#summary-20"><i class="fa fa-check"></i><b>23.1</b> Summary</a></li>
<li class="chapter" data-level="23.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>23.2</b> Details of <code>plot_F2.R</code></a>
<ul>
<li class="chapter" data-level="23.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>23.2.1</b> Config</a></li>
<li class="chapter" data-level="23.2.2" data-path="figure-2.html"><a href="figure-2.html#actual-script-start-1"><i class="fa fa-check"></i><b>23.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>24</b> Figure 3</a>
<ul>
<li class="chapter" data-level="24.1" data-path="figure-3.html"><a href="figure-3.html#summary-21"><i class="fa fa-check"></i><b>24.1</b> Summary</a></li>
<li class="chapter" data-level="24.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>24.2</b> Details of <code>plot_F3.R</code></a>
<ul>
<li class="chapter" data-level="24.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>24.2.1</b> Config</a></li>
<li class="chapter" data-level="24.2.2" data-path="figure-3.html"><a href="figure-3.html#actual-script-start-2"><i class="fa fa-check"></i><b>24.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>25</b> Figure 4</a>
<ul>
<li class="chapter" data-level="25.1" data-path="figure-4.html"><a href="figure-4.html#summary-22"><i class="fa fa-check"></i><b>25.1</b> Summary</a></li>
<li class="chapter" data-level="25.2" data-path="figure-4.html"><a href="figure-4.html#details-of-plot_f4.r"><i class="fa fa-check"></i><b>25.2</b> Details of <code>plot_F4.R</code></a>
<ul>
<li class="chapter" data-level="25.2.1" data-path="figure-4.html"><a href="figure-4.html#config-3"><i class="fa fa-check"></i><b>25.2.1</b> Config</a></li>
<li class="chapter" data-level="25.2.2" data-path="figure-4.html"><a href="figure-4.html#actual-script-start-3"><i class="fa fa-check"></i><b>25.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>26</b> Figure 5</a>
<ul>
<li class="chapter" data-level="26.1" data-path="figure-5.html"><a href="figure-5.html#summary-23"><i class="fa fa-check"></i><b>26.1</b> Summary</a></li>
<li class="chapter" data-level="26.2" data-path="figure-5.html"><a href="figure-5.html#details-of-plot_f5.r"><i class="fa fa-check"></i><b>26.2</b> Details of <code>plot_F5.R</code></a>
<ul>
<li class="chapter" data-level="26.2.1" data-path="figure-5.html"><a href="figure-5.html#config-4"><i class="fa fa-check"></i><b>26.2.1</b> Config</a></li>
<li class="chapter" data-level="26.2.2" data-path="figure-5.html"><a href="figure-5.html#actual-script-start-4"><i class="fa fa-check"></i><b>26.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>27</b> Figure 6</a>
<ul>
<li class="chapter" data-level="27.1" data-path="figure-6.html"><a href="figure-6.html#summary-24"><i class="fa fa-check"></i><b>27.1</b> Summary</a></li>
<li class="chapter" data-level="27.2" data-path="figure-6.html"><a href="figure-6.html#details-of-plot_f6.r"><i class="fa fa-check"></i><b>27.2</b> Details of <code>plot_F6.R</code></a>
<ul>
<li class="chapter" data-level="27.2.1" data-path="figure-6.html"><a href="figure-6.html#config-5"><i class="fa fa-check"></i><b>27.2.1</b> Config</a></li>
<li class="chapter" data-level="27.2.2" data-path="figure-6.html"><a href="figure-6.html#actual-script-start-5"><i class="fa fa-check"></i><b>27.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html"><i class="fa fa-check"></i><b>28</b> Supplementary Figure 1</a>
<ul>
<li class="chapter" data-level="28.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#summary-25"><i class="fa fa-check"></i><b>28.1</b> Summary</a></li>
<li class="chapter" data-level="28.2" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#details-of-plot_sf1.r"><i class="fa fa-check"></i><b>28.2</b> Details of <code>plot_SF1.R</code></a>
<ul>
<li class="chapter" data-level="28.2.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#config-6"><i class="fa fa-check"></i><b>28.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="29" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html"><i class="fa fa-check"></i><b>29</b> Supplementary Figure 2</a>
<ul>
<li class="chapter" data-level="29.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#summary-26"><i class="fa fa-check"></i><b>29.1</b> Summary</a></li>
<li class="chapter" data-level="29.2" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#details-of-plot_sf2.r"><i class="fa fa-check"></i><b>29.2</b> Details of <code>plot_SF2.R</code></a>
<ul>
<li class="chapter" data-level="29.2.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#config-7"><i class="fa fa-check"></i><b>29.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="30" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html"><i class="fa fa-check"></i><b>30</b> Supplementary Figure 3</a>
<ul>
<li class="chapter" data-level="30.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#summary-27"><i class="fa fa-check"></i><b>30.1</b> Summary</a></li>
<li class="chapter" data-level="30.2" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#details-of-plot_sf3.r"><i class="fa fa-check"></i><b>30.2</b> Details of <code>plot_SF3.R</code></a>
<ul>
<li class="chapter" data-level="30.2.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#config-8"><i class="fa fa-check"></i><b>30.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="31" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html"><i class="fa fa-check"></i><b>31</b> Supplementary Figure 4</a>
<ul>
<li class="chapter" data-level="31.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#summary-28"><i class="fa fa-check"></i><b>31.1</b> Summary</a></li>
<li class="chapter" data-level="31.2" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#details-of-plot_sf4.r"><i class="fa fa-check"></i><b>31.2</b> Details of <code>plot_SF4.R</code></a>
<ul>
<li class="chapter" data-level="31.2.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#config-9"><i class="fa fa-check"></i><b>31.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="32" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html"><i class="fa fa-check"></i><b>32</b> Supplementary Figure 5</a>
<ul>
<li class="chapter" data-level="32.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#summary-29"><i class="fa fa-check"></i><b>32.1</b> Summary</a></li>
<li class="chapter" data-level="32.2" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#details-of-plot_sf5.r"><i class="fa fa-check"></i><b>32.2</b> Details of <code>plot_SF5.R</code></a>
<ul>
<li class="chapter" data-level="32.2.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#config-10"><i class="fa fa-check"></i><b>32.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html"><i class="fa fa-check"></i><b>33</b> Supplementary Figure 6</a>
<ul>
<li class="chapter" data-level="33.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#summary-30"><i class="fa fa-check"></i><b>33.1</b> Summary</a></li>
<li class="chapter" data-level="33.2" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#details-of-plot_sf6.r"><i class="fa fa-check"></i><b>33.2</b> Details of <code>plot_SF6.R</code></a>
<ul>
<li class="chapter" data-level="33.2.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#config-11"><i class="fa fa-check"></i><b>33.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html"><i class="fa fa-check"></i><b>34</b> Supplementary Figure 7</a>
<ul>
<li class="chapter" data-level="34.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#summary-31"><i class="fa fa-check"></i><b>34.1</b> Summary</a></li>
<li class="chapter" data-level="34.2" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#details-of-plot_sf7.r"><i class="fa fa-check"></i><b>34.2</b> Details of <code>plot_SF7.R</code></a>
<ul>
<li class="chapter" data-level="34.2.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#config-12"><i class="fa fa-check"></i><b>34.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="35" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html"><i class="fa fa-check"></i><b>35</b> Supplementary Figure 8</a>
<ul>
<li class="chapter" data-level="35.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#summary-32"><i class="fa fa-check"></i><b>35.1</b> Summary</a></li>
<li class="chapter" data-level="35.2" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#details-of-plot_sf8.r"><i class="fa fa-check"></i><b>35.2</b> Details of <code>plot_SF8.R</code></a>
<ul>
<li class="chapter" data-level="35.2.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#config-13"><i class="fa fa-check"></i><b>35.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="36" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html"><i class="fa fa-check"></i><b>36</b> Supplementary Figure 9</a>
<ul>
<li class="chapter" data-level="36.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#summary-33"><i class="fa fa-check"></i><b>36.1</b> Summary</a></li>
<li class="chapter" data-level="36.2" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#details-of-plot_sf9.r"><i class="fa fa-check"></i><b>36.2</b> Details of <code>plot_SF9.R</code></a>
<ul>
<li class="chapter" data-level="36.2.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#config-14"><i class="fa fa-check"></i><b>36.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="37" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html"><i class="fa fa-check"></i><b>37</b> Supplementary Figure 10</a>
<ul>
<li class="chapter" data-level="37.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#summary-34"><i class="fa fa-check"></i><b>37.1</b> Summary</a></li>
<li class="chapter" data-level="37.2" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#details-of-plot_sf10.r"><i class="fa fa-check"></i><b>37.2</b> Details of <code>plot_SF10.R</code></a>
<ul>
<li class="chapter" data-level="37.2.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#config-15"><i class="fa fa-check"></i><b>37.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="38" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html"><i class="fa fa-check"></i><b>38</b> Supplementary Figure 11</a>
<ul>
<li class="chapter" data-level="38.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#summary-35"><i class="fa fa-check"></i><b>38.1</b> Summary</a></li>
<li class="chapter" data-level="38.2" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#details-of-plot_sf11.r"><i class="fa fa-check"></i><b>38.2</b> Details of <code>plot_SF11.R</code></a>
<ul>
<li class="chapter" data-level="38.2.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#config-16"><i class="fa fa-check"></i><b>38.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="39" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html"><i class="fa fa-check"></i><b>39</b> Supplementary Figure 12</a>
<ul>
<li class="chapter" data-level="39.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#summary-36"><i class="fa fa-check"></i><b>39.1</b> Summary</a></li>
<li class="chapter" data-level="39.2" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#details-of-plot_sf12.r"><i class="fa fa-check"></i><b>39.2</b> Details of <code>plot_SF12.R</code></a>
<ul>
<li class="chapter" data-level="39.2.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#config-17"><i class="fa fa-check"></i><b>39.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="40" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html"><i class="fa fa-check"></i><b>40</b> Supplementary Figure 13</a>
<ul>
<li class="chapter" data-level="40.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#summary-37"><i class="fa fa-check"></i><b>40.1</b> Summary</a></li>
<li class="chapter" data-level="40.2" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#details-of-plot_sf13.r"><i class="fa fa-check"></i><b>40.2</b> Details of <code>plot_SF13.R</code></a>
<ul>
<li class="chapter" data-level="40.2.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#config-18"><i class="fa fa-check"></i><b>40.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="41" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html"><i class="fa fa-check"></i><b>41</b> Supplementary Figure 14</a>
<ul>
<li class="chapter" data-level="41.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#summary-38"><i class="fa fa-check"></i><b>41.1</b> Summary</a></li>
<li class="chapter" data-level="41.2" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#details-of-plot_sf14.r"><i class="fa fa-check"></i><b>41.2</b> Details of <code>plot_SF14.R</code></a>
<ul>
<li class="chapter" data-level="41.2.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#config-19"><i class="fa fa-check"></i><b>41.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="42" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html"><i class="fa fa-check"></i><b>42</b> Supplementary Figure 15</a>
<ul>
<li class="chapter" data-level="42.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#summary-39"><i class="fa fa-check"></i><b>42.1</b> Summary</a></li>
<li class="chapter" data-level="42.2" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#details-of-plot_sf15.r"><i class="fa fa-check"></i><b>42.2</b> Details of <code>plot_SF15.R</code></a>
<ul>
<li class="chapter" data-level="42.2.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#config-20"><i class="fa fa-check"></i><b>42.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="43" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html"><i class="fa fa-check"></i><b>43</b> Supplementary Figure 16</a>
<ul>
<li class="chapter" data-level="43.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#summary-40"><i class="fa fa-check"></i><b>43.1</b> Summary</a></li>
<li class="chapter" data-level="43.2" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#details-of-plot_sf16.r"><i class="fa fa-check"></i><b>43.2</b> Details of <code>plot_SF16.R</code></a>
<ul>
<li class="chapter" data-level="43.2.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#config-21"><i class="fa fa-check"></i><b>43.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="44" data-path="supplementary-figure-17.html"><a href="supplementary-figure-17.html"><i class="fa fa-check"></i><b>44</b> Supplementary Figure 17</a>
<ul>
<li class="chapter" data-level="44.1" data-path="supplementary-figure-17.html"><a href="supplementary-figure-17.html#summary-41"><i class="fa fa-check"></i><b>44.1</b> Summary</a></li>
<li class="chapter" data-level="44.2" data-path="supplementary-figure-17.html"><a href="supplementary-figure-17.html#details-of-plot_sf17.r"><i class="fa fa-check"></i><b>44.2</b> Details of <code>plot_SF17.R</code></a>
<ul>
<li class="chapter" data-level="44.2.1" data-path="supplementary-figure-17.html"><a href="supplementary-figure-17.html#config-22"><i class="fa fa-check"></i><b>44.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="45" data-path="supplementary-figure-18.html"><a href="supplementary-figure-18.html"><i class="fa fa-check"></i><b>45</b> Supplementary Figure 18</a>
<ul>
<li class="chapter" data-level="45.1" data-path="supplementary-figure-18.html"><a href="supplementary-figure-18.html#summary-42"><i class="fa fa-check"></i><b>45.1</b> Summary</a></li>
<li class="chapter" data-level="45.2" data-path="supplementary-figure-18.html"><a href="supplementary-figure-18.html#details-of-plot_sf18.r"><i class="fa fa-check"></i><b>45.2</b> Details of <code>plot_SF18.R</code></a>
<ul>
<li class="chapter" data-level="45.2.1" data-path="supplementary-figure-18.html"><a href="supplementary-figure-18.html#config-23"><i class="fa fa-check"></i><b>45.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="46" data-path="supplementary-figure-19.html"><a href="supplementary-figure-19.html"><i class="fa fa-check"></i><b>46</b> Supplementary Figure 19</a>
<ul>
<li class="chapter" data-level="46.1" data-path="supplementary-figure-19.html"><a href="supplementary-figure-19.html#summary-43"><i class="fa fa-check"></i><b>46.1</b> Summary</a></li>
<li class="chapter" data-level="46.2" data-path="supplementary-figure-19.html"><a href="supplementary-figure-19.html#details-of-plot_sf19.r"><i class="fa fa-check"></i><b>46.2</b> Details of <code>plot_SF19.R</code></a>
<ul>
<li class="chapter" data-level="46.2.1" data-path="supplementary-figure-19.html"><a href="supplementary-figure-19.html#config-24"><i class="fa fa-check"></i><b>46.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="47" data-path="supplementary-figure-20.html"><a href="supplementary-figure-20.html"><i class="fa fa-check"></i><b>47</b> Supplementary Figure 20</a>
<ul>
<li class="chapter" data-level="47.1" data-path="supplementary-figure-20.html"><a href="supplementary-figure-20.html#summary-44"><i class="fa fa-check"></i><b>47.1</b> Summary</a></li>
<li class="chapter" data-level="47.2" data-path="supplementary-figure-20.html"><a href="supplementary-figure-20.html#details-of-plot_sf20.r"><i class="fa fa-check"></i><b>47.2</b> Details of <code>plot_SF20.R</code></a>
<ul>
<li class="chapter" data-level="47.2.1" data-path="supplementary-figure-20.html"><a href="supplementary-figure-20.html#config-25"><i class="fa fa-check"></i><b>47.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="48" data-path="supplementary-figure-21.html"><a href="supplementary-figure-21.html"><i class="fa fa-check"></i><b>48</b> Supplementary Figure 21</a>
<ul>
<li class="chapter" data-level="48.1" data-path="supplementary-figure-21.html"><a href="supplementary-figure-21.html#summary-45"><i class="fa fa-check"></i><b>48.1</b> Summary</a></li>
<li class="chapter" data-level="48.2" data-path="supplementary-figure-21.html"><a href="supplementary-figure-21.html#details-of-plot_sf21.r"><i class="fa fa-check"></i><b>48.2</b> Details of <code>plot_SF21.R</code></a>
<ul>
<li class="chapter" data-level="48.2.1" data-path="supplementary-figure-21.html"><a href="supplementary-figure-21.html#config-26"><i class="fa fa-check"></i><b>48.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="git-16-analysis-xiv-identity-by-descent" class="section level1" number="17">
<h1><span class="header-section-number">17</span> (git 16) Analysis XIV (Identity by Descent)</h1>
<p>This pipeline can be executed as follows:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb21-1"><a href="git-16-analysis-xiv-identity-by-descent.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span>/nf/16_analysis_ibd</span>
<span id="cb21-2"><a href="git-16-analysis-xiv-identity-by-descent.html#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="ex">nextflow</span> run analysis_ibd.nf <span class="at">-c</span> ../../nextflow.config <span class="at">-resume</span></span></code></pre></div>
<div id="summary-15" class="section level2" number="17.1">
<h2><span class="header-section-number">17.1</span> Summary</h2>
<p>Identity by descent is computed within the <a href="https://www.nextflow.io/"><strong>nextflow</strong></a> script <code>analysis_ibd.nf</code> (located under <code>$BASE_DIR/nf/16_analysis_ibd/</code>).
It takes the phased genotypes and computes the IBD segments.
Below is an overview of the steps involved in the analysis.</p>
</div>
<div id="details-of-analysis_ibd.nf" class="section level2" number="17.2">
<h2><span class="header-section-number">17.2</span> Details of <code>analysis_ibd.nf</code></h2>
<div id="setup-8" class="section level3" number="17.2.1">
<h3><span class="header-section-number">17.2.1</span> Setup</h3>
<p>The nextflow script starts by opening the phased genotypes.</p>
<div class="kclass">
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode">#<span class="hl opt">!/</span>usr<span class="hl opt">/</span>bin<span class="hl opt">/</span>env nextflow
<span class="hl slc">// This pipeline includes the analysis calculating the pair-wise IBD</span>

<span class="hl slc">// git 16.1</span>
<span class="hl slc">// load genotypes</span>
<span class="hl kwa">Channel</span>
    .fromFilePairs<span class="hl opt">(</span><span class="hl str">&quot;../../1_genotyping/4_phased/phased_mac2.vcf.{gz,gz.tbi}&quot;</span><span class="hl opt">)</span>
    .set<span class="hl opt">{</span> genotypes_raw_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Then, outgroup samples are removed.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 16.2</span>
<span class="hl slc">// drop outgroups</span>
<span class="hl kwa">process</span> drop_outgroups <span class="hl opt">{</span>
    
    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> vcfId<span class="hl opt">,</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">)</span> <span class="hl kwa">from</span> genotypes_raw_ch

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;phased_mac2.no_outgroup.vcf.gz&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> genotypes_ch

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   vcfsamplenames</span> <span class="hl ipl">${vcf[0]}</span> <span class="hl str">| \</span>
<span class="hl str">       grep &quot;tor</span><span class="hl esc">\\</span><span class="hl str">|tab</span><span class="hl esc">\\</span><span class="hl str">|flo&quot; &gt; outrgr.pop</span>
<span class="hl str"></span>
<span class="hl str">   vcftools --gzvcf</span> <span class="hl ipl">${vcf[0]}</span> <span class="hl str">\</span>
<span class="hl str">       --remove outrgr.pop \</span>
<span class="hl str">       --recode \</span>
<span class="hl str">       --stdout | bgzip &gt; phased_mac2.no_outgroup.vcf.gz</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>The IDB analysis is run for several IBD fragment size thresholds, so here we are initializing the different thresholds.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 16.3</span>
<span class="hl slc">// Set IBD fragment sizes</span>
<span class="hl kwa">Channel</span>
    .from<span class="hl opt">([[</span> <span class="hl num">25000</span><span class="hl opt">,</span> <span class="hl num">10000</span><span class="hl opt">,</span> <span class="hl num">7</span> <span class="hl opt">],</span>
           <span class="hl opt">[</span> <span class="hl num">15000</span><span class="hl opt">,</span> <span class="hl num">7500</span><span class="hl opt">,</span> <span class="hl num">10</span> <span class="hl opt">],</span>
           <span class="hl opt">[</span> <span class="hl num">10000</span><span class="hl opt">,</span> <span class="hl num">5000</span><span class="hl opt">,</span> <span class="hl num">8</span> <span class="hl opt">]])</span>
    .set<span class="hl opt">{</span> seq_sizes_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Also, the IBD analysis is additionally run with specific parts of the genome excluded - here the different exclusion models are initialized.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 16.4</span>
<span class="hl slc">// Set filter mode</span>
<span class="hl kwa">Channel</span>
    .from<span class="hl opt">([[</span><span class="hl str">&quot;direct&quot;</span><span class="hl opt">,</span> <span class="hl str">&quot;&quot;</span><span class="hl opt">],</span>
           <span class="hl opt">[</span><span class="hl str">&quot;bed&quot;</span><span class="hl opt">,</span> <span class="hl str">&quot;95&quot;</span><span class="hl opt">]])</span>
    .set<span class="hl opt">{</span> filtermode_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Truffle is run to compute the IBD segments.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 16.5</span>
<span class="hl slc">// run truffle</span>
<span class="hl kwa">process</span> run_truffle <span class="hl opt">{</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../2_analysis/ibd/&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sz1 <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sz2 <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sz3 <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> mode <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> excluding <span class="hl opt">)</span> <span class="hl kwa">from</span> genotypes_ch.combine<span class="hl opt">(</span> seq_sizes_ch <span class="hl opt">)</span>.combine<span class="hl opt">(</span> filtermode_ch <span class="hl opt">)</span>

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sz3 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.ibd.tsv&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.segments.tsv&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> truffle_result

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    if<span class="hl opt">(</span> mode <span class="hl opt">==</span> <span class="hl str">&#39;direct&#39;</span> <span class="hl opt">)</span>
        <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">       truffle \</span>
<span class="hl str">           --vcf</span> <span class="hl ipl">${vcf}</span> <span class="hl str">\</span>
<span class="hl str">           --segments \</span>
<span class="hl str">           --nofiltering \</span>
<span class="hl str">           --ibs1markers</span> <span class="hl ipl">${sz1}</span> <span class="hl str">\</span>
<span class="hl str">           --ibs2markers</span> <span class="hl ipl">${sz2}</span> <span class="hl str">\</span>
<span class="hl str">           --out no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span> <span class="hl str">\</span>
<span class="hl str">           --cpu 8</span>
<span class="hl str">       </span>
<span class="hl str">       sed &#39;s/^</span><span class="hl esc">\\</span><span class="hl str">s*//g; s/</span><span class="hl esc">\\</span><span class="hl str">s</span><span class="hl esc">\\</span><span class="hl str">+/</span><span class="hl esc">\\</span><span class="hl str">t/g&#39; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.ibd &gt; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.ibd.tsv</span>
<span class="hl str">       sed &#39;s/^</span><span class="hl esc">\\</span><span class="hl str">s*//g; s/</span><span class="hl esc">\\</span><span class="hl str">s</span><span class="hl esc">\\</span><span class="hl str">+/</span><span class="hl esc">\\</span><span class="hl str">t/g&#39; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.segments &gt; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.segments.tsv</span>
<span class="hl str">       &quot;&quot;&quot;</span>
    else if<span class="hl opt">(</span> mode <span class="hl opt">==</span> <span class="hl str">&#39;filter&#39;</span> <span class="hl opt">)</span>
        <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">       vcftools --gzvcf</span> <span class="hl ipl">${vcf}</span> <span class="hl str">\</span>
<span class="hl str">           --not-chr</span> <span class="hl ipl">${excluding}</span> <span class="hl str">\</span>
<span class="hl str">           --recode \</span>
<span class="hl str">           --stdout | bgzip &gt; tmp.vcf.gz </span>
<span class="hl str">       </span>
<span class="hl str">       truffle \</span>
<span class="hl str">           --vcf tmp.vcf.gz  \</span>
<span class="hl str">           --segments \</span>
<span class="hl str">           --nofiltering \</span>
<span class="hl str">           --ibs1markers</span> <span class="hl ipl">${sz1}</span> <span class="hl str">\</span>
<span class="hl str">           --ibs2markers</span> <span class="hl ipl">${sz2}</span> <span class="hl str">\</span>
<span class="hl str">           --out no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span> <span class="hl str">\</span>
<span class="hl str">           --cpu 8</span>
<span class="hl str">       </span>
<span class="hl str">       sed &#39;s/^</span><span class="hl esc">\\</span><span class="hl str">s*//g; s/</span><span class="hl esc">\\</span><span class="hl str">s</span><span class="hl esc">\\</span><span class="hl str">+/</span><span class="hl esc">\\</span><span class="hl str">t/g&#39; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.ibd &gt; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.ibd.tsv</span>
<span class="hl str">       sed &#39;s/^</span><span class="hl esc">\\</span><span class="hl str">s*//g; s/</span><span class="hl esc">\\</span><span class="hl str">s</span><span class="hl esc">\\</span><span class="hl str">+/</span><span class="hl esc">\\</span><span class="hl str">t/g&#39; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.segments &gt; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.segments.tsv</span>
<span class="hl str">       rm tmp.vcf.gz </span>
<span class="hl str">       &quot;&quot;&quot;</span>
    else if<span class="hl opt">(</span> mode <span class="hl opt">==</span> <span class="hl str">&#39;bed&#39;</span> <span class="hl opt">)</span>
        <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">       vcftools --gzvcf</span> <span class="hl ipl">${vcf}</span> <span class="hl str">\</span>
<span class="hl str">           --exclude-bed ../../../../../ressources/plugin/idb_above_</span><span class="hl ipl">${excluding}</span><span class="hl str">.bed \</span>
<span class="hl str">           --recode \</span>
<span class="hl str">           --stdout | bgzip &gt; tmp.vcf.gz </span>
<span class="hl str">       </span>
<span class="hl str">       truffle \</span>
<span class="hl str">           --vcf tmp.vcf.gz  \</span>
<span class="hl str">           --segments \</span>
<span class="hl str">           --nofiltering \</span>
<span class="hl str">           --ibs1markers</span> <span class="hl ipl">${sz1}</span> <span class="hl str">\</span>
<span class="hl str">           --ibs2markers</span> <span class="hl ipl">${sz2}</span> <span class="hl str">\</span>
<span class="hl str">           --out no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span> <span class="hl str">\</span>
<span class="hl str">           --cpu 8</span>
<span class="hl str">       </span>
<span class="hl str">       sed &#39;s/^</span><span class="hl esc">\\</span><span class="hl str">s*//g; s/</span><span class="hl esc">\\</span><span class="hl str">s</span><span class="hl esc">\\</span><span class="hl str">+/</span><span class="hl esc">\\</span><span class="hl str">t/g&#39; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.ibd &gt; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.ibd.tsv</span>
<span class="hl str">       sed &#39;s/^</span><span class="hl esc">\\</span><span class="hl str">s*//g; s/</span><span class="hl esc">\\</span><span class="hl str">s</span><span class="hl esc">\\</span><span class="hl str">+/</span><span class="hl esc">\\</span><span class="hl str">t/g&#39; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.segments &gt; no_outgr_</span><span class="hl ipl">${mode}${excluding}</span><span class="hl str">_</span><span class="hl ipl">${sz3}</span><span class="hl str">.segments.tsv</span>
<span class="hl str">       rm tmp.vcf.gz </span>
<span class="hl str">       &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Within R, the genomic coordinates of the IBD segments are converted to cM positions.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 16.6</span>
<span class="hl slc">// convert IBD segments to cM</span>
<span class="hl kwa">process</span> convert_to_cM <span class="hl opt">{</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../2_analysis/ibd/cM_converted&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> sz3 <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> truffle_summary <span class="hl opt">) ,</span> <span class="hl kwc">file</span><span class="hl opt">(</span> truffle_segments <span class="hl opt">)</span> <span class="hl kwa">from</span> truffle_result

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*.converted.tsv&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*.conv_summary.tsv&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*.conv_filterd.tsv&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> cM_result

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   #!/usr/bin/env Rscript</span>
<span class="hl str">   base_dir &lt;- Sys.getenv(&quot;BASE_DIR&quot;)</span>
<span class="hl str"></span>
<span class="hl str">   args &lt;- c(&quot;ressources/recombination/&quot;,</span>
<span class="hl str">          &quot;MAP1cm.txt&quot;, &quot;MAP1bp.txt&quot;,</span>
<span class="hl str">          &quot;MAP2cm.txt&quot;, &quot;MAP2bp.txt&quot;,</span>
<span class="hl str">          &quot;</span><span class="hl ipl">${truffle_segments}</span><span class="hl str">&quot;,</span>
<span class="hl str">          &quot;</span><span class="hl ipl">${truffle_summary}</span><span class="hl str">&quot;)</span>
<span class="hl str">   </span>
<span class="hl str">   renv::activate(base_dir)</span>
<span class="hl str"></span>
<span class="hl str">   library(GenomicOriginsScripts)</span>
<span class="hl str">   library(hypogen)</span>
<span class="hl str">   library(patchwork)</span>
<span class="hl str">   library(plyranges)</span>
<span class="hl str"></span>
<span class="hl str">   rec_path &lt;- str_c(base_dir, as.character(args[1]))</span>
<span class="hl str">   hypo_map1_cm &lt;- as.character(args[2])</span>
<span class="hl str">   hypo_map1_bp &lt;- as.character(args[3])</span>
<span class="hl str">   hypo_map2_cm &lt;- as.character(args[4])</span>
<span class="hl str">   hypo_map2_bp &lt;- as.character(args[5])</span>
<span class="hl str">   segment_file &lt;- as.character(args[6])</span>
<span class="hl str">   summary_file &lt;- as.character(args[7])</span>
<span class="hl str"></span>
<span class="hl str">   truffle_conv &lt;- segment_file %&gt;% str_replace(pattern = &quot;.segments.tsv&quot;, replacement = &quot;.converted.tsv&quot;) %&gt;% str_remove(&quot;.*/&quot;)</span>
<span class="hl str">   truffle_sum &lt;- segment_file %&gt;% str_replace(pattern = &quot;.segments.tsv&quot;, replacement = &quot;.conv_summary.tsv&quot;) %&gt;% str_remove(&quot;.*/&quot;)</span>
<span class="hl str">   truffle_filt &lt;- segment_file %&gt;% str_replace(pattern = &quot;.segments.tsv&quot;, replacement = &quot;.conv_filterd.tsv&quot;) %&gt;% str_remove(&quot;.*/&quot;)</span>
<span class="hl str"></span>
<span class="hl str">   read_maps &lt;- function(cm_file, bp_file){</span>
<span class="hl str">   read_tsv(cm_file) %&gt;% </span>
<span class="hl str">       group_by(LG) %&gt;% </span>
<span class="hl str">       mutate(LGnm = as.roman(LG) %&gt;% as.numeric(),</span>
<span class="hl str">           CHROM = str_c(&quot;LG&quot;, str_pad(LGnm, width = 2, pad = 0)),</span>
<span class="hl str">           cM = ifelse(LG == &quot;VIII&quot;, max(cM)-cM,cM)) %&gt;% </span>
<span class="hl str">       ungroup() %&gt;% </span>
<span class="hl str">       dplyr::select(Loci, cM, CHROM)  %&gt;% </span>
<span class="hl str">       full_join(read_tsv(bp_file) %&gt;% </span>
<span class="hl str">                   filter(!(duplicated(Loci) | duplicated(Loci, fromlast = TRUE))),</span>
<span class="hl str">               by = c(Loci = &quot;Loci&quot;, CHROM = &quot;LG&quot;)) %&gt;% </span>
<span class="hl str">       left_join(hypo_chrom_start) %&gt;% </span>
<span class="hl str">       mutate(GPOS = GSTART + bp) %&gt;% </span>
<span class="hl str">       filter(!is.na(GPOS),</span>
<span class="hl str">           !is.na(cM)) %&gt;% </span>
<span class="hl str">       arrange(GPOS)</span>
<span class="hl str">   }</span>
<span class="hl str"></span>
<span class="hl str">   make_lg_seg &lt;- function(lg = &quot;LG08&quot;, n = 31, gmap = gmap1){</span>
<span class="hl str">   data_pos &lt;- tibble(CHROM = rep(lg, n),</span>
<span class="hl str">                       start = seq(from = hypo_karyotype\$GSTART[hypo_karyotype\$CHROM == lg],</span>
<span class="hl str">                                   to = hypo_karyotype\$GEND[hypo_karyotype\$CHROM == lg],</span>
<span class="hl str">                                   length = n) %&gt;%</span>
<span class="hl str">                       floor(),</span>
<span class="hl str">                       GSTART = hypo_karyotype\$GSTART[hypo_karyotype\$CHROM == lg]) %&gt;% </span>
<span class="hl str">       mutate(GPOS = start,</span>
<span class="hl str">           start = start - GSTART,</span>
<span class="hl str">           end = start) %&gt;% </span>
<span class="hl str">       as_iranges()</span>
<span class="hl str">   </span>
<span class="hl str">   map_pos &lt;- gmap %&gt;%</span>
<span class="hl str">       filter(CHROM == lg) %&gt;% </span>
<span class="hl str">       attach_end(LG = lg) %&gt;% </span>
<span class="hl str">       mutate(start = lag(bp, default = 0),</span>
<span class="hl str">           start_cM = lag(cM, default = 0)) %&gt;% </span>
<span class="hl str">       dplyr::select(CHROM, start, end = bp, start_cM, end_cM = cM) %&gt;%</span>
<span class="hl str">       mutate(start_bp = start, end_bp = end) %&gt;% </span>
<span class="hl str">       as_iranges()</span>
<span class="hl str">   </span>
<span class="hl str">   list(data = data_pos, map = map_pos)</span>
<span class="hl str">   }</span>
<span class="hl str"></span>
<span class="hl str">   attach_end &lt;- function(data, LG = &quot;LG01&quot;){</span>
<span class="hl str">   data %&gt;% </span>
<span class="hl str">       bind_rows(., </span>
<span class="hl str">               data %&gt;%</span>
<span class="hl str">                   filter(row_number() == last(row_number())) %&gt;% </span>
<span class="hl str">                   mutate(GPOS = hypo_karyotype\$GEND[hypo_karyotype\$CHROM == CHROM],</span>
<span class="hl str">                       bp = hypo_karyotype\$LENGTH[hypo_karyotype\$CHROM == CHROM],</span>
<span class="hl str">                       Loci = as.numeric(</span>
<span class="hl str">                           str_c(&quot;-99&quot;,</span>
<span class="hl str">                               str_remove(string = CHROM, &quot;LG&quot;))</span>
<span class="hl str">                       )</span>
<span class="hl str">                   ))</span>
<span class="hl str">   }</span>
<span class="hl str"></span>
<span class="hl str">   bin_rescaler &lt;- function(bp, start_cM, end_cM, start_bp, end_bp,...){</span>
<span class="hl str">   scales::rescale(x = bp,</span>
<span class="hl str">                   to = c(start_cM, end_cM),</span>
<span class="hl str">                   from = c(start_bp, end_bp))</span>
<span class="hl str">   }</span>
<span class="hl str"></span>
<span class="hl str">   interpol_data &lt;- function(lg, ...){</span>
<span class="hl str">   data_pair &lt;- make_lg_seg(lg = lg, ...)</span>
<span class="hl str">   </span>
<span class="hl str">   plyranges::join_overlap_inner(data_pair\$data,</span>
<span class="hl str">                                   data_pair\$map) %&gt;% </span>
<span class="hl str">       as.data.frame() %&gt;% </span>
<span class="hl str">       as_tibble() %&gt;% </span>
<span class="hl str">       dplyr::select(CHROM = CHROM.x, bp = start, GSTART, GPOS, start_cM:end_bp) %&gt;% </span>
<span class="hl str">       mutate(interpol_cM = pmap_dbl(cur_data(), bin_rescaler)) </span>
<span class="hl str">   }</span>
<span class="hl str"></span>
<span class="hl str">   na_to_zero &lt;- function(x){</span>
<span class="hl str">   x_type &lt;- typeof(x)</span>
<span class="hl str">   if_else(is.na(x), as(0,Class = x_type), x) %&gt;% </span>
<span class="hl str">       as.double() %&gt;% as(Class = x_type)</span>
<span class="hl str">   }</span>
<span class="hl str"></span>
<span class="hl str">   convert_bp_to_cm &lt;- function(data, lg = &quot;LG08&quot;, gmap = gmap1){</span>
<span class="hl str">   gmap_in &lt;- deparse(substitute(gmap))</span>
<span class="hl str">   </span>
<span class="hl str">   data_pos &lt;- data %&gt;% </span>
<span class="hl str">       filter( CHROM == lg ) %&gt;% </span>
<span class="hl str">       dplyr::select(PAIR, TYPE, CHROM, START, END, NMARKERS) %&gt;% </span>
<span class="hl str">       mutate(seg_id = str_c(PAIR,&quot;_&quot;,CHROM,&quot;_&quot;,START)) %&gt;% </span>
<span class="hl str">       pivot_longer(cols = START:END, names_to = &quot;PART&quot;, values_to = &quot;start&quot;) %&gt;%</span>
<span class="hl str">       mutate(end = start) %&gt;% </span>
<span class="hl str">       as_iranges()</span>
<span class="hl str">   </span>
<span class="hl str">   map_pos &lt;- gmap %&gt;%</span>
<span class="hl str">       filter(CHROM == lg) %&gt;%</span>
<span class="hl str">       attach_end(LG = lg) %&gt;%</span>
<span class="hl str">       mutate(start = lag(bp, default = 0) + 1, # avoid overlapping segments - causes duplications in joining</span>
<span class="hl str">           start_cM = lag(cM, default = 0)) %&gt;%</span>
<span class="hl str">       dplyr::select(start, end = bp, start_cM, end_cM = cM) %&gt;%</span>
<span class="hl str">       mutate(start_bp = start, end_bp = end) %&gt;%</span>
<span class="hl str">       as_iranges()</span>
<span class="hl str">   </span>
<span class="hl str">   map_nr &lt;- str_replace(gmap_in, pattern = &quot;gmap&quot;, replacement = &quot;_m&quot;)</span>
<span class="hl str">   </span>
<span class="hl str">   plyranges::join_overlap_inner(data_pos,</span>
<span class="hl str">                                   map_pos) %&gt;%</span>
<span class="hl str">       as.data.frame() %&gt;%</span>
<span class="hl str">       as_tibble() %&gt;%</span>
<span class="hl str">       dplyr::select(CHROM = CHROM, bp = start, PAIR, TYPE, PART, start_cM:end_bp, seg_id, NMARKERS) %&gt;%</span>
<span class="hl str">       mutate(interpol_cM = pmap_dbl(cur_data(), bin_rescaler)) %&gt;%</span>
<span class="hl str">       pivot_wider(id_cols = c(CHROM,PAIR,TYPE,seg_id, PART,NMARKERS),</span>
<span class="hl str">                   values_from = c(bp,interpol_cM),</span>
<span class="hl str">                   names_from = PART) %&gt;%</span>
<span class="hl str">       dplyr::select(-seg_id) %&gt;%</span>
<span class="hl str">       mutate(length_bp = bp_END - bp_START,</span>
<span class="hl str">           length_cM = interpol_cM_END - interpol_cM_START) %&gt;% </span>
<span class="hl str">       set_names(value = c(&quot;CHROM&quot;, &quot;PAIR&quot;, &quot;TYPE&quot;, &quot;NMARKERS&quot;, &quot;bp_START&quot;, &quot;bp_END&quot;,</span>
<span class="hl str">                           str_c(c(&quot;interpol_cM_START&quot;, &quot;interpol_cM_END&quot;), map_nr),</span>
<span class="hl str">                           &quot;length_bp&quot;, str_c(&quot;length_cM&quot;, map_nr)))</span>
<span class="hl str">   }</span>
<span class="hl str"></span>
<span class="hl str">   # actual script -------------------</span>
<span class="hl str">   gmap1 &lt;- read_maps(cm_file = str_c(rec_path, hypo_map1_cm),</span>
<span class="hl str">                   bp_file = str_c(rec_path, hypo_map1_bp))</span>
<span class="hl str">   gmap2 &lt;- read_maps(cm_file = str_c(rec_path, hypo_map2_cm),</span>
<span class="hl str">                   bp_file = str_c(rec_path, hypo_map2_bp))</span>
<span class="hl str"></span>
<span class="hl str"></span>
<span class="hl str">   lgs &lt;- 1:24 %&gt;%</span>
<span class="hl str">   str_pad(width = 2, pad = 0) %&gt;%</span>
<span class="hl str">   str_c(&quot;LG&quot;,.)</span>
<span class="hl str"></span>
<span class="hl str">   segments_individual_interpol_map1 &lt;- lgs %&gt;% map_dfr(interpol_data, n = 51)</span>
<span class="hl str">   segments_individual_interpol_map2 &lt;- lgs %&gt;% map_dfr(interpol_data, n = 51, gmap = gmap2)</span>
<span class="hl str"></span>
<span class="hl str">   segments_individual &lt;- vroom::vroom(segment_file) %&gt;%</span>
<span class="hl str">   mutate(LENGTH = LENGTH * 10^6,</span>
<span class="hl str">           START = POS * 10^6,</span>
<span class="hl str">           END = START + LENGTH,</span>
<span class="hl str">           PAIR = str_c(ID1, &quot;-&quot;, ID2))</span>
<span class="hl str"></span>
<span class="hl str">   segments_summary &lt;- vroom::vroom(summary_file) %&gt;%</span>
<span class="hl str">   mutate(PAIR = str_c(ID1, &quot;-&quot;, ID2))</span>
<span class="hl str"></span>
<span class="hl str">   bounds_gmap1 &lt;- gmap1 %&gt;% </span>
<span class="hl str">   group_by(CHROM) %&gt;% </span>
<span class="hl str">   filter(cM == max(cM)) %&gt;% </span>
<span class="hl str">   filter(bp == max(bp)) %&gt;% </span>
<span class="hl str">   ungroup() %&gt;% </span>
<span class="hl str">   dplyr::select(CHROM, cM) %&gt;%</span>
<span class="hl str">   mutate(GSTART_cM = cumsum(lag(cM, default = 0)),</span>
<span class="hl str">           GEND_cM = cM + GSTART_cM,</span>
<span class="hl str">           GMID_cM = (GSTART_cM + GEND_cM) / 2,</span>
<span class="hl str">           grp = c(&quot;even&quot;, &quot;odd&quot;)[ 1+row_number() %% 2 ])</span>
<span class="hl str"></span>
<span class="hl str">   bounds_gmap2 &lt;- gmap2 %&gt;% </span>
<span class="hl str">   group_by(CHROM) %&gt;% </span>
<span class="hl str">   filter(cM == max(cM)) %&gt;% </span>
<span class="hl str">   filter(bp == max(bp)) %&gt;% </span>
<span class="hl str">   ungroup() %&gt;% </span>
<span class="hl str">   dplyr::select(CHROM, cM) %&gt;%</span>
<span class="hl str">   mutate(GSTART_cM = cumsum(lag(cM, default = 0)),</span>
<span class="hl str">           GEND_cM = cM + GSTART_cM,</span>
<span class="hl str">           GMID_cM = (GSTART_cM + GEND_cM) / 2,</span>
<span class="hl str">           grp = c(&quot;even&quot;, &quot;odd&quot;)[ 1+row_number() %% 2 ])</span>
<span class="hl str"></span>
<span class="hl str">   hypo_all_starts &lt;- hypo_karyotype %&gt;% </span>
<span class="hl str">   dplyr::select(CHROM, GSTART, GEND) %&gt;% </span>
<span class="hl str">   left_join(bounds_gmap1 %&gt;% </span>
<span class="hl str">               dplyr::select(CHROM, GSTART_cM_m1 = GSTART_cM, GEND_cM_m1 = GEND_cM)) %&gt;% </span>
<span class="hl str">   left_join(bounds_gmap2 %&gt;% </span>
<span class="hl str">               dplyr::select(CHROM, GSTART_cM_m2 = GSTART_cM, GEND_cM_m2 = GEND_cM))</span>
<span class="hl str"></span>
<span class="hl str">   converted_segments &lt;- lgs %&gt;% </span>
<span class="hl str">   map_dfr(convert_bp_to_cm, data = segments_individual) %&gt;% </span>
<span class="hl str">   left_join( lgs %&gt;% </span>
<span class="hl str">               map_dfr(convert_bp_to_cm, data = segments_individual, gmap = gmap2) ) %&gt;%</span>
<span class="hl str">   left_join(hypo_all_starts) %&gt;% </span>
<span class="hl str">   mutate(G_SEG_START = GSTART + bp_START,</span>
<span class="hl str">           G_SEG_END = GSTART + bp_END,</span>
<span class="hl str">           G_SEG_START_cM_m1 = GSTART_cM_m1 + interpol_cM_START_m1,</span>
<span class="hl str">           G_SEG_END_cM_m1 = GSTART_cM_m1 + interpol_cM_END_m1,</span>
<span class="hl str">           G_SEG_START_cM_m2 = GSTART_cM_m2 + interpol_cM_START_m2,</span>
<span class="hl str">           G_SEG_END_cM_m2 = GSTART_cM_m2 + interpol_cM_END_m2)</span>
<span class="hl str"></span>
<span class="hl str">   hypo_cM_length_map1 &lt;- max(hypo_all_starts\$GEND_cM_m1)</span>
<span class="hl str">   hypo_cM_length_map2 &lt;- max(hypo_all_starts\$GEND_cM_m2)</span>
<span class="hl str">   hypo_bp_length &lt;- hypo_karyotype\$GEND[hypo_karyotype\$CHROM == &quot;LG24&quot;]</span>
<span class="hl str"></span>
<span class="hl str">   control &lt;- converted_segments %&gt;%</span>
<span class="hl str">   ungroup() %&gt;%</span>
<span class="hl str">   group_by(PAIR, TYPE) %&gt;%</span>
<span class="hl str">   summarise(seq_length = sum(length_bp),</span>
<span class="hl str">               n_mark = sum(NMARKERS),</span>
<span class="hl str">               cm_length_m1 = sum(length_cM_m1),</span>
<span class="hl str">               cm_length_m2 = sum(length_cM_m2)) %&gt;%</span>
<span class="hl str">   ungroup() %&gt;%</span>
<span class="hl str">   pivot_wider(id_cols = PAIR, names_from = TYPE, values_from = seq_length:cm_length_m2, values_fill = 0) %&gt;% </span>
<span class="hl str">   left_join(segments_summary, .,  ) %&gt;% </span>
<span class="hl str">   mutate(across(.cols = seq_length_IBD1:cm_length_m2_IBD2, .fns = na_to_zero)) %&gt;% </span>
<span class="hl str">   mutate(IBD0_manual = (NMARK - (n_mark_IBD1 + n_mark_IBD2)) / NMARK,</span>
<span class="hl str">           IBD1_manual = n_mark_IBD1 / NMARK,</span>
<span class="hl str">           IBD2_manual = n_mark_IBD2 / NMARK,</span>
<span class="hl str">           icheck_0 = IBD0_manual - IBD0,</span>
<span class="hl str">           icheck_1 = IBD1_manual - IBD1,</span>
<span class="hl str">           icheck_2 = IBD2 - IBD2,</span>
<span class="hl str">           # compile ibd by sequence map</span>
<span class="hl str">           ibd0_bp = (hypo_bp_length - (seq_length_IBD1 + seq_length_IBD2)) / hypo_bp_length,</span>
<span class="hl str">           ibd1_bp = seq_length_IBD1 / hypo_bp_length,</span>
<span class="hl str">           ibd2_bp = seq_length_IBD2 / hypo_bp_length,</span>
<span class="hl str">           # compile ibd by genetic map 1</span>
<span class="hl str">           ibd0_cM_m1 = (hypo_cM_length_map1 - (cm_length_m1_IBD1 + cm_length_m1_IBD2)) / hypo_cM_length_map1,</span>
<span class="hl str">           ibd1_cM_m1 = cm_length_m1_IBD1 / hypo_cM_length_map1,</span>
<span class="hl str">           ibd2_cM_m1 = cm_length_m1_IBD2 / hypo_cM_length_map1,</span>
<span class="hl str">           # compile ibd by genetic map 2</span>
<span class="hl str">           ibd0_cM_m2 = (hypo_cM_length_map2 - (cm_length_m2_IBD1 + cm_length_m2_IBD2)) / hypo_cM_length_map2,</span>
<span class="hl str">           ibd1_cM_m2 = cm_length_m2_IBD1 / hypo_cM_length_map2,</span>
<span class="hl str">           ibd2_cM_m2 = cm_length_m2_IBD2 / hypo_cM_length_map2)</span>
<span class="hl str"></span>
<span class="hl str">   cM_treshold &lt;- 0.2</span>
<span class="hl str">   summary_filterd &lt;- converted_segments %&gt;%</span>
<span class="hl str">   filter(length_cM_m1 &gt; cM_treshold &amp; length_cM_m2 &gt; cM_treshold) %&gt;% </span>
<span class="hl str">   ungroup() %&gt;%</span>
<span class="hl str">   group_by(PAIR, TYPE) %&gt;%</span>
<span class="hl str">   summarise(seq_length = sum(length_bp),</span>
<span class="hl str">               n_mark = sum(NMARKERS),</span>
<span class="hl str">               cm_length_m1 = sum(length_cM_m1),</span>
<span class="hl str">               cm_length_m2 = sum(length_cM_m2)) %&gt;%</span>
<span class="hl str">   ungroup() %&gt;%</span>
<span class="hl str">   pivot_wider(id_cols = PAIR, names_from = TYPE, values_from = seq_length:cm_length_m2, values_fill = 0) %&gt;% </span>
<span class="hl str">   left_join(segments_summary, .,  ) %&gt;% </span>
<span class="hl str">   mutate(across(.cols = seq_length_IBD1:cm_length_m2_IBD2, .fns = na_to_zero)) %&gt;% </span>
<span class="hl str">   mutate(IBD0_manual = (NMARK - (n_mark_IBD1 + n_mark_IBD2)) / NMARK,</span>
<span class="hl str">           IBD1_manual = n_mark_IBD1 / NMARK,</span>
<span class="hl str">           IBD2_manual = n_mark_IBD2 / NMARK,</span>
<span class="hl str">           icheck_0 = IBD0_manual - IBD0,</span>
<span class="hl str">           icheck_1 = IBD1_manual - IBD1,</span>
<span class="hl str">           icheck_2 = IBD2 - IBD2_manual,</span>
<span class="hl str">           # compile ibd by sequence map</span>
<span class="hl str">           ibd0_bp = (hypo_bp_length - (seq_length_IBD1 + seq_length_IBD2)) / hypo_bp_length,</span>
<span class="hl str">           ibd1_bp = seq_length_IBD1 / hypo_bp_length,</span>
<span class="hl str">           ibd2_bp = seq_length_IBD2 / hypo_bp_length,</span>
<span class="hl str">           # compile ibd by genetic map 1</span>
<span class="hl str">           ibd0_cM_m1 = (hypo_cM_length_map1 - (cm_length_m1_IBD1 + cm_length_m1_IBD2)) / hypo_cM_length_map1,</span>
<span class="hl str">           ibd1_cM_m1 = cm_length_m1_IBD1 / hypo_cM_length_map1,</span>
<span class="hl str">           ibd2_cM_m1 = cm_length_m1_IBD2 / hypo_cM_length_map1,</span>
<span class="hl str">           # compile ibd by genetic map 2</span>
<span class="hl str">           ibd0_cM_m2 = (hypo_cM_length_map2 - (cm_length_m2_IBD1 + cm_length_m2_IBD2)) / hypo_cM_length_map2,</span>
<span class="hl str">           ibd1_cM_m2 = cm_length_m2_IBD1 / hypo_cM_length_map2,</span>
<span class="hl str">           ibd2_cM_m2 = cm_length_m2_IBD2 / hypo_cM_length_map2)</span>
<span class="hl str"></span>
<span class="hl str">   write_tsv(x = converted_segments, file = truffle_conv)</span>
<span class="hl str">   write_tsv(x = control, file = truffle_sum)</span>
<span class="hl str">   write_tsv(x = summary_filterd, file = truffle_filt)</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
</div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="git-17-analysis-xv-dstats.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
