<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>14 (git 13) Analysis XI (Whole Genome Phylogenies) | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Rapid radiation in a highly diverse marine environment’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="14 (git 13) Analysis XI (Whole Genome Phylogenies) | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Rapid radiation in a highly diverse marine environment’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="14 (git 13) Analysis XI (Whole Genome Phylogenies) | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Rapid radiation in a highly diverse marine environment’ by Hench, Helmkampf, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench and Martin Helmkampf" />


<meta name="date" content="2022-01-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="git-12-analysis-x-fst-permutation-test.html"/>
<link rel="next" href="git-14-analysis-xii-outlier-region-phylogenies.html"/>
<script src="libs/header-attrs-2.7.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#r-setup"><i class="fa fa-check"></i><b>1.4</b> R setup</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a>
<ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a>
<ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy.html"><a href="git-4-analysis-ii-dxy.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i>)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy.html"><a href="git-4-analysis-ii-dxy.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy.html"><a href="git-4-analysis-ii-dxy.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy.html"><a href="git-4-analysis-ii-dxy.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (topology weighting)</a>
<ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (Principal Component Analysis)</a>
<ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#details-of-analysis_pca.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_pca.nf</code></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#setup-2"><i class="fa fa-check"></i><b>8.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-4"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-5"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-6"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html"><i class="fa fa-check"></i><b>12</b> (git 11) Analysis IX (Allele Age)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#summary-10"><i class="fa fa-check"></i><b>12.1</b> Summary</a></li>
<li class="chapter" data-level="12.2" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#details-of-analysis_allele_age.nf"><i class="fa fa-check"></i><b>12.2</b> Details of <code>analysis_allele_age.nf</code></a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#setup-3"><i class="fa fa-check"></i><b>12.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html"><i class="fa fa-check"></i><b>13</b> (git 12) Analysis X (<i>F<sub>ST</sub></i> Permutation Test)</a>
<ul>
<li class="chapter" data-level="13.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#summary-11"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#details-of-analysis_fst_sign.nf"><i class="fa fa-check"></i><b>13.2</b> Details of <code>analysis_fst_sign.nf</code></a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#setup-4"><i class="fa fa-check"></i><b>13.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html"><i class="fa fa-check"></i><b>14</b> (git 13) Analysis XI (Whole Genome Phylogenies)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#summary-12"><i class="fa fa-check"></i><b>14.1</b> Summary</a></li>
<li class="chapter" data-level="14.2" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#details-of-analysis_phylo_whg.nf"><i class="fa fa-check"></i><b>14.2</b> Details of <code>analysis_phylo_whg.nf</code></a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#setup-5"><i class="fa fa-check"></i><b>14.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html"><i class="fa fa-check"></i><b>15</b> (git 14) Analysis XII (Outlier Region Phylogenies)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#summary-13"><i class="fa fa-check"></i><b>15.1</b> Summary</a></li>
<li class="chapter" data-level="15.2" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#details-of-analysis_phylo_regions.nf"><i class="fa fa-check"></i><b>15.2</b> Details of <code>analysis_phylo_regions.nf</code></a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#setup-6"><i class="fa fa-check"></i><b>15.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"><a href="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"><i class="fa fa-check"></i><b>16</b> (git 15) Analysis XIII (diversity with and without outlier regions)</a>
<ul>
<li class="chapter" data-level="16.1" data-path="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"><a href="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html#summary-14"><i class="fa fa-check"></i><b>16.1</b> Summary</a></li>
<li class="chapter" data-level="16.2" data-path="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"><a href="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html#details-of-analysis_pi.nf"><i class="fa fa-check"></i><b>16.2</b> Details of <code>analysis_pi.nf</code></a>
<ul>
<li class="chapter" data-level="16.2.1" data-path="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html"><a href="git-15-analysis-xiii-diversity-with-and-without-outlier-regions.html#setup-7"><i class="fa fa-check"></i><b>16.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="git-16-analysis-xiv-identity-by-descent.html"><a href="git-16-analysis-xiv-identity-by-descent.html"><i class="fa fa-check"></i><b>17</b> (git 16) Analysis XIV (Identity by Descent)</a>
<ul>
<li class="chapter" data-level="17.1" data-path="git-16-analysis-xiv-identity-by-descent.html"><a href="git-16-analysis-xiv-identity-by-descent.html#summary-15"><i class="fa fa-check"></i><b>17.1</b> Summary</a></li>
<li class="chapter" data-level="17.2" data-path="git-16-analysis-xiv-identity-by-descent.html"><a href="git-16-analysis-xiv-identity-by-descent.html#details-of-analysis_ibd.nf"><i class="fa fa-check"></i><b>17.2</b> Details of <code>analysis_ibd.nf</code></a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="git-16-analysis-xiv-identity-by-descent.html"><a href="git-16-analysis-xiv-identity-by-descent.html#setup-8"><i class="fa fa-check"></i><b>17.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="git-17-analysis-xv-dstats.html"><a href="git-17-analysis-xv-dstats.html"><i class="fa fa-check"></i><b>18</b> (git 17) Analysis XV (dstats)</a>
<ul>
<li class="chapter" data-level="18.1" data-path="git-17-analysis-xv-dstats.html"><a href="git-17-analysis-xv-dstats.html#summary-16"><i class="fa fa-check"></i><b>18.1</b> Summary</a></li>
<li class="chapter" data-level="18.2" data-path="git-17-analysis-xv-dstats.html"><a href="git-17-analysis-xv-dstats.html#details-of-analysis_dstats.nf"><i class="fa fa-check"></i><b>18.2</b> Details of <code>analysis_dstats.nf</code></a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="git-17-analysis-xv-dstats.html"><a href="git-17-analysis-xv-dstats.html#setup-9"><i class="fa fa-check"></i><b>18.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html"><a href="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html"><i class="fa fa-check"></i><b>19</b> (git 18) Genotyping III (all callable sites for mtDNA and unplaced contigs)</a>
<ul>
<li class="chapter" data-level="19.1" data-path="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html"><a href="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html#summary-17"><i class="fa fa-check"></i><b>19.1</b> Summary</a></li>
<li class="chapter" data-level="19.2" data-path="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html"><a href="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html#details-of-genotyping_all_basepairs_mt.nf"><i class="fa fa-check"></i><b>19.2</b> Details of <code>genotyping_all_basepairs_mt.nf</code></a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html"><a href="git-18-genotyping-iii-all-callable-sites-for-mtdna-and-unplaced-contigs.html#data-preparation-7"><i class="fa fa-check"></i><b>19.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="git-19-analysis-xvi-serraninae-phylogeny.html"><a href="git-19-analysis-xvi-serraninae-phylogeny.html"><i class="fa fa-check"></i><b>20</b> (git 19) Analysis XVI (Serraninae Phylogeny)</a>
<ul>
<li class="chapter" data-level="20.1" data-path="git-19-analysis-xvi-serraninae-phylogeny.html"><a href="git-19-analysis-xvi-serraninae-phylogeny.html#summary-18"><i class="fa fa-check"></i><b>20.1</b> Summary</a></li>
<li class="chapter" data-level="20.2" data-path="git-19-analysis-xvi-serraninae-phylogeny.html"><a href="git-19-analysis-xvi-serraninae-phylogeny.html#details-of-analysis_phylo_serraninae.nf"><i class="fa fa-check"></i><b>20.2</b> Details of <code>analysis_phylo_serraninae.nf</code></a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="git-19-analysis-xvi-serraninae-phylogeny.html"><a href="git-19-analysis-xvi-serraninae-phylogeny.html#setup-10"><i class="fa fa-check"></i><b>20.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="git-20-data-visualization.html"><a href="git-20-data-visualization.html"><i class="fa fa-check"></i><b>21</b> (git 20) Data Visualization</a></li>
<li class="chapter" data-level="22" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>22</b> Figure 1</a>
<ul>
<li class="chapter" data-level="22.1" data-path="figure-1.html"><a href="figure-1.html#summary-19"><i class="fa fa-check"></i><b>22.1</b> Summary</a></li>
<li class="chapter" data-level="22.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>22.2</b> Details of <code>plot_F1.R</code></a>
<ul>
<li class="chapter" data-level="22.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>22.2.1</b> Config</a></li>
<li class="chapter" data-level="22.2.2" data-path="figure-1.html"><a href="figure-1.html#actual-script-start"><i class="fa fa-check"></i><b>22.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>23</b> Figure 2</a>
<ul>
<li class="chapter" data-level="23.1" data-path="figure-2.html"><a href="figure-2.html#summary-20"><i class="fa fa-check"></i><b>23.1</b> Summary</a></li>
<li class="chapter" data-level="23.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>23.2</b> Details of <code>plot_F2.R</code></a>
<ul>
<li class="chapter" data-level="23.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>23.2.1</b> Config</a></li>
<li class="chapter" data-level="23.2.2" data-path="figure-2.html"><a href="figure-2.html#actual-script-start-1"><i class="fa fa-check"></i><b>23.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>24</b> Figure 3</a>
<ul>
<li class="chapter" data-level="24.1" data-path="figure-3.html"><a href="figure-3.html#summary-21"><i class="fa fa-check"></i><b>24.1</b> Summary</a></li>
<li class="chapter" data-level="24.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>24.2</b> Details of <code>plot_F3.R</code></a>
<ul>
<li class="chapter" data-level="24.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>24.2.1</b> Config</a></li>
<li class="chapter" data-level="24.2.2" data-path="figure-3.html"><a href="figure-3.html#actual-script-start-2"><i class="fa fa-check"></i><b>24.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>25</b> Figure 4</a>
<ul>
<li class="chapter" data-level="25.1" data-path="figure-4.html"><a href="figure-4.html#summary-22"><i class="fa fa-check"></i><b>25.1</b> Summary</a></li>
<li class="chapter" data-level="25.2" data-path="figure-4.html"><a href="figure-4.html#details-of-plot_f4.r"><i class="fa fa-check"></i><b>25.2</b> Details of <code>plot_F4.R</code></a>
<ul>
<li class="chapter" data-level="25.2.1" data-path="figure-4.html"><a href="figure-4.html#config-3"><i class="fa fa-check"></i><b>25.2.1</b> Config</a></li>
<li class="chapter" data-level="25.2.2" data-path="figure-4.html"><a href="figure-4.html#actual-script-start-3"><i class="fa fa-check"></i><b>25.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>26</b> Figure 5</a>
<ul>
<li class="chapter" data-level="26.1" data-path="figure-5.html"><a href="figure-5.html#summary-23"><i class="fa fa-check"></i><b>26.1</b> Summary</a></li>
<li class="chapter" data-level="26.2" data-path="figure-5.html"><a href="figure-5.html#details-of-plot_f5.r"><i class="fa fa-check"></i><b>26.2</b> Details of <code>plot_F5.R</code></a>
<ul>
<li class="chapter" data-level="26.2.1" data-path="figure-5.html"><a href="figure-5.html#config-4"><i class="fa fa-check"></i><b>26.2.1</b> Config</a></li>
<li class="chapter" data-level="26.2.2" data-path="figure-5.html"><a href="figure-5.html#actual-script-start-4"><i class="fa fa-check"></i><b>26.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>27</b> Figure 6</a>
<ul>
<li class="chapter" data-level="27.1" data-path="figure-6.html"><a href="figure-6.html#summary-24"><i class="fa fa-check"></i><b>27.1</b> Summary</a></li>
<li class="chapter" data-level="27.2" data-path="figure-6.html"><a href="figure-6.html#details-of-plot_f6.r"><i class="fa fa-check"></i><b>27.2</b> Details of <code>plot_F6.R</code></a>
<ul>
<li class="chapter" data-level="27.2.1" data-path="figure-6.html"><a href="figure-6.html#config-5"><i class="fa fa-check"></i><b>27.2.1</b> Config</a></li>
<li class="chapter" data-level="27.2.2" data-path="figure-6.html"><a href="figure-6.html#actual-script-start-5"><i class="fa fa-check"></i><b>27.2.2</b> Actual Script Start</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html"><i class="fa fa-check"></i><b>28</b> Supplementary Figure 1</a>
<ul>
<li class="chapter" data-level="28.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#summary-25"><i class="fa fa-check"></i><b>28.1</b> Summary</a></li>
<li class="chapter" data-level="28.2" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#details-of-plot_sf1.r"><i class="fa fa-check"></i><b>28.2</b> Details of <code>plot_SF1.R</code></a>
<ul>
<li class="chapter" data-level="28.2.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#config-6"><i class="fa fa-check"></i><b>28.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="29" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html"><i class="fa fa-check"></i><b>29</b> Supplementary Figure 2</a>
<ul>
<li class="chapter" data-level="29.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#summary-26"><i class="fa fa-check"></i><b>29.1</b> Summary</a></li>
<li class="chapter" data-level="29.2" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#details-of-plot_sf2.r"><i class="fa fa-check"></i><b>29.2</b> Details of <code>plot_SF2.R</code></a>
<ul>
<li class="chapter" data-level="29.2.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#config-7"><i class="fa fa-check"></i><b>29.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="30" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html"><i class="fa fa-check"></i><b>30</b> Supplementary Figure 3</a>
<ul>
<li class="chapter" data-level="30.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#summary-27"><i class="fa fa-check"></i><b>30.1</b> Summary</a></li>
<li class="chapter" data-level="30.2" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#details-of-plot_sf3.r"><i class="fa fa-check"></i><b>30.2</b> Details of <code>plot_SF3.R</code></a>
<ul>
<li class="chapter" data-level="30.2.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#config-8"><i class="fa fa-check"></i><b>30.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="31" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html"><i class="fa fa-check"></i><b>31</b> Supplementary Figure 4</a>
<ul>
<li class="chapter" data-level="31.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#summary-28"><i class="fa fa-check"></i><b>31.1</b> Summary</a></li>
<li class="chapter" data-level="31.2" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#details-of-plot_sf4.r"><i class="fa fa-check"></i><b>31.2</b> Details of <code>plot_SF4.R</code></a>
<ul>
<li class="chapter" data-level="31.2.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#config-9"><i class="fa fa-check"></i><b>31.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="32" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html"><i class="fa fa-check"></i><b>32</b> Supplementary Figure 5</a>
<ul>
<li class="chapter" data-level="32.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#summary-29"><i class="fa fa-check"></i><b>32.1</b> Summary</a></li>
<li class="chapter" data-level="32.2" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#details-of-plot_sf5.r"><i class="fa fa-check"></i><b>32.2</b> Details of <code>plot_SF5.R</code></a>
<ul>
<li class="chapter" data-level="32.2.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#config-10"><i class="fa fa-check"></i><b>32.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html"><i class="fa fa-check"></i><b>33</b> Supplementary Figure 6</a>
<ul>
<li class="chapter" data-level="33.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#summary-30"><i class="fa fa-check"></i><b>33.1</b> Summary</a></li>
<li class="chapter" data-level="33.2" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#details-of-plot_sf6.r"><i class="fa fa-check"></i><b>33.2</b> Details of <code>plot_SF6.R</code></a>
<ul>
<li class="chapter" data-level="33.2.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#config-11"><i class="fa fa-check"></i><b>33.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html"><i class="fa fa-check"></i><b>34</b> Supplementary Figure 7</a>
<ul>
<li class="chapter" data-level="34.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#summary-31"><i class="fa fa-check"></i><b>34.1</b> Summary</a></li>
<li class="chapter" data-level="34.2" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#details-of-plot_sf7.r"><i class="fa fa-check"></i><b>34.2</b> Details of <code>plot_SF7.R</code></a>
<ul>
<li class="chapter" data-level="34.2.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#config-12"><i class="fa fa-check"></i><b>34.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="35" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html"><i class="fa fa-check"></i><b>35</b> Supplementary Figure 8</a>
<ul>
<li class="chapter" data-level="35.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#summary-32"><i class="fa fa-check"></i><b>35.1</b> Summary</a></li>
<li class="chapter" data-level="35.2" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#details-of-plot_sf8.r"><i class="fa fa-check"></i><b>35.2</b> Details of <code>plot_SF8.R</code></a>
<ul>
<li class="chapter" data-level="35.2.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#config-13"><i class="fa fa-check"></i><b>35.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="36" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html"><i class="fa fa-check"></i><b>36</b> Supplementary Figure 9</a>
<ul>
<li class="chapter" data-level="36.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#summary-33"><i class="fa fa-check"></i><b>36.1</b> Summary</a></li>
<li class="chapter" data-level="36.2" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#details-of-plot_sf9.r"><i class="fa fa-check"></i><b>36.2</b> Details of <code>plot_SF9.R</code></a>
<ul>
<li class="chapter" data-level="36.2.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#config-14"><i class="fa fa-check"></i><b>36.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="37" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html"><i class="fa fa-check"></i><b>37</b> Supplementary Figure 10</a>
<ul>
<li class="chapter" data-level="37.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#summary-34"><i class="fa fa-check"></i><b>37.1</b> Summary</a></li>
<li class="chapter" data-level="37.2" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#details-of-plot_sf10.r"><i class="fa fa-check"></i><b>37.2</b> Details of <code>plot_SF10.R</code></a>
<ul>
<li class="chapter" data-level="37.2.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#config-15"><i class="fa fa-check"></i><b>37.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="38" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html"><i class="fa fa-check"></i><b>38</b> Supplementary Figure 11</a>
<ul>
<li class="chapter" data-level="38.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#summary-35"><i class="fa fa-check"></i><b>38.1</b> Summary</a></li>
<li class="chapter" data-level="38.2" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#details-of-plot_sf11.r"><i class="fa fa-check"></i><b>38.2</b> Details of <code>plot_SF11.R</code></a>
<ul>
<li class="chapter" data-level="38.2.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#config-16"><i class="fa fa-check"></i><b>38.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="39" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html"><i class="fa fa-check"></i><b>39</b> Supplementary Figure 12</a>
<ul>
<li class="chapter" data-level="39.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#summary-36"><i class="fa fa-check"></i><b>39.1</b> Summary</a></li>
<li class="chapter" data-level="39.2" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#details-of-plot_sf12.r"><i class="fa fa-check"></i><b>39.2</b> Details of <code>plot_SF12.R</code></a>
<ul>
<li class="chapter" data-level="39.2.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#config-17"><i class="fa fa-check"></i><b>39.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="40" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html"><i class="fa fa-check"></i><b>40</b> Supplementary Figure 13</a>
<ul>
<li class="chapter" data-level="40.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#summary-37"><i class="fa fa-check"></i><b>40.1</b> Summary</a></li>
<li class="chapter" data-level="40.2" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#details-of-plot_sf13.r"><i class="fa fa-check"></i><b>40.2</b> Details of <code>plot_SF13.R</code></a>
<ul>
<li class="chapter" data-level="40.2.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#config-18"><i class="fa fa-check"></i><b>40.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="41" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html"><i class="fa fa-check"></i><b>41</b> Supplementary Figure 14</a>
<ul>
<li class="chapter" data-level="41.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#summary-38"><i class="fa fa-check"></i><b>41.1</b> Summary</a></li>
<li class="chapter" data-level="41.2" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#details-of-plot_sf14.r"><i class="fa fa-check"></i><b>41.2</b> Details of <code>plot_SF14.R</code></a>
<ul>
<li class="chapter" data-level="41.2.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#config-19"><i class="fa fa-check"></i><b>41.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="42" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html"><i class="fa fa-check"></i><b>42</b> Supplementary Figure 15</a>
<ul>
<li class="chapter" data-level="42.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#summary-39"><i class="fa fa-check"></i><b>42.1</b> Summary</a></li>
<li class="chapter" data-level="42.2" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#details-of-plot_sf15.r"><i class="fa fa-check"></i><b>42.2</b> Details of <code>plot_SF15.R</code></a>
<ul>
<li class="chapter" data-level="42.2.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#config-20"><i class="fa fa-check"></i><b>42.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="43" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html"><i class="fa fa-check"></i><b>43</b> Supplementary Figure 16</a>
<ul>
<li class="chapter" data-level="43.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#summary-40"><i class="fa fa-check"></i><b>43.1</b> Summary</a></li>
<li class="chapter" data-level="43.2" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#details-of-plot_sf16.r"><i class="fa fa-check"></i><b>43.2</b> Details of <code>plot_SF16.R</code></a>
<ul>
<li class="chapter" data-level="43.2.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#config-21"><i class="fa fa-check"></i><b>43.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="44" data-path="supplementary-figure-17.html"><a href="supplementary-figure-17.html"><i class="fa fa-check"></i><b>44</b> Supplementary Figure 17</a>
<ul>
<li class="chapter" data-level="44.1" data-path="supplementary-figure-17.html"><a href="supplementary-figure-17.html#summary-41"><i class="fa fa-check"></i><b>44.1</b> Summary</a></li>
<li class="chapter" data-level="44.2" data-path="supplementary-figure-17.html"><a href="supplementary-figure-17.html#details-of-plot_sf17.r"><i class="fa fa-check"></i><b>44.2</b> Details of <code>plot_SF17.R</code></a>
<ul>
<li class="chapter" data-level="44.2.1" data-path="supplementary-figure-17.html"><a href="supplementary-figure-17.html#config-22"><i class="fa fa-check"></i><b>44.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="45" data-path="supplementary-figure-18.html"><a href="supplementary-figure-18.html"><i class="fa fa-check"></i><b>45</b> Supplementary Figure 18</a>
<ul>
<li class="chapter" data-level="45.1" data-path="supplementary-figure-18.html"><a href="supplementary-figure-18.html#summary-42"><i class="fa fa-check"></i><b>45.1</b> Summary</a></li>
<li class="chapter" data-level="45.2" data-path="supplementary-figure-18.html"><a href="supplementary-figure-18.html#details-of-plot_sf18.r"><i class="fa fa-check"></i><b>45.2</b> Details of <code>plot_SF18.R</code></a>
<ul>
<li class="chapter" data-level="45.2.1" data-path="supplementary-figure-18.html"><a href="supplementary-figure-18.html#config-23"><i class="fa fa-check"></i><b>45.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="46" data-path="supplementary-figure-19.html"><a href="supplementary-figure-19.html"><i class="fa fa-check"></i><b>46</b> Supplementary Figure 19</a>
<ul>
<li class="chapter" data-level="46.1" data-path="supplementary-figure-19.html"><a href="supplementary-figure-19.html#summary-43"><i class="fa fa-check"></i><b>46.1</b> Summary</a></li>
<li class="chapter" data-level="46.2" data-path="supplementary-figure-19.html"><a href="supplementary-figure-19.html#details-of-plot_sf19.r"><i class="fa fa-check"></i><b>46.2</b> Details of <code>plot_SF19.R</code></a>
<ul>
<li class="chapter" data-level="46.2.1" data-path="supplementary-figure-19.html"><a href="supplementary-figure-19.html#config-24"><i class="fa fa-check"></i><b>46.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="47" data-path="supplementary-figure-20.html"><a href="supplementary-figure-20.html"><i class="fa fa-check"></i><b>47</b> Supplementary Figure 20</a>
<ul>
<li class="chapter" data-level="47.1" data-path="supplementary-figure-20.html"><a href="supplementary-figure-20.html#summary-44"><i class="fa fa-check"></i><b>47.1</b> Summary</a></li>
<li class="chapter" data-level="47.2" data-path="supplementary-figure-20.html"><a href="supplementary-figure-20.html#details-of-plot_sf20.r"><i class="fa fa-check"></i><b>47.2</b> Details of <code>plot_SF20.R</code></a>
<ul>
<li class="chapter" data-level="47.2.1" data-path="supplementary-figure-20.html"><a href="supplementary-figure-20.html#config-25"><i class="fa fa-check"></i><b>47.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="48" data-path="supplementary-figure-21.html"><a href="supplementary-figure-21.html"><i class="fa fa-check"></i><b>48</b> Supplementary Figure 21</a>
<ul>
<li class="chapter" data-level="48.1" data-path="supplementary-figure-21.html"><a href="supplementary-figure-21.html#summary-45"><i class="fa fa-check"></i><b>48.1</b> Summary</a></li>
<li class="chapter" data-level="48.2" data-path="supplementary-figure-21.html"><a href="supplementary-figure-21.html#details-of-plot_sf21.r"><i class="fa fa-check"></i><b>48.2</b> Details of <code>plot_SF21.R</code></a>
<ul>
<li class="chapter" data-level="48.2.1" data-path="supplementary-figure-21.html"><a href="supplementary-figure-21.html#config-26"><i class="fa fa-check"></i><b>48.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="git-13-analysis-xi-whole-genome-phylogenies" class="section level1" number="14">
<h1><span class="header-section-number">14</span> (git 13) Analysis XI (Whole Genome Phylogenies)</h1>
<p>This pipeline can be executed as follows:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb18-1"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span>/nf/13_analysis_phylo_whg</span>
<span id="cb18-2"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="ex">nextflow</span> run analysis_phylo_whg.nf</span></code></pre></div>
<div id="summary-12" class="section level2" number="14.1">
<h2><span class="header-section-number">14.1</span> Summary</h2>
<p>The whole genome phylogenies can be reconstructed within the <a href="https://www.nextflow.io/"><strong>nextflow</strong></a> script <code>analysis_phylo_whg.nf</code> (located under <code>$BASE_DIR/nf/analysis_phylo_whg/</code>).</p>
</div>
<div id="details-of-analysis_phylo_whg.nf" class="section level2" number="14.2">
<h2><span class="header-section-number">14.2</span> Details of <code>analysis_phylo_whg.nf</code></h2>
<blockquote>
<p>This part of the analysis was actually manged manually and not via <code>nextflow</code>.
We still report the analysis as a <code>.nf</code> script as we believe this is a cleaner and more concise report of the conducted analysis.</p>
</blockquote>
<div id="setup-5" class="section level3" number="14.2.1">
<h3><span class="header-section-number">14.2.1</span> Setup</h3>
<p>The nextflow script starts by opening the genotype data and feeding it into two different streams.</p>
<div class="kclass">
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode">#<span class="hl opt">!/</span>usr<span class="hl opt">/</span>bin<span class="hl opt">/</span>env nextflow

<span class="hl slc">// git 13.1</span>
<span class="hl slc">// Open the SNP data set</span>
<span class="hl kwa">Channel</span>
    .fromFilePairs<span class="hl opt">(</span><span class="hl str">&quot;../../1_genotyping/4_phased/phased_mac2.vcf.{gz,gz.tbi}&quot;</span><span class="hl opt">)</span>
    .into<span class="hl opt">{</span> vcf_snps_ch<span class="hl opt">;</span> vcf_snps_ch2 <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>We also open the <em>allBP</em> data set (genotypes of SNPS + invariant sites).</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.2</span>
<span class="hl slc">// Open the allBP data set (will be expanded x 24 LGs)</span>
<span class="hl kwa">Channel</span>
    .fromFilePairs<span class="hl opt">(</span><span class="hl str">&quot;../../1_genotyping/3_gatk_filtered/filterd.allBP.non_ref.vcf.{gz,gz.tbi}&quot;</span><span class="hl opt">)</span>
    .set<span class="hl opt">{</span> vcf_allbp_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Then, we initialize the different window sizes considered (1-50 kb).</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.3</span>
<span class="hl kwa">Channel</span>
    .from<span class="hl opt">( [</span> <span class="hl num">1</span><span class="hl opt">,</span> <span class="hl num">5</span><span class="hl opt">,</span> <span class="hl num">10</span><span class="hl opt">,</span> <span class="hl num">50</span> <span class="hl opt">] )</span>
    .set<span class="hl opt">{</span> window_size_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Then we create one bed file per windowsize, containing the window positions.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.4</span>
<span class="hl slc">// Compile summary table</span>
<span class="hl kwa">process</span> segment_windows <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_loc_slice_windows&#39;</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../2_analysis/window_stats/windows/&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span> 

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwc">val</span><span class="hl opt">(</span> kb_size <span class="hl opt">)</span> <span class="hl kwa">from</span> window_size_ch

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> kb_size <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;windows_</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb.bed.gz&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> <span class="hl opt">(</span> windows_ch<span class="hl opt">,</span> windows_ch2<span class="hl opt">,</span> windows_ch3 <span class="hl opt">)</span>

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   #!/usr/bin/env Rscript</span>
<span class="hl str">   library(hypogen)</span>
<span class="hl str"></span>
<span class="hl str">   x_bp &lt;-</span> <span class="hl ipl">${kb_size}</span> <span class="hl str">* 1000</span>
<span class="hl str">   </span>
<span class="hl str">   window_non_overlap &lt;- function(CHROM, LENGTH, windowsize = x_bp ){</span>
<span class="hl str">     tibble(CHROM = CHROM, </span>
<span class="hl str">            START = seq(from = 1, to = LENGTH, by = windowsize) - 1, # (produces overlap of one which is needed for bedtools)</span>
<span class="hl str">            END = lead(START, default = LENGTH) ) }</span>
<span class="hl str"></span>
<span class="hl str">   hypo_karyotype %&gt;%</span>
<span class="hl str">     select(CHROM, LENGTH) %&gt;%</span>
<span class="hl str">     pmap_dfr(window_non_overlap) %&gt;%</span>
<span class="hl str">     write_tsv(file = &quot;windows_</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb.bed.gz&quot;)</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>We drop the Serranid samples (the outgroup) from the genotypes.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.5</span>
<span class="hl kwa">process</span> filter_hamlets <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_20g2h_outgroup_drop&#39;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span>  <span class="hl kwc">val</span><span class="hl opt">(</span> vcfId <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">)</span> <span class="hl kwa">from</span> vcf_snps_ch2

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> <span class="hl str">&quot;hamlets_only&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;hamlets_only.vcf.gz*&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> vcf_hamlets_ch

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   echo -e &quot;20478tabhon</span><span class="hl esc">\\</span><span class="hl str">n28393torpan</span><span class="hl esc">\\</span><span class="hl str">ns_tort_3torpan&quot; &gt; outgroup.pop</span>
<span class="hl str"></span>
<span class="hl str">   vcftools  --gzvcf</span> <span class="hl ipl">${vcf[0]}</span> <span class="hl str">\</span>
<span class="hl str">       --remove outgroup.pop \</span>
<span class="hl str">       --mac 1 \</span>
<span class="hl str">       --recode \</span>
<span class="hl str">       --stdout | \</span>
<span class="hl str">       bgzip &gt; hamlets_only.vcf.gz</span>
<span class="hl str">   </span>
<span class="hl str">   tabix hamlets_only.vcf.gz</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Then, we compute the coverage (n of SNPs) within each window.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.6</span>
<span class="hl slc">// Coverage of SNPs vcf for SNPdensity, allBP for Ns</span>
<span class="hl kwa">process</span> compute_coverage <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_140g1h_coverage&#39;</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../2_analysis/window_stats/coverages/&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span> 

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl slc">//set vcfId, file( vcf ), val( kb_size ), file( window ) from vcf_snps_filterd_ch.combine( windows_ch )</span>
    <span class="hl kwa">set</span>  <span class="hl kwc">val</span><span class="hl opt">(</span> vcfId <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> kb_size <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> window <span class="hl opt">)</span> <span class="hl kwa">from</span> vcf_snps_ch.concat<span class="hl opt">(</span> vcf_hamlets_ch <span class="hl opt">)</span>.map<span class="hl opt">{ [</span>it<span class="hl opt">[</span><span class="hl num">0</span><span class="hl opt">]</span>.minus<span class="hl opt">(</span><span class="hl str">&quot;.vcf&quot;</span><span class="hl opt">),</span> it<span class="hl opt">[</span><span class="hl num">1</span><span class="hl opt">]]}</span>.combine<span class="hl opt">(</span> windows_ch <span class="hl opt">)</span>
    
    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> kb_size <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;</span><span class="hl ipl">${vcfId}</span><span class="hl str">.</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb_cov.tsv.gz&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> coverage_ch

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   bedtools coverage \</span>
<span class="hl str">       -a</span> <span class="hl ipl">${window}</span> <span class="hl str">\</span>
<span class="hl str">       -b</span> <span class="hl ipl">${vcf[0]}</span> <span class="hl str">\</span>
<span class="hl str">       -counts | gzip &gt;</span> <span class="hl ipl">${vcfId}</span><span class="hl str">.</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb_cov.tsv.gz</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>To split the genotypes by linkage group, we initialize a LG-channel…</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.7</span>
<span class="hl kwa">Channel</span>
    .from<span class="hl opt">( (</span><span class="hl str">&#39;01&#39;</span>..<span class="hl str">&#39;09&#39;</span><span class="hl opt">) + (</span><span class="hl str">&#39;10&#39;</span>..<span class="hl str">&#39;19&#39;</span><span class="hl opt">) + (</span><span class="hl str">&#39;20&#39;</span>..<span class="hl str">&#39;24&#39;</span><span class="hl opt">) )</span>
    .set<span class="hl opt">{</span> lg_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>…and split the genotypes.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.8</span>
<span class="hl kwa">process</span> subset_allBP <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_140g10h_coverage&#39;</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../1_genotyping/3_gatk_filtered/non_ref_byLG/&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span> 

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl slc">//set vcfId, file( vcf ), val( kb_size ), file( window ) from vcf_snps_filterd_ch.combine( windows_ch )</span>
    <span class="hl kwa">set</span>  <span class="hl kwc">val</span><span class="hl opt">(</span> lg <span class="hl opt">)</span> <span class="hl kwa">from</span> lg_ch

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> lg <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;filterd.allBP.non_ref.LG</span><span class="hl ipl">${lg}</span><span class="hl str">.vcf.gz&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;filterd.allBP.non_ref.LG</span><span class="hl ipl">${lg}</span><span class="hl str">.vcf.gz.tbi&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> allbp_non_ref_ch

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   vcftools \</span>
<span class="hl str">       --gzvcf \$BASE_DIR/1_genotyping/3_gatk_filtered//filterd.allBP.non_ref.vcf.gz \</span>
<span class="hl str">       --chr LG</span><span class="hl ipl">${lg}</span> <span class="hl str">\</span>
<span class="hl str">       --recode \</span>
<span class="hl str">       --stdout | bgzip &gt; filterd.allBP.non_ref.LG</span><span class="hl ipl">${lg}</span><span class="hl str">.vcf.gz</span>
<span class="hl str"></span>
<span class="hl str">   tabix filterd.allBP.non_ref.LG</span><span class="hl ipl">${lg}</span><span class="hl str">.vcf.gz</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Then we also compute the coverage (in terms of callable sites, not just SNPs) within windows.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.9</span>
<span class="hl kwa">process</span> compute_coverage_allBP <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_140g1h_coverage&#39;</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../2_analysis/window_stats/coverages/&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span> 

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span>  <span class="hl kwc">val</span><span class="hl opt">(</span> lg <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> tbi <span class="hl opt">),</span> <span class="hl kwc">val</span><span class="hl opt">(</span> kb_size <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> window <span class="hl opt">)</span> <span class="hl kwa">from</span> allbp_non_ref_ch.combine<span class="hl opt">(</span> windows_ch2 <span class="hl opt">)</span>
    
    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> kb_size <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;filterd.allBP.LG</span><span class="hl ipl">${lg}</span><span class="hl str">.</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb_cov.tsv.gz&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> coverage_allbp_ch

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   echo -e &quot;CHROM</span><span class="hl esc">\\</span><span class="hl str">tSTART</span><span class="hl esc">\\</span><span class="hl str">tEND&quot; &gt; bed_LG</span><span class="hl ipl">${lg}</span><span class="hl str">.bed</span>
<span class="hl str"></span>
<span class="hl str">   zgrep &quot;LG</span><span class="hl ipl">${lg}</span><span class="hl str">&quot;</span> <span class="hl ipl">${window}</span> <span class="hl str">&gt;&gt; bed_LG</span><span class="hl ipl">${lg}</span><span class="hl str">.bed</span>
<span class="hl str">   gzip bed_LG</span><span class="hl ipl">${lg}</span><span class="hl str">.bed</span>
<span class="hl str"></span>
<span class="hl str">   bedtools coverage \</span>
<span class="hl str">       -a bed_LG</span><span class="hl ipl">${lg}</span><span class="hl str">.bed.gz \</span>
<span class="hl str">       -b</span> <span class="hl ipl">${vcf}</span> <span class="hl str">\</span>
<span class="hl str">       -counts | gzip &gt; filterd.allBP.LG</span><span class="hl ipl">${lg}</span><span class="hl str">.</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb_cov.tsv.gz</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>So, now we can merge the coverage information for each window.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.10</span>
<span class="hl slc">// Compile summary table</span>
<span class="hl kwa">process</span> complie_window_stats <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_20g2h_windows_stats&#39;</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../2_analysis/window_stats/window_stats/&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span> 

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> kb_size <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> windows <span class="hl opt">)</span> <span class="hl kwa">from</span> coverage_ch.concat<span class="hl opt">(</span> coverage_allbp_ch <span class="hl opt">)</span>.groupTuple<span class="hl opt">()</span>

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> kb_size <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;window_stats.</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb.tsv.gz&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> window_out_ch

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   #!/usr/bin/env Rscript</span>
<span class="hl str"></span>
<span class="hl str">   library(tidyverse)</span>
<span class="hl str"></span>
<span class="hl str">   data_SNPs &lt;- read_tsv(&quot;phased_mac2.</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb_cov.tsv.gz&quot;,</span>
<span class="hl str">                         col_names = c(&quot;CHROM&quot;, &quot;START&quot;, &quot;END&quot;, &quot;COV_SNP&quot;))</span>
<span class="hl str"></span>
<span class="hl str">   data_HYP &lt;- read_tsv(&quot;hamlets_only.</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb_cov.tsv.gz&quot;,</span>
<span class="hl str">                         col_names = c(&quot;CHROM&quot;, &quot;START&quot;, &quot;END&quot;, &quot;COV_HYP&quot;))</span>
<span class="hl str"></span>
<span class="hl str">   all_bp_files &lt;- dir(pattern = &quot;filterd.allBP.LG*&quot;)</span>
<span class="hl str"></span>
<span class="hl str">   data_allBPs &lt;- map_dfr(all_bp_files, .f = function(x){read_tsv(x, col_names = c(&quot;CHROM&quot;, &quot;START&quot;, &quot;END&quot;, &quot;COV_ALL&quot;))})</span>
<span class="hl str"></span>
<span class="hl str">   data &lt;- data_SNPs %&gt;%</span>
<span class="hl str">       left_join(data_HYP, by = c(CHROM = &quot;CHROM&quot;, START = &quot;START&quot;, END = &quot;END&quot;))  %&gt;%</span>
<span class="hl str">       left_join(data_allBPs, by = c(CHROM = &quot;CHROM&quot;, START = &quot;START&quot;, END = &quot;END&quot;))  %&gt;%</span>
<span class="hl str">       filter(COV_ALL &gt; 0 ) %&gt;%</span>
<span class="hl str">       mutate(SNP_density = round(COV_SNP/ COV_ALL, 2), </span>
<span class="hl str">       REL_COV =  round(COV_ALL/ (END-START), 2))</span>
<span class="hl str">   </span>
<span class="hl str">   write_tsv(x = data, file = &quot;window_stats.</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb.tsv.gz&quot;)</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>From this point onward, the analysis was actually done manually (we still format it according to nextflow for clarity).</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// ----------------------- DISCLAIMER ----------------------</span>
<span class="hl slc">// form here on, this pipeline was not actually run using</span>
<span class="hl slc">// nextflow, but managed manually</span>
<span class="hl slc">// ---------------------------------------------------------</span>
</code>
</pre>
</div>
<p>Now, the subset of windows to be considered for the phylogeny was sampled.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.11</span>
<span class="hl slc">// Subset to 5000 random windows</span>
<span class="hl kwa">process</span> subset_windows <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_loc_subset_windows&#39;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> kb_size <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> windows <span class="hl opt">)</span> <span class="hl kwa">from</span> window_out_ch.filter<span class="hl opt">({</span> it<span class="hl opt">[</span><span class="hl num">1</span><span class="hl opt">] ==</span> <span class="hl num">5</span> <span class="hl opt">})</span>

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> kb_size <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;5000x_</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb_v1.bed&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> window_subset_ch

    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   #!/usr/bin/env Rscript</span>
<span class="hl str">   library(hypogen)   # attaches tidyverse automatically</span>
<span class="hl str"></span>
<span class="hl str">   ### Draw windows with coverage (sequence contiguity) and SNP-based cutoffs</span>
<span class="hl str"></span>
<span class="hl str">   stats5 &lt;- read_tsv(&quot;window_stats.</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb.tsv.gz&quot;)</span>
<span class="hl str"></span>
<span class="hl str">   set.seed(64)</span>
<span class="hl str"></span>
<span class="hl str">   selected_windows &lt;- stats5 %&gt;%</span>
<span class="hl str">   filter(REL_COV &gt;= 0.80 &amp;</span>
<span class="hl str">           COV_HYP &gt;= 50) %&gt;%</span>
<span class="hl str">   sample_n(5000) %&gt;%</span>
<span class="hl str">   arrange(CHROM, START)</span>
<span class="hl str"></span>
<span class="hl str">   write_tsv(selected_windows %&gt;% select(CHROM, START, END), file = &quot;5000x_</span><span class="hl ipl">${kb_size}</span><span class="hl str">kb_v1.bed&quot;)</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>To parallelize the analysis, several looping-channels are opened.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.12</span>
<span class="hl slc">// index for sub-channels</span>
<span class="hl kwa">Channel</span>.from<span class="hl opt">(</span> <span class="hl num">1</span>..50 <span class="hl opt">)</span>.into<span class="hl opt">{</span> loop50_idx_ch1<span class="hl opt">;</span> loop50_idx_ch2<span class="hl opt">;</span> loop50_idx_ch3<span class="hl opt">;</span> loop50_idx_ch4 <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>For the selected windows, the genotypes are extracted.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.13</span>
<span class="hl slc">// Extract windows from all BP</span>
<span class="hl kwa">process</span> extract_windows <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_20g2h_extract_windows&#39;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> idx <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> windows <span class="hl opt">)</span> <span class="hl kwa">from</span> loop50_idx_ch1.combine<span class="hl opt">(</span> window_out_ch <span class="hl opt">)</span>

    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*_v1_all.vcf.gz&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> window_extract_loop
    
    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   PER_TASK=100</span>
<span class="hl str"></span>
<span class="hl str">   START_NUM=\$(( (</span><span class="hl ipl">${idx}</span> <span class="hl str">- 1) * \$PER_TASK + 1 ))</span>
<span class="hl str">   END_NUM=\$((</span> <span class="hl ipl">${idx}</span> <span class="hl str">* \$PER_TASK ))</span>
<span class="hl str"></span>
<span class="hl str">   echo This is task</span> <span class="hl ipl">${idx}</span><span class="hl str">, which will do runs \$START_NUM to \$END_NUM</span>
<span class="hl str"></span>
<span class="hl str">   for (( run=\$START_NUM ; run&lt;=END_NUM ; run++ ))</span>
<span class="hl str">       do</span>
<span class="hl str">       echo This is task</span> <span class="hl ipl">${idx}</span><span class="hl str">, run number \$run</span>
<span class="hl str">       chr=\$(awk -v line=&quot;\$run&quot; &#39;BEGIN { FS = &quot;</span><span class="hl esc">\\</span><span class="hl str">t&quot; } ; NR==line+1 { print \$1 }&#39; 5000x_5kb_v1.bed)</span>
<span class="hl str">       sta=\$(awk -v line=&quot;\$run&quot; &#39;BEGIN { FS = &quot;</span><span class="hl esc">\\</span><span class="hl str">t&quot; } ; NR==line+1 { print \$2 }&#39; 5000x_5kb_v1.bed)</span>
<span class="hl str">       end=\$(awk -v line=&quot;\$run&quot; &#39;BEGIN { FS = &quot;</span><span class="hl esc">\\</span><span class="hl str">t&quot; } ; NR==line+1 { print \$3 }&#39; 5000x_5kb_v1.bed)</span>
<span class="hl str">       printf -v i &quot;%04d&quot; \$run</span>
<span class="hl str"></span>
<span class="hl str">       vcftools \</span>
<span class="hl str">           --gzvcf \$BASE_DIR/1_genotyping/3_gatk_filtered/byLG/filterd.allBP.&quot;\$chr&quot;.vcf.gz \</span>
<span class="hl str">           --chr &quot;\$chr&quot; \</span>
<span class="hl str">           --from-bp &quot;\$sta&quot; \</span>
<span class="hl str">           --to-bp &quot;\$end&quot; \</span>
<span class="hl str">           --recode --stdout | \</span>
<span class="hl str">           grep -v &quot;##&quot; | bgzip &gt; window_&quot;\$i&quot;_v1_all.vcf.gz</span>
<span class="hl str">   done</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Again, a channel is created for parallelizing…</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.14</span>
<span class="hl slc">// index for sub-channels</span>
<span class="hl kwa">Channel</span>.from<span class="hl opt">(</span> <span class="hl num">1</span>..20 <span class="hl opt">)</span>.set<span class="hl opt">{</span> loop20_idx_ch <span class="hl opt">}</span>
<span class="hl slc">// bundle the distributed sub-channels</span>
window_extract_loop.collect<span class="hl opt">()</span>.map<span class="hl opt">{ [</span> it <span class="hl opt">] }</span>.set<span class="hl opt">{</span> window_extract_ch1<span class="hl opt">,</span> window_extract_ch2 <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>…and again, the outgroup samples are removed from the genoptypes.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.15</span>
<span class="hl slc">// Remove samples (all / noS data sets)</span>
<span class="hl kwa">process</span> remove_samples <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_20g2h_remove_samples&#39;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> idx <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf <span class="hl opt">)</span> <span class="hl kwa">from</span> loop20_idx_ch.combine<span class="hl opt">(</span> window_extract_ch1 <span class="hl opt">)</span>
    
    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*_noS.vcf.gz&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> samples_removed_loop
    
    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   PER_TASK=250</span>
<span class="hl str"></span>
<span class="hl str">   START_NUM=\$(( (</span><span class="hl ipl">${idx}</span> <span class="hl str">- 1) * \$PER_TASK + 1 ))</span>
<span class="hl str">   END_NUM=\$((</span> <span class="hl ipl">${idx}</span> <span class="hl str">* \$PER_TASK ))</span>
<span class="hl str"></span>
<span class="hl str">   echo This is task</span> <span class="hl ipl">${idx}</span><span class="hl str">, which will do runs \$START_NUM to \$END_NUM</span>
<span class="hl str"></span>
<span class="hl str">   for (( run=\$START_NUM ; run&lt;=END_NUM ; run++ ))</span>
<span class="hl str">       do</span>
<span class="hl str"></span>
<span class="hl str">       echo This is task</span> <span class="hl ipl">${idx}</span><span class="hl str">, run number \$run</span>
<span class="hl str">       </span>
<span class="hl str">       printf -v i &quot;%04d&quot; \$run</span>
<span class="hl str"></span>
<span class="hl str">       vcftools \</span>
<span class="hl str">           --gzvcf window_&quot;\$i&quot;*.vcf.gz \</span>
<span class="hl str">           --remove samples_Serr.txt \</span>
<span class="hl str">           --recode \</span>
<span class="hl str">           --stdout | bgzip &gt; window_&quot;\$i&quot;_v1_noS.vcf.gz</span>
<span class="hl str">   done</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
<span class="hl slc">// bundle the distributed sub-channels</span>
samples_removed_loop.collect<span class="hl opt">()</span>.map<span class="hl opt">{ [</span> it <span class="hl opt">] }</span>.set<span class="hl opt">{</span> samples_removed_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Then, we translate the genotypes from vcf to fasta format.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.16</span>
<span class="hl slc">// Convert to Fasta (IUPAC-encoded)</span>
<span class="hl kwa">process</span> fasta_convert <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_20g2h_convert_to_fasta&#39;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> idx <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf_noS <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> vcf_all <span class="hl opt">)</span> <span class="hl kwa">from</span> loop50_idx_ch2.combine<span class="hl opt">(</span> samples_removed_ch <span class="hl opt">)</span>.combine<span class="hl opt">(</span> window_extract_ch2 <span class="hl opt">)</span>
    
    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*.fas&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> fasta_convert_loop
    
    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   PER_TASK=100</span>
<span class="hl str"></span>
<span class="hl str">   START_NUM=\$(( (</span><span class="hl ipl">${idx}</span> <span class="hl str">- 1) * \$PER_TASK + 1 ))</span>
<span class="hl str">   END_NUM=\$((</span> <span class="hl ipl">${idx}</span> <span class="hl str">* \$PER_TASK ))</span>
<span class="hl str"></span>
<span class="hl str">   echo This is task</span> <span class="hl ipl">${idx}</span><span class="hl str">, which will do runs \$START_NUM to \$END_NUM</span>
<span class="hl str"></span>
<span class="hl str">   for (( run=\$START_NUM ; run&lt;=END_NUM ; run++ ))</span>
<span class="hl str">       do</span>
<span class="hl str">       echo This is task</span> <span class="hl ipl">${idx}</span><span class="hl str">, run number \$run</span>
<span class="hl str">       printf -v i &quot;%04d&quot; \$run</span>
<span class="hl str">       for j in all noS</span>
<span class="hl str">           do</span>
<span class="hl str">               bgzip -cd window_&quot;\$i&quot;_v1_&quot;\$j&quot;.vcf.gz | vcf-to-tab &gt; window_&quot;\$i&quot;_v1_&quot;\$j&quot;.tab</span>
<span class="hl str"></span>
<span class="hl str">               perl \$SFTWR/vcf-tab-to-fasta/vcf_tab_to_fasta_alignment.pl -i window_&quot;\$i&quot;_v1_&quot;\$j&quot;.tab &gt; window_&quot;\$i&quot;_v1_&quot;\$j&quot;.fas</span>
<span class="hl str"></span>
<span class="hl str">               rm window_&quot;\$i&quot;_v1_&quot;\$j&quot;.tab*</span>
<span class="hl str">       done</span>
<span class="hl str">   done</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
<span class="hl slc">// bundle the distributed sub-channels</span>
fasta_convert_loop.collect<span class="hl opt">()</span>.map<span class="hl opt">{ [</span> it <span class="hl opt">] }</span>.set<span class="hl opt">{</span> fasta_convert_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>The fasta sequences are aligned.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.17</span>
<span class="hl slc">// Align sequences in windows</span>
<span class="hl kwa">process</span> fasta_align <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_20g2h_fasta_align&#39;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> idx <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> fa <span class="hl opt">)</span> <span class="hl kwa">from</span> loop50_idx_ch3.combine<span class="hl opt">(</span> fasta_convert_ch <span class="hl opt">)</span>
    
    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*.aln&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> fasta_align_loop
    
    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   PER_TASK=100</span>
<span class="hl str"></span>
<span class="hl str">   START_NUM=\$(( (</span><span class="hl ipl">${idx}</span> <span class="hl str">- 1) * \$PER_TASK + 1 ))</span>
<span class="hl str">   END_NUM=\$((</span> <span class="hl ipl">${idx}</span> <span class="hl str">* \$PER_TASK ))</span>
<span class="hl str"></span>
<span class="hl str">   echo This is task</span> <span class="hl ipl">${idx}</span><span class="hl str">, which will do runs \$START_NUM to \$END_NUM</span>
<span class="hl str"></span>
<span class="hl str">   for (( run=</span><span class="hl ipl">${idx}</span> <span class="hl str">; run&lt;=END_NUM ; run++ ))</span>
<span class="hl str">       do</span>
<span class="hl str">       echo This is task</span> <span class="hl ipl">${idx}</span><span class="hl str">, run number \$run</span>
<span class="hl str">       printf -v i &quot;%04d&quot; \$run</span>
<span class="hl str">       for j in all noS</span>
<span class="hl str">           do</span>
<span class="hl str">               mafft --auto window_&quot;\$i&quot;_v1_&quot;\$j&quot;.fas &gt; &quot;window_&quot;\$i&quot;_v1_&quot;\$j&quot;.aln</span>
<span class="hl str">       done</span>
<span class="hl str">   done</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
<span class="hl slc">// bundle the distributed sub-channels</span>
fasta_align_loop.collect<span class="hl opt">()</span>.map<span class="hl opt">{ [</span> it <span class="hl opt">] }</span>.set<span class="hl opt">{</span> fasta_align_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Next, the trees are inferred within each window.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.18</span>
<span class="hl slc">// Infer local trees</span>
<span class="hl kwa">process</span> local_trees <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_20g2h_local_trees&#39;</span>

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">val</span><span class="hl opt">(</span> idx <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> aln <span class="hl opt">)</span> <span class="hl kwa">from</span> loop50_idx_ch4.combine<span class="hl opt">(</span> fasta_align_ch <span class="hl opt">)</span>
    
    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;*.treefile&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> local_trees_loop
    
    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   PER_TASK=100</span>
<span class="hl str"></span>
<span class="hl str">   START_NUM=$(( (</span><span class="hl ipl">${idx}</span><span class="hl str">- 1) * \$PER_TASK + 1 ))</span>
<span class="hl str">   END_NUM=$((</span> <span class="hl ipl">${idx}</span> <span class="hl str">* \$PER_TASK ))</span>
<span class="hl str"></span>
<span class="hl str">   echo This is task</span> <span class="hl ipl">${idx}</span><span class="hl str">, which will do runs \$START_NUM to \$END_NUM</span>
<span class="hl str"></span>
<span class="hl str">   for (( run=</span><span class="hl ipl">${idx}</span> <span class="hl str">; run&lt;=END_NUM ; run++ ))</span>
<span class="hl str">       do</span>
<span class="hl str">       echo This is task</span> <span class="hl ipl">${idx}</span><span class="hl str">, run number \$run</span>
<span class="hl str">       printf -v i &quot;%04d&quot; \$run</span>
<span class="hl str">       for j in all noS</span>
<span class="hl str">           do</span>
<span class="hl str">               iqtree2 -s window_&quot;\$i&quot;_v1_&quot;\$j&quot;.aln  --prefix locus_&quot;\$i&quot;_v1_&quot;\$j&quot; -o PL17_160floflo -T 1</span>
<span class="hl str">       done</span>
<span class="hl str">   done</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
<span class="hl slc">// bundle the distributed sub-channels</span>
local_trees_loop.collect<span class="hl opt">()</span>.map<span class="hl opt">{ [</span> it <span class="hl opt">] }</span>.set<span class="hl opt">{</span> local_trees_ch <span class="hl opt">}</span>
</code>
</pre>
</div>
<p>Finally, the individual trees are summarized.</p>
<div class="sourceCode">
<pre class="sourceCode">
<code class="sourceCode"><span class="hl slc">// git 13.19</span>
<span class="hl slc">// Calculate summary tree</span>
<span class="hl kwa">process</span> summary_trees <span class="hl opt">{</span>
    <span class="hl kwb">label</span> <span class="hl str">&#39;L_20g2h_summary_trees&#39;</span>
    <span class="hl kwb">publishDir</span> <span class="hl str">&quot;../../2_analysis/astral/&quot;</span><span class="hl opt">,</span> mode<span class="hl opt">:</span> <span class="hl str">&#39;copy&#39;</span> 

    <span class="hl kwb">input</span><span class="hl opt">:</span>
    <span class="hl kwc">file</span><span class="hl opt">(</span> tree <span class="hl opt">)</span> <span class="hl kwa">from</span> local_trees_ch
    
    <span class="hl kwb">output</span><span class="hl opt">:</span>
    <span class="hl kwa">set</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;astral*.tre&quot;</span> <span class="hl opt">),</span> <span class="hl kwc">file</span><span class="hl opt">(</span> <span class="hl str">&quot;astral*.log&quot;</span> <span class="hl opt">)</span> <span class="hl kwa">into</span> summary_trees_ch
    
    <span class="hl kwb">script</span><span class="hl opt">:</span>
    <span class="hl str">&quot;&quot;&quot;</span>
<span class="hl str">   cat ./*_noS.treefile &gt; genetrees_5000x_5kb_v1_noS.tre</span>
<span class="hl str">   java -jar \$SFTWR/ASTRAL_5.7.5/astral.5.7.5.jar -i genetrees_5000x_5kb_v1_noS.tre -o astral_5000x_5kb_v1_noS.tre 2&gt; astral_5000x_5kb_v1_noS.log</span>
<span class="hl str"></span>
<span class="hl str">   cat ./*_all.treefile &gt; genetrees_5000x_5kb_v1_all.tre</span>
<span class="hl str">   java -jar \$SFTWR/ASTRAL_5.7.5/astral.5.7.5.jar -i genetrees_5000x_5kb_v1_all.tre -o astral_5000x_5kb_v1_all.tre 2&gt; astral_5000x_5kb_v1_all.log</span>
<span class="hl str">   &quot;&quot;&quot;</span>
<span class="hl opt">}</span>
</code>
</pre>
</div>
</div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="git-12-analysis-x-fst-permutation-test.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="git-14-analysis-xii-outlier-region-phylogenies.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
