<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>35 Supplementary Figure 13 | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="35 Supplementary Figure 13 | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="35 Supplementary Figure 13 | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of ‘Ancestral variation, hybridization and modularity fuel a marine radiation’ by Hench, Helmkampf, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench and Martin Helmkampf" />


<meta name="date" content="2021-08-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="supplementary-figure-12.html"/>
<link rel="next" href="supplementary-figure-14.html"/>
<script src="libs/header-attrs-2.7.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#r-setup"><i class="fa fa-check"></i><b>1.4</b> R setup</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a>
<ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a>
<ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i> &amp; pi)</a>
<ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (topology weighting)</a>
<ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-topology-weighting.html"><a href="git-5-analysis-iii-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a>
<ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (Principal Component Analysis)</a>
<ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#details-of-analysis_pca.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_pca.nf</code></a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-principal-component-analysis.html"><a href="git-7-analysis-v-principal-component-analysis.html#setup-2"><i class="fa fa-check"></i><b>8.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a>
<ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-4"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-5"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-6"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html"><i class="fa fa-check"></i><b>12</b> (git 11) Analysis IX (Allele Age)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#summary-10"><i class="fa fa-check"></i><b>12.1</b> Summary</a></li>
<li class="chapter" data-level="12.2" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#details-of-analysis_allele_age.nf"><i class="fa fa-check"></i><b>12.2</b> Details of <code>analysis_allele_age.nf</code></a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="git-11-analysis-ix-allele-age.html"><a href="git-11-analysis-ix-allele-age.html#setup-3"><i class="fa fa-check"></i><b>12.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html"><i class="fa fa-check"></i><b>13</b> (git 12) Analysis X (<i>F<sub>ST</sub></i> Permutation Test)</a>
<ul>
<li class="chapter" data-level="13.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#summary-11"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#details-of-analysis_fst_sign.nf"><i class="fa fa-check"></i><b>13.2</b> Details of <code>analysis_fst_sign.nf</code></a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="git-12-analysis-x-fst-permutation-test.html"><a href="git-12-analysis-x-fst-permutation-test.html#setup-4"><i class="fa fa-check"></i><b>13.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html"><i class="fa fa-check"></i><b>14</b> (git 13) Analysis XI (Whole Genome Phylogenies)</a>
<ul>
<li class="chapter" data-level="14.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#summary-12"><i class="fa fa-check"></i><b>14.1</b> Summary</a></li>
<li class="chapter" data-level="14.2" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#details-of-analysis_phylo_whg.nf"><i class="fa fa-check"></i><b>14.2</b> Details of <code>analysis_phylo_whg.nf</code></a>
<ul>
<li class="chapter" data-level="14.2.1" data-path="git-13-analysis-xi-whole-genome-phylogenies.html"><a href="git-13-analysis-xi-whole-genome-phylogenies.html#setup-5"><i class="fa fa-check"></i><b>14.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html"><i class="fa fa-check"></i><b>15</b> (git 14) Analysis XII (Outlier Region Phylogenies)</a>
<ul>
<li class="chapter" data-level="15.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#summary-13"><i class="fa fa-check"></i><b>15.1</b> Summary</a></li>
<li class="chapter" data-level="15.2" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#details-of-analysis_phylo_regions.nf"><i class="fa fa-check"></i><b>15.2</b> Details of <code>analysis_phylo_regions.nf</code></a>
<ul>
<li class="chapter" data-level="15.2.1" data-path="git-14-analysis-xii-outlier-region-phylogenies.html"><a href="git-14-analysis-xii-outlier-region-phylogenies.html#setup-6"><i class="fa fa-check"></i><b>15.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="git-15-data-visualization.html"><a href="git-15-data-visualization.html"><i class="fa fa-check"></i><b>16</b> (git 15) Data Visualization</a></li>
<li class="chapter" data-level="17" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>17</b> Figure 1</a>
<ul>
<li class="chapter" data-level="17.1" data-path="figure-1.html"><a href="figure-1.html#summary-14"><i class="fa fa-check"></i><b>17.1</b> Summary</a></li>
<li class="chapter" data-level="17.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>17.2</b> Details of <code>plot_F1.R</code></a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>17.2.1</b> Config</a></li>
<li class="chapter" data-level="17.2.2" data-path="figure-1.html"><a href="figure-1.html#fish-tree-of-life-subtree"><i class="fa fa-check"></i><b>17.2.2</b> Fish Tree Of Life Subtree</a></li>
<li class="chapter" data-level="17.2.3" data-path="figure-1.html"><a href="figure-1.html#genetic-differentiation-boxplots"><i class="fa fa-check"></i><b>17.2.3</b> Genetic differentiation boxplots</a></li>
<li class="chapter" data-level="17.2.4" data-path="figure-1.html"><a href="figure-1.html#principal-component-analyses"><i class="fa fa-check"></i><b>17.2.4</b> Principal Component Analyses</a></li>
<li class="chapter" data-level="17.2.5" data-path="figure-1.html"><a href="figure-1.html#legend-and-final-figure"><i class="fa fa-check"></i><b>17.2.5</b> Legend and Final Figure</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>18</b> Figure 2</a>
<ul>
<li class="chapter" data-level="18.1" data-path="figure-2.html"><a href="figure-2.html#summary-15"><i class="fa fa-check"></i><b>18.1</b> Summary</a></li>
<li class="chapter" data-level="18.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>18.2</b> Details of <code>plot_F2.R</code></a>
<ul>
<li class="chapter" data-level="18.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>18.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>19</b> Figure 3</a>
<ul>
<li class="chapter" data-level="19.1" data-path="figure-3.html"><a href="figure-3.html#summary-16"><i class="fa fa-check"></i><b>19.1</b> Summary</a></li>
<li class="chapter" data-level="19.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>19.2</b> Details of <code>plot_F3.R</code></a>
<ul>
<li class="chapter" data-level="19.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>19.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>20</b> Figure 4</a>
<ul>
<li class="chapter" data-level="20.1" data-path="figure-4.html"><a href="figure-4.html#summary-17"><i class="fa fa-check"></i><b>20.1</b> Summary</a></li>
<li class="chapter" data-level="20.2" data-path="figure-4.html"><a href="figure-4.html#details-of-plot_f4.r"><i class="fa fa-check"></i><b>20.2</b> Details of <code>plot_F4.R</code></a>
<ul>
<li class="chapter" data-level="20.2.1" data-path="figure-4.html"><a href="figure-4.html#config-3"><i class="fa fa-check"></i><b>20.2.1</b> Config</a></li>
<li class="chapter" data-level="20.2.2" data-path="figure-4.html"><a href="figure-4.html#data-import"><i class="fa fa-check"></i><b>20.2.2</b> Data import</a></li>
<li class="chapter" data-level="20.2.3" data-path="figure-4.html"><a href="figure-4.html#plotting"><i class="fa fa-check"></i><b>20.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>21</b> Figure 5</a>
<ul>
<li class="chapter" data-level="21.1" data-path="figure-5.html"><a href="figure-5.html#summary-18"><i class="fa fa-check"></i><b>21.1</b> Summary</a></li>
<li class="chapter" data-level="21.2" data-path="figure-5.html"><a href="figure-5.html#details-of-plot_f5.r"><i class="fa fa-check"></i><b>21.2</b> Details of <code>plot_F5.R</code></a>
<ul>
<li class="chapter" data-level="21.2.1" data-path="figure-5.html"><a href="figure-5.html#config-4"><i class="fa fa-check"></i><b>21.2.1</b> Config</a></li>
<li class="chapter" data-level="21.2.2" data-path="figure-5.html"><a href="figure-5.html#data-import-1"><i class="fa fa-check"></i><b>21.2.2</b> Data import</a></li>
<li class="chapter" data-level="21.2.3" data-path="figure-5.html"><a href="figure-5.html#plotting-1"><i class="fa fa-check"></i><b>21.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="figure-6.html"><a href="figure-6.html"><i class="fa fa-check"></i><b>22</b> Figure 6</a>
<ul>
<li class="chapter" data-level="22.1" data-path="figure-6.html"><a href="figure-6.html#summary-19"><i class="fa fa-check"></i><b>22.1</b> Summary</a></li>
<li class="chapter" data-level="22.2" data-path="figure-6.html"><a href="figure-6.html#details-of-plot_f6.r"><i class="fa fa-check"></i><b>22.2</b> Details of <code>plot_F6.R</code></a>
<ul>
<li class="chapter" data-level="22.2.1" data-path="figure-6.html"><a href="figure-6.html#config-5"><i class="fa fa-check"></i><b>22.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html"><i class="fa fa-check"></i><b>23</b> Supplementary Figure 1</a>
<ul>
<li class="chapter" data-level="23.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#summary-20"><i class="fa fa-check"></i><b>23.1</b> Summary</a></li>
<li class="chapter" data-level="23.2" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#details-of-plot_sf1.r"><i class="fa fa-check"></i><b>23.2</b> Details of <code>plot_SF1.R</code></a>
<ul>
<li class="chapter" data-level="23.2.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#config-6"><i class="fa fa-check"></i><b>23.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html"><i class="fa fa-check"></i><b>24</b> Supplementary Figure 2</a>
<ul>
<li class="chapter" data-level="24.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#summary-21"><i class="fa fa-check"></i><b>24.1</b> Summary</a></li>
<li class="chapter" data-level="24.2" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#details-of-plot_sf2.r"><i class="fa fa-check"></i><b>24.2</b> Details of <code>plot_SF2.R</code></a>
<ul>
<li class="chapter" data-level="24.2.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#config-7"><i class="fa fa-check"></i><b>24.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html"><i class="fa fa-check"></i><b>25</b> Supplementary Figure 3</a>
<ul>
<li class="chapter" data-level="25.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#summary-22"><i class="fa fa-check"></i><b>25.1</b> Summary</a></li>
<li class="chapter" data-level="25.2" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#details-of-plot_sf3.r"><i class="fa fa-check"></i><b>25.2</b> Details of <code>plot_SF3.R</code></a>
<ul>
<li class="chapter" data-level="25.2.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#config-8"><i class="fa fa-check"></i><b>25.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html"><i class="fa fa-check"></i><b>26</b> Supplementary Figure 4</a>
<ul>
<li class="chapter" data-level="26.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#summary-23"><i class="fa fa-check"></i><b>26.1</b> Summary</a></li>
<li class="chapter" data-level="26.2" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#details-of-plot_sf4.r"><i class="fa fa-check"></i><b>26.2</b> Details of <code>plot_SF4.R</code></a>
<ul>
<li class="chapter" data-level="26.2.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#config-9"><i class="fa fa-check"></i><b>26.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="27" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html"><i class="fa fa-check"></i><b>27</b> Supplementary Figure 5</a>
<ul>
<li class="chapter" data-level="27.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#summary-24"><i class="fa fa-check"></i><b>27.1</b> Summary</a></li>
<li class="chapter" data-level="27.2" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#details-of-plot_sf5.r"><i class="fa fa-check"></i><b>27.2</b> Details of <code>plot_SF5.R</code></a>
<ul>
<li class="chapter" data-level="27.2.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#config-10"><i class="fa fa-check"></i><b>27.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="28" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html"><i class="fa fa-check"></i><b>28</b> Supplementary Figure 6</a>
<ul>
<li class="chapter" data-level="28.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#summary-25"><i class="fa fa-check"></i><b>28.1</b> Summary</a></li>
<li class="chapter" data-level="28.2" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#details-of-plot_sf6.r"><i class="fa fa-check"></i><b>28.2</b> Details of <code>plot_SF6.R</code></a>
<ul>
<li class="chapter" data-level="28.2.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#config-11"><i class="fa fa-check"></i><b>28.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="29" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html"><i class="fa fa-check"></i><b>29</b> Supplementary Figure 7</a>
<ul>
<li class="chapter" data-level="29.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#summary-26"><i class="fa fa-check"></i><b>29.1</b> Summary</a></li>
<li class="chapter" data-level="29.2" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#details-of-plot_sf7.r"><i class="fa fa-check"></i><b>29.2</b> Details of <code>plot_SF7.R</code></a>
<ul>
<li class="chapter" data-level="29.2.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#config-12"><i class="fa fa-check"></i><b>29.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="30" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html"><i class="fa fa-check"></i><b>30</b> Supplementary Figure 8</a>
<ul>
<li class="chapter" data-level="30.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#summary-27"><i class="fa fa-check"></i><b>30.1</b> Summary</a></li>
<li class="chapter" data-level="30.2" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#details-of-plot_sf8.r"><i class="fa fa-check"></i><b>30.2</b> Details of <code>plot_SF8.R</code></a>
<ul>
<li class="chapter" data-level="30.2.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#config-13"><i class="fa fa-check"></i><b>30.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="31" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html"><i class="fa fa-check"></i><b>31</b> Supplementary Figure 9</a>
<ul>
<li class="chapter" data-level="31.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#summary-28"><i class="fa fa-check"></i><b>31.1</b> Summary</a></li>
<li class="chapter" data-level="31.2" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#details-of-plot_sf9.r"><i class="fa fa-check"></i><b>31.2</b> Details of <code>plot_SF9.R</code></a>
<ul>
<li class="chapter" data-level="31.2.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#config-14"><i class="fa fa-check"></i><b>31.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="32" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html"><i class="fa fa-check"></i><b>32</b> Supplementary Figure 10</a>
<ul>
<li class="chapter" data-level="32.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#summary-29"><i class="fa fa-check"></i><b>32.1</b> Summary</a></li>
<li class="chapter" data-level="32.2" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#details-of-plot_sf10.r"><i class="fa fa-check"></i><b>32.2</b> Details of <code>plot_SF10.R</code></a>
<ul>
<li class="chapter" data-level="32.2.1" data-path="supplementary-figure-10.html"><a href="supplementary-figure-10.html#config-15"><i class="fa fa-check"></i><b>32.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="33" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html"><i class="fa fa-check"></i><b>33</b> Supplementary Figure 11</a>
<ul>
<li class="chapter" data-level="33.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#summary-30"><i class="fa fa-check"></i><b>33.1</b> Summary</a></li>
<li class="chapter" data-level="33.2" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#details-of-plot_sf11.r"><i class="fa fa-check"></i><b>33.2</b> Details of <code>plot_SF11.R</code></a>
<ul>
<li class="chapter" data-level="33.2.1" data-path="supplementary-figure-11.html"><a href="supplementary-figure-11.html#config-16"><i class="fa fa-check"></i><b>33.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="34" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html"><i class="fa fa-check"></i><b>34</b> Supplementary Figure 12</a>
<ul>
<li class="chapter" data-level="34.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#summary-31"><i class="fa fa-check"></i><b>34.1</b> Summary</a></li>
<li class="chapter" data-level="34.2" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#details-of-plot_sf12.r"><i class="fa fa-check"></i><b>34.2</b> Details of <code>plot_SF12.R</code></a>
<ul>
<li class="chapter" data-level="34.2.1" data-path="supplementary-figure-12.html"><a href="supplementary-figure-12.html#config-17"><i class="fa fa-check"></i><b>34.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="35" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html"><i class="fa fa-check"></i><b>35</b> Supplementary Figure 13</a>
<ul>
<li class="chapter" data-level="35.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#summary-32"><i class="fa fa-check"></i><b>35.1</b> Summary</a></li>
<li class="chapter" data-level="35.2" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#details-of-plot_sf13.r"><i class="fa fa-check"></i><b>35.2</b> Details of <code>plot_SF13.R</code></a>
<ul>
<li class="chapter" data-level="35.2.1" data-path="supplementary-figure-13.html"><a href="supplementary-figure-13.html#config-18"><i class="fa fa-check"></i><b>35.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="36" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html"><i class="fa fa-check"></i><b>36</b> Supplementary Figure 14</a>
<ul>
<li class="chapter" data-level="36.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#summary-33"><i class="fa fa-check"></i><b>36.1</b> Summary</a></li>
<li class="chapter" data-level="36.2" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#details-of-plot_sf14.r"><i class="fa fa-check"></i><b>36.2</b> Details of <code>plot_SF14.R</code></a>
<ul>
<li class="chapter" data-level="36.2.1" data-path="supplementary-figure-14.html"><a href="supplementary-figure-14.html#config-19"><i class="fa fa-check"></i><b>36.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="37" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html"><i class="fa fa-check"></i><b>37</b> Supplementary Figure 15</a>
<ul>
<li class="chapter" data-level="37.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#summary-34"><i class="fa fa-check"></i><b>37.1</b> Summary</a></li>
<li class="chapter" data-level="37.2" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#details-of-plot_sf15.r"><i class="fa fa-check"></i><b>37.2</b> Details of <code>plot_SF15.R</code></a>
<ul>
<li class="chapter" data-level="37.2.1" data-path="supplementary-figure-15.html"><a href="supplementary-figure-15.html#config-20"><i class="fa fa-check"></i><b>37.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="38" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html"><i class="fa fa-check"></i><b>38</b> Supplementary Figure 16</a>
<ul>
<li class="chapter" data-level="38.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#summary-35"><i class="fa fa-check"></i><b>38.1</b> Summary</a></li>
<li class="chapter" data-level="38.2" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#details-of-plot_sf16.r"><i class="fa fa-check"></i><b>38.2</b> Details of <code>plot_SF16.R</code></a>
<ul>
<li class="chapter" data-level="38.2.1" data-path="supplementary-figure-16.html"><a href="supplementary-figure-16.html#config-21"><i class="fa fa-check"></i><b>38.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="supplementary-figure-13" class="section level1" number="35">
<h1><span class="header-section-number">35</span> Supplementary Figure 13</h1>
<div id="summary-32" class="section level2" number="35.1">
<h2><span class="header-section-number">35.1</span> Summary</h2>
<p>This is the accessory documentation of Figure S13.
The Figure can be recreated by running the <strong>R</strong> script <code>plot_SF13.R</code>:</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb290-1"><a href="supplementary-figure-13.html#cb290-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span></span>
<span id="cb290-2"><a href="supplementary-figure-13.html#cb290-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb290-3"><a href="supplementary-figure-13.html#cb290-3" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> <span class="at">--vanilla</span> R/fig/plot_SF13.R<span class="dt">\</span></span>
<span id="cb290-4"><a href="supplementary-figure-13.html#cb290-4" aria-hidden="true" tabindex="-1"></a>  2_analysis/admixture/ <span class="dt">\</span></span>
<span id="cb290-5"><a href="supplementary-figure-13.html#cb290-5" aria-hidden="true" tabindex="-1"></a>  metadata/phenotypes.sc</span></code></pre></div>
</div>
<div id="details-of-plot_sf13.r" class="section level2" number="35.2">
<h2><span class="header-section-number">35.2</span> Details of <code>plot_SF13.R</code></h2>
<p>In the following, the individual steps of the R script are documented.
It is an executable R script that depends on the accessory R package <a href="https://k-hench.github.io/GenomicOriginsScripts"><strong>GenomicOriginsScripts</strong></a>, as well as on the packages <a href="https://k-hench.github.io/hypoimg"><strong>hypoimg</strong></a>, <a href="https://k-hench.github.io/hypogen"><strong>hypogen</strong></a>, <a href="https://wilkelab.org/ggtext/"><strong>ggtext</strong></a>, <a href="https://patchwork.data-imaginist.com/"><strong>patchwork</strong></a> and <a href="https://emilhvitfeldt.github.io/paletteer/"><strong>paletteer</strong></a>.</p>
<div id="config-18" class="section level3" number="35.2.1">
<h3><span class="header-section-number">35.2.1</span> Config</h3>
<p>The scripts start with a header that contains copy &amp; paste templates to execute or debug the script:</p>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="supplementary-figure-13.html#cb291-1" aria-hidden="true" tabindex="-1"></a><span class="co">#!/usr/bin/env Rscript</span></span>
<span id="cb291-2"><a href="supplementary-figure-13.html#cb291-2" aria-hidden="true" tabindex="-1"></a><span class="co"># run from terminal:</span></span>
<span id="cb291-3"><a href="supplementary-figure-13.html#cb291-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Rscript --vanilla R/fig/plot_SF13.R 2_analysis/admixture/ metadata/phenotypes.sc</span></span>
<span id="cb291-4"><a href="supplementary-figure-13.html#cb291-4" aria-hidden="true" tabindex="-1"></a><span class="co"># ===============================================================</span></span>
<span id="cb291-5"><a href="supplementary-figure-13.html#cb291-5" aria-hidden="true" tabindex="-1"></a><span class="co"># This script produces Suppl. Figure 13 of the study &quot;Ancestral variation,</span></span>
<span id="cb291-6"><a href="supplementary-figure-13.html#cb291-6" aria-hidden="true" tabindex="-1"></a><span class="co"># hybridization and modularity fuel a marine radiation&quot;</span></span>
<span id="cb291-7"><a href="supplementary-figure-13.html#cb291-7" aria-hidden="true" tabindex="-1"></a><span class="co"># by Hench, Helmkampf, McMillan and Puebla</span></span>
<span id="cb291-8"><a href="supplementary-figure-13.html#cb291-8" aria-hidden="true" tabindex="-1"></a><span class="co"># ---------------------------------------------------------------</span></span>
<span id="cb291-9"><a href="supplementary-figure-13.html#cb291-9" aria-hidden="true" tabindex="-1"></a><span class="co"># ===============================================================</span></span>
<span id="cb291-10"><a href="supplementary-figure-13.html#cb291-10" aria-hidden="true" tabindex="-1"></a><span class="co"># args &lt;- c( &quot;2_analysis/admixture/&quot;, &quot;metadata/phenotypes.sc&quot;)</span></span>
<span id="cb291-11"><a href="supplementary-figure-13.html#cb291-11" aria-hidden="true" tabindex="-1"></a><span class="co"># script_name &lt;- &quot;R/fig/plot_SF13.R&quot;</span></span></code></pre></div>
<p>The next section processes the input from the command line.
It stores the arguments in the vector <code>args</code>.
The needed R packages are and the script name and the current working directory are stored inside variables (<code>script_name</code>, <code>plot_comment</code>).
This information will later be written into the meta data of the figure to help us tracing back the scripts that created the figures in the future.</p>
<p>Then we drop all the imported information besides the arguments following the script name and print the information to the terminal.</p>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="supplementary-figure-13.html#cb292-1" aria-hidden="true" tabindex="-1"></a>args <span class="ot">&lt;-</span> <span class="fu">commandArgs</span>(<span class="at">trailingOnly=</span><span class="cn">FALSE</span>)</span>
<span id="cb292-2"><a href="supplementary-figure-13.html#cb292-2" aria-hidden="true" tabindex="-1"></a><span class="co"># setup -----------------------</span></span>
<span id="cb292-3"><a href="supplementary-figure-13.html#cb292-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(paletteer)</span>
<span id="cb292-4"><a href="supplementary-figure-13.html#cb292-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb292-5"><a href="supplementary-figure-13.html#cb292-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicOriginsScripts)</span>
<span id="cb292-6"><a href="supplementary-figure-13.html#cb292-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hypoimg)</span>
<span id="cb292-7"><a href="supplementary-figure-13.html#cb292-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hypogen)</span>
<span id="cb292-8"><a href="supplementary-figure-13.html#cb292-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtext)</span>
<span id="cb292-9"><a href="supplementary-figure-13.html#cb292-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-10"><a href="supplementary-figure-13.html#cb292-10" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>)</span>
<span id="cb292-11"><a href="supplementary-figure-13.html#cb292-11" aria-hidden="true" tabindex="-1"></a>script_name <span class="ot">&lt;-</span> args[<span class="dv">5</span>] <span class="sc">%&gt;%</span></span>
<span id="cb292-12"><a href="supplementary-figure-13.html#cb292-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_remove</span>(.,<span class="st">&#39;--file=&#39;</span>)</span>
<span id="cb292-13"><a href="supplementary-figure-13.html#cb292-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-14"><a href="supplementary-figure-13.html#cb292-14" aria-hidden="true" tabindex="-1"></a>plot_comment <span class="ot">&lt;-</span> script_name <span class="sc">%&gt;%</span></span>
<span id="cb292-15"><a href="supplementary-figure-13.html#cb292-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_c</span>(<span class="st">&#39;mother-script = &#39;</span>,<span class="fu">getwd</span>(),<span class="st">&#39;/&#39;</span>,.)</span>
<span id="cb292-16"><a href="supplementary-figure-13.html#cb292-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-17"><a href="supplementary-figure-13.html#cb292-17" aria-hidden="true" tabindex="-1"></a>cli<span class="sc">::</span><span class="fu">rule</span>( <span class="at">left =</span> <span class="fu">str_c</span>(crayon<span class="sc">::</span><span class="fu">bold</span>(<span class="st">&#39;Script: &#39;</span>),crayon<span class="sc">::</span><span class="fu">red</span>(script_name)))</span>
<span id="cb292-18"><a href="supplementary-figure-13.html#cb292-18" aria-hidden="true" tabindex="-1"></a>args <span class="ot">=</span> args[<span class="dv">7</span><span class="sc">:</span><span class="fu">length</span>(args)]</span>
<span id="cb292-19"><a href="supplementary-figure-13.html#cb292-19" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&#39; &#39;</span>)</span>
<span id="cb292-20"><a href="supplementary-figure-13.html#cb292-20" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">str_c</span>(crayon<span class="sc">::</span><span class="fu">green</span>(cli<span class="sc">::</span>symbol<span class="sc">$</span>star),<span class="st">&#39; &#39;</span>, <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(args),<span class="st">&#39;: &#39;</span>,crayon<span class="sc">::</span><span class="fu">green</span>(args),<span class="st">&#39;</span><span class="sc">\n</span><span class="st">&#39;</span>))</span>
<span id="cb292-21"><a href="supplementary-figure-13.html#cb292-21" aria-hidden="true" tabindex="-1"></a>cli<span class="sc">::</span><span class="fu">rule</span>(<span class="at">right =</span> <span class="fu">getwd</span>())</span></code></pre></div>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="supplementary-figure-13.html#cb293-1" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ── Script: R/fig/plot_SF13.R ────────────────────────────────────────────</span></span>
<span id="cb293-2"><a href="supplementary-figure-13.html#cb293-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Parameters read:</span></span>
<span id="cb293-3"><a href="supplementary-figure-13.html#cb293-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ★ 1: 2_analysis/admixture/</span></span>
<span id="cb293-4"><a href="supplementary-figure-13.html#cb293-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ★ 2: metadata/phenotypes.sc</span></span>
<span id="cb293-5"><a href="supplementary-figure-13.html#cb293-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ─────────────────────────────────────────── /current/working/directory ──</span></span></code></pre></div>
<p>The path containing the admixture results and the phenotype data file are received and stored inside more descriptive variables.</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="supplementary-figure-13.html#cb294-1" aria-hidden="true" tabindex="-1"></a><span class="co"># config -----------------------</span></span>
<span id="cb294-2"><a href="supplementary-figure-13.html#cb294-2" aria-hidden="true" tabindex="-1"></a>admx_path <span class="ot">&lt;-</span> <span class="fu">as.character</span>(args[<span class="dv">1</span>])</span>
<span id="cb294-3"><a href="supplementary-figure-13.html#cb294-3" aria-hidden="true" tabindex="-1"></a>pheno_file <span class="ot">&lt;-</span> <span class="fu">as.character</span>(args[<span class="dv">2</span>])</span></code></pre></div>
<p>Then, we extract the outlier region IDs from the admixture data file names for later use.</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="supplementary-figure-13.html#cb295-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load outlier window IDs (crop from admixture result file names)</span></span>
<span id="cb295-2"><a href="supplementary-figure-13.html#cb295-2" aria-hidden="true" tabindex="-1"></a>gids <span class="ot">&lt;-</span> <span class="fu">dir</span>(admx_path,</span>
<span id="cb295-3"><a href="supplementary-figure-13.html#cb295-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">pattern =</span> <span class="st">&quot;pop.*15.txt&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb295-4"><a href="supplementary-figure-13.html#cb295-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_remove</span>(<span class="st">&quot;pop.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb295-5"><a href="supplementary-figure-13.html#cb295-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_remove</span>(<span class="st">&quot;.15.txt&quot;</span>)</span></code></pre></div>
<p>We start the data import by loading the phenotype data.</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="supplementary-figure-13.html#cb296-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load phenotype data</span></span>
<span id="cb296-2"><a href="supplementary-figure-13.html#cb296-2" aria-hidden="true" tabindex="-1"></a>pheno_data <span class="ot">&lt;-</span> <span class="fu">read_sc</span>(pheno_file) <span class="sc">%&gt;%</span></span>
<span id="cb296-3"><a href="supplementary-figure-13.html#cb296-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(id, Bars, Peduncle, Snout) <span class="sc">%&gt;%</span></span>
<span id="cb296-4"><a href="supplementary-figure-13.html#cb296-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Bars))</span></code></pre></div>
<p>Then we load all the admixture results that used two clusters (<code>k = 2</code>).</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="supplementary-figure-13.html#cb297-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load admixture data</span></span>
<span id="cb297-2"><a href="supplementary-figure-13.html#cb297-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> gids <span class="sc">%&gt;%</span></span>
<span id="cb297-3"><a href="supplementary-figure-13.html#cb297-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map_dfr</span>(data_amdx, <span class="at">admx_path =</span> admx_path,</span>
<span id="cb297-4"><a href="supplementary-figure-13.html#cb297-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">k =</span> <span class="dv">2</span>)</span></code></pre></div>
<p>We want to indicate the phenotype of each trait under the outlier region that is expected to be most relevant for this trait.
Therefore we assign each trait to a specific outlier ID.</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="supplementary-figure-13.html#cb298-1" aria-hidden="true" tabindex="-1"></a><span class="co"># associate phenotypic trait with outlier region</span></span>
<span id="cb298-2"><a href="supplementary-figure-13.html#cb298-2" aria-hidden="true" tabindex="-1"></a>pheno_facet <span class="ot">&lt;-</span> <span class="fu">tibble</span>( <span class="at">trait =</span> <span class="fu">c</span>(<span class="st">&quot;Snout&quot;</span>,<span class="st">&quot;Bars&quot;</span>,  <span class="st">&quot;Peduncle&quot;</span>),</span>
<span id="cb298-3"><a href="supplementary-figure-13.html#cb298-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">gid =</span> <span class="fu">c</span>(<span class="st">&quot;LG04_1&quot;</span>, <span class="st">&quot;LG12_3&quot;</span>, <span class="st">&quot;LG12_4&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb298-4"><a href="supplementary-figure-13.html#cb298-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">facet_label =</span> <span class="fu">str_c</span>(gid, <span class="st">&quot; / &quot;</span>, trait))</span></code></pre></div>
<p>Then we format the outlier ID labels for the final plot.</p>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="supplementary-figure-13.html#cb299-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set outlier region labels</span></span>
<span id="cb299-2"><a href="supplementary-figure-13.html#cb299-2" aria-hidden="true" tabindex="-1"></a>gid_labels <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="at">LG04_1 =</span> <span class="st">&quot;LG04 (A)&quot;</span>,</span>
<span id="cb299-3"><a href="supplementary-figure-13.html#cb299-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">LG12_3 =</span> <span class="st">&quot;LG12 (B)&quot;</span>,</span>
<span id="cb299-4"><a href="supplementary-figure-13.html#cb299-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">LG12_4 =</span> <span class="st">&quot;LG12 (C)&quot;</span>)</span></code></pre></div>
<p>To due the historic development of this script, there is a redundant assignment of outlier ID and trait (both are needed within functions of <strong>GenomicOriginsScripts</strong> 🤷).</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="supplementary-figure-13.html#cb300-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set outlier region phenotypic traits</span></span>
<span id="cb300-2"><a href="supplementary-figure-13.html#cb300-2" aria-hidden="true" tabindex="-1"></a>gid_traits <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="at">LG04_1 =</span> <span class="st">&quot;Snout&quot;</span>,</span>
<span id="cb300-3"><a href="supplementary-figure-13.html#cb300-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">LG12_3 =</span> <span class="st">&quot;Bars&quot;</span>,</span>
<span id="cb300-4"><a href="supplementary-figure-13.html#cb300-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">LG12_4 =</span> <span class="st">&quot;Peduncle&quot;</span>)</span></code></pre></div>
<p>Using the <code>markdown</code> functionality of <strong>ggtext</strong>, we store each trait image in the axis labels using <code>html</code> syntax.</p>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="supplementary-figure-13.html#cb301-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set path to trait images</span></span>
<span id="cb301-2"><a href="supplementary-figure-13.html#cb301-2" aria-hidden="true" tabindex="-1"></a>trait_icons <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">LG04_1 =</span> <span class="st">&quot;&lt;img src=&#39;ressources/img/snout_c.png&#39; width=&#39;60&#39; /&gt;   &quot;</span>,</span>
<span id="cb301-3"><a href="supplementary-figure-13.html#cb301-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">LG12_3 =</span> <span class="st">&quot;&lt;img src=&#39;ressources/img/bars_c.png&#39; width=&#39;60&#39; /&gt;    &quot;</span>,</span>
<span id="cb301-4"><a href="supplementary-figure-13.html#cb301-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">LG12_4 =</span> <span class="st">&quot;&lt;img src=&#39;ressources/img/peduncle_c.png&#39; width=&#39;60&#39; /&gt;    &quot;</span>)</span></code></pre></div>
<p>Then, we clean and format the phenotype data for plotting.</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="supplementary-figure-13.html#cb302-1" aria-hidden="true" tabindex="-1"></a><span class="co"># format phenotype data</span></span>
<span id="cb302-2"><a href="supplementary-figure-13.html#cb302-2" aria-hidden="true" tabindex="-1"></a>pheno_plot_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb302-3"><a href="supplementary-figure-13.html#cb302-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(id)) <span class="sc">%&gt;%</span></span>
<span id="cb302-4"><a href="supplementary-figure-13.html#cb302-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(id<span class="sc">:</span>id_order) <span class="sc">%&gt;%</span></span>
<span id="cb302-5"><a href="supplementary-figure-13.html#cb302-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(pheno_data,<span class="at">by =</span> <span class="fu">c</span>( <span class="at">id_nr =</span> <span class="st">&quot;id&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb302-6"><a href="supplementary-figure-13.html#cb302-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(spec, Bars, Peduncle, Snout, id) <span class="sc">%&gt;%</span></span>
<span id="cb302-7"><a href="supplementary-figure-13.html#cb302-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ord_nr =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb302-8"><a href="supplementary-figure-13.html#cb302-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">&quot;trait&quot;</span>,</span>
<span id="cb302-9"><a href="supplementary-figure-13.html#cb302-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">&quot;phenotype&quot;</span>,</span>
<span id="cb302-10"><a href="supplementary-figure-13.html#cb302-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">cols =</span> Bars<span class="sc">:</span>Snout) <span class="sc">%&gt;%</span></span>
<span id="cb302-11"><a href="supplementary-figure-13.html#cb302-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(pheno_facet)</span></code></pre></div>
<p>The separate panels of the figure are actually individual plots.
To assure a consistent sample order, this is defined externally and stored within <code>sample_order</code>.</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="supplementary-figure-13.html#cb303-1" aria-hidden="true" tabindex="-1"></a><span class="co"># helper for consistent sample order across all panels</span></span>
<span id="cb303-2"><a href="supplementary-figure-13.html#cb303-2" aria-hidden="true" tabindex="-1"></a>sample_order <span class="ot">&lt;-</span> pheno_plot_data <span class="sc">%&gt;%</span></span>
<span id="cb303-3"><a href="supplementary-figure-13.html#cb303-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">duplicated</span>(id)) <span class="sc">%&gt;%</span></span>
<span id="cb303-4"><a href="supplementary-figure-13.html#cb303-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(id, ord_nr)</span></code></pre></div>
<p>Then, the admixture panels (a - c) are created as three sepearted plots and stored within a list.</p>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="supplementary-figure-13.html#cb304-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create plot panels a-c</span></span>
<span id="cb304-2"><a href="supplementary-figure-13.html#cb304-2" aria-hidden="true" tabindex="-1"></a>p_ad <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;LG04_1&quot;</span>, <span class="st">&quot;LG12_3&quot;</span>, <span class="st">&quot;LG12_4&quot;</span>) <span class="sc">%&gt;%</span> purrr<span class="sc">::</span><span class="fu">map</span>(adm_plot, <span class="at">data =</span> data)</span></code></pre></div>
<p>(Just as an example - here is the plot for panel c)</p>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="supplementary-figure-13.html#cb305-1" aria-hidden="true" tabindex="-1"></a>p_ad[[<span class="dv">3</span>]] <span class="co"># not part of the actual script</span></span></code></pre></div>
<p>Next, we create a dummy plot to extract the legend for the phenotypes.</p>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="supplementary-figure-13.html#cb306-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create dummy plot for the phenotype legend</span></span>
<span id="cb306-2"><a href="supplementary-figure-13.html#cb306-2" aria-hidden="true" tabindex="-1"></a>p_phno <span class="ot">&lt;-</span> pheno_plot_data <span class="sc">%&gt;%</span></span>
<span id="cb306-3"><a href="supplementary-figure-13.html#cb306-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> ord_nr))<span class="sc">+</span></span>
<span id="cb306-4"><a href="supplementary-figure-13.html#cb306-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> trait, <span class="at">fill =</span> <span class="fu">factor</span>(phenotype)),<span class="at">shape =</span> <span class="dv">21</span>)<span class="sc">+</span></span>
<span id="cb306-5"><a href="supplementary-figure-13.html#cb306-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="st">&quot;Phenotype&lt;br&gt;&lt;img src=&#39;ressources/img/all_traits_c.png&#39; width=&#39;110&#39; /&gt;&quot;</span>,</span>
<span id="cb306-6"><a href="supplementary-figure-13.html#cb306-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">0</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;white&quot;</span>, <span class="st">`</span><span class="at">1</span><span class="st">`</span> <span class="ot">=</span> <span class="st">&quot;black&quot;</span>),</span>
<span id="cb306-7"><a href="supplementary-figure-13.html#cb306-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">na.value =</span> <span class="st">&quot;gray&quot;</span>,</span>
<span id="cb306-8"><a href="supplementary-figure-13.html#cb306-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;absent&quot;</span>, <span class="st">&quot;present&quot;</span>, <span class="st">&quot;not scored&quot;</span>))<span class="sc">+</span></span>
<span id="cb306-9"><a href="supplementary-figure-13.html#cb306-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">ncol =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb306-10"><a href="supplementary-figure-13.html#cb306-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()<span class="sc">+</span></span>
<span id="cb306-11"><a href="supplementary-figure-13.html#cb306-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_markdown</span>(<span class="at">hjust =</span> .<span class="dv">5</span>),</span>
<span id="cb306-12"><a href="supplementary-figure-13.html#cb306-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p>Below the panels a - c the species of the samples are indicated.
To do this, we prepare a small table from the species affiliation of each sample.</p>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="supplementary-figure-13.html#cb307-1" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare table with fish annotations for the species indication</span></span>
<span id="cb307-2"><a href="supplementary-figure-13.html#cb307-2" aria-hidden="true" tabindex="-1"></a>tib_drawing <span class="ot">&lt;-</span> pheno_plot_data <span class="sc">%&gt;%</span></span>
<span id="cb307-3"><a href="supplementary-figure-13.html#cb307-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(spec) <span class="sc">%&gt;%</span></span>
<span id="cb307-4"><a href="supplementary-figure-13.html#cb307-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">pos =</span> (<span class="fu">min</span>(ord_nr)<span class="sc">+</span><span class="fu">max</span>(ord_nr))<span class="sc">*</span>.<span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb307-5"><a href="supplementary-figure-13.html#cb307-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code></pre></div>
<p>This table is then used to create a small plot.</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="supplementary-figure-13.html#cb308-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create sub-plot for species indication</span></span>
<span id="cb308-2"><a href="supplementary-figure-13.html#cb308-2" aria-hidden="true" tabindex="-1"></a>p_spec <span class="ot">&lt;-</span> pheno_plot_data <span class="sc">%&gt;%</span></span>
<span id="cb308-3"><a href="supplementary-figure-13.html#cb308-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(spec) <span class="sc">%&gt;%</span></span>
<span id="cb308-4"><a href="supplementary-figure-13.html#cb308-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">start =</span> <span class="fu">min</span>(ord_nr)<span class="sc">-</span><span class="dv">1</span>,</span>
<span id="cb308-5"><a href="supplementary-figure-13.html#cb308-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">end =</span> <span class="fu">max</span>(ord_nr)) <span class="sc">%&gt;%</span></span>
<span id="cb308-6"><a href="supplementary-figure-13.html#cb308-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">xmin =</span> start, <span class="at">xmax =</span> end,</span>
<span id="cb308-7"><a href="supplementary-figure-13.html#cb308-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">ymin =</span> <span class="sc">-</span><span class="cn">Inf</span>,</span>
<span id="cb308-8"><a href="supplementary-figure-13.html#cb308-8" aria-hidden="true" tabindex="-1"></a>             <span class="at">ymax =</span> <span class="cn">Inf</span>))<span class="sc">+</span></span>
<span id="cb308-9"><a href="supplementary-figure-13.html#cb308-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add colored backgroud boxes</span></span>
<span id="cb308-10"><a href="supplementary-figure-13.html#cb308-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_rect</span>(<span class="fu">aes</span>(<span class="at">fill =</span> spec), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)<span class="sc">+</span></span>
<span id="cb308-11"><a href="supplementary-figure-13.html#cb308-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add fish images</span></span>
<span id="cb308-12"><a href="supplementary-figure-13.html#cb308-12" aria-hidden="true" tabindex="-1"></a>  (tib_drawing <span class="sc">%&gt;%</span> <span class="fu">pmap</span>(add_spec_drawing))<span class="sc">+</span></span>
<span id="cb308-13"><a href="supplementary-figure-13.html#cb308-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set axis layout</span></span>
<span id="cb308-14"><a href="supplementary-figure-13.html#cb308-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> .<span class="dv">5</span>, <span class="at">labels =</span> <span class="fu">c</span>( <span class="st">&quot;Species&quot;</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb308-15"><a href="supplementary-figure-13.html#cb308-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">breaks =</span> sample_order<span class="sc">$</span>ord_nr,</span>
<span id="cb308-16"><a href="supplementary-figure-13.html#cb308-16" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> sample_order<span class="sc">$</span>id,</span>
<span id="cb308-17"><a href="supplementary-figure-13.html#cb308-17" aria-hidden="true" tabindex="-1"></a>                   <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb308-18"><a href="supplementary-figure-13.html#cb308-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set species color scheme</span></span>
<span id="cb308-19"><a href="supplementary-figure-13.html#cb308-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="st">&quot;Species&quot;</span>, <span class="at">values =</span> clr, <span class="at">labels =</span> sp_labs)<span class="sc">+</span></span>
<span id="cb308-20"><a href="supplementary-figure-13.html#cb308-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set general plot layout</span></span>
<span id="cb308-21"><a href="supplementary-figure-13.html#cb308-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()<span class="sc">+</span></span>
<span id="cb308-22"><a href="supplementary-figure-13.html#cb308-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>),</span>
<span id="cb308-23"><a href="supplementary-figure-13.html#cb308-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb308-24"><a href="supplementary-figure-13.html#cb308-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text.align =</span> <span class="dv">0</span>,</span>
<span id="cb308-25"><a href="supplementary-figure-13.html#cb308-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb308-26"><a href="supplementary-figure-13.html#cb308-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb308-27"><a href="supplementary-figure-13.html#cb308-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p>Similarly to the species affiliation, we create a second plot to indicate the sampling location.</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="supplementary-figure-13.html#cb309-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create sub-plot for samplong location indication</span></span>
<span id="cb309-2"><a href="supplementary-figure-13.html#cb309-2" aria-hidden="true" tabindex="-1"></a>p_loc <span class="ot">&lt;-</span> pheno_plot_data <span class="sc">%&gt;%</span></span>
<span id="cb309-3"><a href="supplementary-figure-13.html#cb309-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(ord_nr)))<span class="sc">+</span></span>
<span id="cb309-4"><a href="supplementary-figure-13.html#cb309-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add colored boxes</span></span>
<span id="cb309-5"><a href="supplementary-figure-13.html#cb309-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_raster</span>(<span class="fu">aes</span>(<span class="at">y =</span> <span class="dv">0</span>, <span class="at">fill =</span> loc))<span class="sc">+</span></span>
<span id="cb309-6"><a href="supplementary-figure-13.html#cb309-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set axis layout</span></span>
<span id="cb309-7"><a href="supplementary-figure-13.html#cb309-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>),<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Location&quot;</span>))<span class="sc">+</span></span>
<span id="cb309-8"><a href="supplementary-figure-13.html#cb309-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">breaks =</span> sample_order<span class="sc">$</span>ord_nr,</span>
<span id="cb309-9"><a href="supplementary-figure-13.html#cb309-9" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> sample_order<span class="sc">$</span>id) <span class="sc">+</span></span>
<span id="cb309-10"><a href="supplementary-figure-13.html#cb309-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set location color scheme</span></span>
<span id="cb309-11"><a href="supplementary-figure-13.html#cb309-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="st">&quot;Location&quot;</span>, <span class="at">values =</span>  clr_loc, loc_names)<span class="sc">+</span></span>
<span id="cb309-12"><a href="supplementary-figure-13.html#cb309-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set general plot layout</span></span>
<span id="cb309-13"><a href="supplementary-figure-13.html#cb309-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()<span class="sc">+</span></span>
<span id="cb309-14"><a href="supplementary-figure-13.html#cb309-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>),</span>
<span id="cb309-15"><a href="supplementary-figure-13.html#cb309-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb309-16"><a href="supplementary-figure-13.html#cb309-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb309-17"><a href="supplementary-figure-13.html#cb309-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb309-18"><a href="supplementary-figure-13.html#cb309-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p>Then, we compose the figure legend from individual sub-legends.</p>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="supplementary-figure-13.html#cb310-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compose legend from individual legend parts</span></span>
<span id="cb310-2"><a href="supplementary-figure-13.html#cb310-2" aria-hidden="true" tabindex="-1"></a>p_l <span class="ot">&lt;-</span> (<span class="fu">get_legend</span>(p_phno) <span class="sc">%&gt;%</span> <span class="fu">ggdraw</span>()) <span class="sc">+</span></span>
<span id="cb310-3"><a href="supplementary-figure-13.html#cb310-3" aria-hidden="true" tabindex="-1"></a>  (<span class="fu">get_legend</span>(p_spec) <span class="sc">%&gt;%</span> <span class="fu">ggdraw</span>()) <span class="sc">+</span></span>
<span id="cb310-4"><a href="supplementary-figure-13.html#cb310-4" aria-hidden="true" tabindex="-1"></a>  (<span class="fu">get_legend</span>(p_loc) <span class="sc">%&gt;%</span> <span class="fu">ggdraw</span>()) <span class="sc">+</span></span>
<span id="cb310-5"><a href="supplementary-figure-13.html#cb310-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">nrow =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>At this point, we can assemble the final figure.
Yet, due to the positioning of the trait images with <strong>ggtext</strong> we need to crop the left margin a little (hence we need <code>p_done</code>).</p>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="supplementary-figure-13.html#cb311-1" aria-hidden="true" tabindex="-1"></a><span class="co"># finalize figure</span></span>
<span id="cb311-2"><a href="supplementary-figure-13.html#cb311-2" aria-hidden="true" tabindex="-1"></a>p_prep <span class="ot">&lt;-</span> p_ad[[<span class="dv">1</span>]] <span class="sc">+</span></span>
<span id="cb311-3"><a href="supplementary-figure-13.html#cb311-3" aria-hidden="true" tabindex="-1"></a>  p_ad[[<span class="dv">2</span>]] <span class="sc">+</span></span>
<span id="cb311-4"><a href="supplementary-figure-13.html#cb311-4" aria-hidden="true" tabindex="-1"></a>  p_ad[[<span class="dv">3</span>]]<span class="sc">+</span></span>
<span id="cb311-5"><a href="supplementary-figure-13.html#cb311-5" aria-hidden="true" tabindex="-1"></a>  p_spec <span class="sc">+</span> p_loc <span class="sc">+</span> p_l <span class="sc">+</span></span>
<span id="cb311-6"><a href="supplementary-figure-13.html#cb311-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">heights =</span> <span class="fu">c</span>(.<span class="dv">4</span>,.<span class="dv">4</span>,.<span class="dv">4</span>,.<span class="dv">08</span>,.<span class="dv">02</span>,.<span class="dv">1</span>)) <span class="sc">&amp;</span></span>
<span id="cb311-7"><a href="supplementary-figure-13.html#cb311-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb311-8"><a href="supplementary-figure-13.html#cb311-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>))</span></code></pre></div>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="supplementary-figure-13.html#cb312-1" aria-hidden="true" tabindex="-1"></a><span class="co"># crop final figure (remove whitespace on left margin)</span></span>
<span id="cb312-2"><a href="supplementary-figure-13.html#cb312-2" aria-hidden="true" tabindex="-1"></a>p_done <span class="ot">&lt;-</span> <span class="fu">ggdraw</span>(p_prep, <span class="at">xlim =</span> <span class="fu">c</span>(.<span class="dv">023</span>,<span class="dv">1</span>))</span></code></pre></div>
<p>Finally, we can export Figure S13.</p>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="supplementary-figure-13.html#cb313-1" aria-hidden="true" tabindex="-1"></a><span class="co"># export final figure</span></span>
<span id="cb313-2"><a href="supplementary-figure-13.html#cb313-2" aria-hidden="true" tabindex="-1"></a>scl <span class="ot">&lt;-</span> .<span class="dv">9</span></span>
<span id="cb313-3"><a href="supplementary-figure-13.html#cb313-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;figures/SF13.pdf&quot;</span>,</span>
<span id="cb313-4"><a href="supplementary-figure-13.html#cb313-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> p_done,</span>
<span id="cb313-5"><a href="supplementary-figure-13.html#cb313-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">16</span><span class="sc">*</span>scl,</span>
<span id="cb313-6"><a href="supplementary-figure-13.html#cb313-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">10</span><span class="sc">*</span>scl,</span>
<span id="cb313-7"><a href="supplementary-figure-13.html#cb313-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">device =</span> cairo_pdf)</span></code></pre></div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="supplementary-figure-12.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="supplementary-figure-14.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
