<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>24 Supplementary Figure 7 | Script repository</title>
  <meta name="description" content="Scripts used to produce Figures and Supplementary Figures of â€˜Ancestral variation, hybridization and modularity fuel a marine radiationâ€™ by Hench, McMillan an Puebla" />
  <meta name="generator" content="bookdown 0.19 and GitBook 2.6.7" />

  <meta property="og:title" content="24 Supplementary Figure 7 | Script repository" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Scripts used to produce Figures and Supplementary Figures of â€˜Ancestral variation, hybridization and modularity fuel a marine radiationâ€™ by Hench, McMillan an Puebla" />
  <meta name="github-repo" content="k-hench/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="24 Supplementary Figure 7 | Script repository" />
  
  <meta name="twitter:description" content="Scripts used to produce Figures and Supplementary Figures of â€˜Ancestral variation, hybridization and modularity fuel a marine radiationâ€™ by Hench, McMillan an Puebla" />
  

<meta name="author" content="Kosmas Hench" />


<meta name="date" content="2020-06-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="supplementary-figure-6.html"/>
<link rel="next" href="supplementary-figure-8.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="libs/d3-5.7.0/d3.min.js"></script>
<script src="libs/d3-lasso-0.0.5/d3-lasso.min.js"></script>
<link href="libs/ggiraphjs-0.2.0/styles.css" rel="stylesheet" />
<script src="libs/ggiraphjs-0.2.0/ggiraphjs.min.js"></script>
<script src="libs/girafe-binding-0.7.0/girafe.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://github.com/k-hench/chapter2">github repository</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Intro</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#analysis"><i class="fa fa-check"></i><b>1.1</b> Analysis</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#prerequesites"><i class="fa fa-check"></i><b>1.2</b> Prerequesites</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#figures"><i class="fa fa-check"></i><b>1.3</b> Figures</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html"><i class="fa fa-check"></i><b>2</b> (git 1) Genotyping I (SNPs only)</a><ul>
<li class="chapter" data-level="2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#summary"><i class="fa fa-check"></i><b>2.1</b> Summary</a></li>
<li class="chapter" data-level="2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#details-of-genotyping.nf"><i class="fa fa-check"></i><b>2.2</b> Details of <code>genotyping.nf</code></a><ul>
<li class="chapter" data-level="2.2.1" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#data-preparation"><i class="fa fa-check"></i><b>2.2.1</b> Data preparation</a></li>
<li class="chapter" data-level="2.2.2" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#genotying"><i class="fa fa-check"></i><b>2.2.2</b> Genotying</a></li>
<li class="chapter" data-level="2.2.3" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#phasing"><i class="fa fa-check"></i><b>2.2.3</b> Phasing</a></li>
<li class="chapter" data-level="2.2.4" data-path="git-1-genotyping-i-snps-only.html"><a href="git-1-genotyping-i-snps-only.html#indel-masks"><i class="fa fa-check"></i><b>2.2.4</b> Indel masks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html"><i class="fa fa-check"></i><b>3</b> (git 2) Genotyping II (all callable sites)</a><ul>
<li class="chapter" data-level="3.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#summary-1"><i class="fa fa-check"></i><b>3.1</b> Summary</a></li>
<li class="chapter" data-level="3.2" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#details-of-genotyping_all_basepairs.nf"><i class="fa fa-check"></i><b>3.2</b> Details of <code>genotyping_all_basepairs.nf</code></a><ul>
<li class="chapter" data-level="3.2.1" data-path="git-2-genotyping-ii-all-callable-sites.html"><a href="git-2-genotyping-ii-all-callable-sites.html#data-preparation-1"><i class="fa fa-check"></i><b>3.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html"><i class="fa fa-check"></i><b>4</b> (git 3) Analysis I (<i>F<sub>ST</sub></i> &amp; GxP)</a><ul>
<li class="chapter" data-level="4.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#summary-2"><i class="fa fa-check"></i><b>4.1</b> Summary</a></li>
<li class="chapter" data-level="4.2" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#details-of-analysis_fst_gxp.nf"><i class="fa fa-check"></i><b>4.2</b> Details of <code>analysis_fst_gxp.nf</code></a><ul>
<li class="chapter" data-level="4.2.1" data-path="git-3-analysis-i-fst-gxp.html"><a href="git-3-analysis-i-fst-gxp.html#setup"><i class="fa fa-check"></i><b>4.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html"><i class="fa fa-check"></i><b>5</b> (git 4) Analysis II (<i>d<sub>XY</sub></i> &amp; pi)</a><ul>
<li class="chapter" data-level="5.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#summary-3"><i class="fa fa-check"></i><b>5.1</b> Summary</a></li>
<li class="chapter" data-level="5.2" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#details-of-analysis_dxy.nf"><i class="fa fa-check"></i><b>5.2</b> Details of <code>analysis_dxy.nf</code></a><ul>
<li class="chapter" data-level="5.2.1" data-path="git-4-analysis-ii-dxy-pi.html"><a href="git-4-analysis-ii-dxy-pi.html#data-preparation-2"><i class="fa fa-check"></i><b>5.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html"><i class="fa fa-check"></i><b>6</b> (git 5) Analysis III (phylogeny &amp; topology weighting)</a><ul>
<li class="chapter" data-level="6.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#summary-4"><i class="fa fa-check"></i><b>6.1</b> Summary</a></li>
<li class="chapter" data-level="6.2" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#details-of-analysis_fasttree_twisst.nf"><i class="fa fa-check"></i><b>6.2</b> Details of <code>analysis_fasttree_twisst.nf</code></a><ul>
<li class="chapter" data-level="6.2.1" data-path="git-5-analysis-iii-phylogeny-topology-weighting.html"><a href="git-5-analysis-iii-phylogeny-topology-weighting.html#setup-1"><i class="fa fa-check"></i><b>6.2.1</b> Setup</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html"><i class="fa fa-check"></i><b>7</b> (git 6) Analysis IV (rho)</a><ul>
<li class="chapter" data-level="7.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#summary-5"><i class="fa fa-check"></i><b>7.1</b> Summary</a></li>
<li class="chapter" data-level="7.2" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#details-of-analysis_recombination.nf"><i class="fa fa-check"></i><b>7.2</b> Details of <code>analysis_recombination.nf</code></a><ul>
<li class="chapter" data-level="7.2.1" data-path="git-6-analysis-iv-rho.html"><a href="git-6-analysis-iv-rho.html#data-preparation-3"><i class="fa fa-check"></i><b>7.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html"><i class="fa fa-check"></i><b>8</b> (git 7) Analysis V (poptrees)</a><ul>
<li class="chapter" data-level="8.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#summary-6"><i class="fa fa-check"></i><b>8.1</b> Summary</a></li>
<li class="chapter" data-level="8.2" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#details-of-analysis_fst_poptree.nf"><i class="fa fa-check"></i><b>8.2</b> Details of <code>analysis_fst_poptree.nf</code></a><ul>
<li class="chapter" data-level="8.2.1" data-path="git-7-analysis-v-poptrees.html"><a href="git-7-analysis-v-poptrees.html#data-preparation-4"><i class="fa fa-check"></i><b>8.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html"><i class="fa fa-check"></i><b>9</b> (git 8) Analysis VI (Demographic History)</a><ul>
<li class="chapter" data-level="9.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#summary-7"><i class="fa fa-check"></i><b>9.1</b> Summary</a></li>
<li class="chapter" data-level="9.2" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#details-of-analysis_msmc.nf"><i class="fa fa-check"></i><b>9.2</b> Details of <code>analysis_msmc.nf</code></a><ul>
<li class="chapter" data-level="9.2.1" data-path="git-8-analysis-vi-demographic-history.html"><a href="git-8-analysis-vi-demographic-history.html#data-preparation-5"><i class="fa fa-check"></i><b>9.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html"><i class="fa fa-check"></i><b>10</b> (git 9) Analysis VII (hybridization)</a><ul>
<li class="chapter" data-level="10.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#summary-8"><i class="fa fa-check"></i><b>10.1</b> Summary</a></li>
<li class="chapter" data-level="10.2" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#details-of-analysis_hybridization.nf"><i class="fa fa-check"></i><b>10.2</b> Details of <code>analysis_hybridization.nf</code></a><ul>
<li class="chapter" data-level="10.2.1" data-path="git-9-analysis-vii-hybridization.html"><a href="git-9-analysis-vii-hybridization.html#data-preparation-6"><i class="fa fa-check"></i><b>10.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html"><i class="fa fa-check"></i><b>11</b> (git 10) Analysis VIII (admixture)</a><ul>
<li class="chapter" data-level="11.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#summary-9"><i class="fa fa-check"></i><b>11.1</b> Summary</a></li>
<li class="chapter" data-level="11.2" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#details-of-analysis_admixture.nf"><i class="fa fa-check"></i><b>11.2</b> Details of <code>analysis_admixture.nf</code></a><ul>
<li class="chapter" data-level="11.2.1" data-path="git-10-analysis-viii-admixture.html"><a href="git-10-analysis-viii-admixture.html#data-preparation-7"><i class="fa fa-check"></i><b>11.2.1</b> Data preparation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="git-11-visualization.html"><a href="git-11-visualization.html"><i class="fa fa-check"></i><b>12</b> (git 11) Visualization</a></li>
<li class="chapter" data-level="13" data-path="figure-1.html"><a href="figure-1.html"><i class="fa fa-check"></i><b>13</b> Figure 1</a><ul>
<li class="chapter" data-level="13.1" data-path="figure-1.html"><a href="figure-1.html#summary-10"><i class="fa fa-check"></i><b>13.1</b> Summary</a></li>
<li class="chapter" data-level="13.2" data-path="figure-1.html"><a href="figure-1.html#details-of-plot_f1.r"><i class="fa fa-check"></i><b>13.2</b> Details of <code>plot_F1.R</code></a><ul>
<li class="chapter" data-level="13.2.1" data-path="figure-1.html"><a href="figure-1.html#config"><i class="fa fa-check"></i><b>13.2.1</b> Config</a></li>
<li class="chapter" data-level="13.2.2" data-path="figure-1.html"><a href="figure-1.html#data-import"><i class="fa fa-check"></i><b>13.2.2</b> Data import</a></li>
<li class="chapter" data-level="13.2.3" data-path="figure-1.html"><a href="figure-1.html#plotting"><i class="fa fa-check"></i><b>13.2.3</b> Plotting</a></li>
<li class="chapter" data-level="13.2.4" data-path="figure-1.html"><a href="figure-1.html#table-export"><i class="fa fa-check"></i><b>13.2.4</b> Table export</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="figure-2.html"><a href="figure-2.html"><i class="fa fa-check"></i><b>14</b> Figure 2</a><ul>
<li class="chapter" data-level="14.1" data-path="figure-2.html"><a href="figure-2.html#summary-11"><i class="fa fa-check"></i><b>14.1</b> Summary</a></li>
<li class="chapter" data-level="14.2" data-path="figure-2.html"><a href="figure-2.html#details-of-plot_f2.r"><i class="fa fa-check"></i><b>14.2</b> Details of <code>plot_F2.R</code></a><ul>
<li class="chapter" data-level="14.2.1" data-path="figure-2.html"><a href="figure-2.html#config-1"><i class="fa fa-check"></i><b>14.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="figure-3.html"><a href="figure-3.html"><i class="fa fa-check"></i><b>15</b> Figure 3</a><ul>
<li class="chapter" data-level="15.1" data-path="figure-3.html"><a href="figure-3.html#summary-12"><i class="fa fa-check"></i><b>15.1</b> Summary</a></li>
<li class="chapter" data-level="15.2" data-path="figure-3.html"><a href="figure-3.html#details-of-plot_f3.r"><i class="fa fa-check"></i><b>15.2</b> Details of <code>plot_F3.R</code></a><ul>
<li class="chapter" data-level="15.2.1" data-path="figure-3.html"><a href="figure-3.html#config-2"><i class="fa fa-check"></i><b>15.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="figure-4.html"><a href="figure-4.html"><i class="fa fa-check"></i><b>16</b> Figure 4</a><ul>
<li class="chapter" data-level="16.1" data-path="figure-4.html"><a href="figure-4.html#summary-13"><i class="fa fa-check"></i><b>16.1</b> Summary</a></li>
<li class="chapter" data-level="16.2" data-path="figure-4.html"><a href="figure-4.html#details-of-plot_f4.r"><i class="fa fa-check"></i><b>16.2</b> Details of <code>plot_F4.R</code></a><ul>
<li class="chapter" data-level="16.2.1" data-path="figure-4.html"><a href="figure-4.html#config-3"><i class="fa fa-check"></i><b>16.2.1</b> Config</a></li>
<li class="chapter" data-level="16.2.2" data-path="figure-4.html"><a href="figure-4.html#data-import-1"><i class="fa fa-check"></i><b>16.2.2</b> Data import</a></li>
<li class="chapter" data-level="16.2.3" data-path="figure-4.html"><a href="figure-4.html#plotting-1"><i class="fa fa-check"></i><b>16.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="figure-5.html"><a href="figure-5.html"><i class="fa fa-check"></i><b>17</b> Figure 5</a><ul>
<li class="chapter" data-level="17.1" data-path="figure-5.html"><a href="figure-5.html#summary-14"><i class="fa fa-check"></i><b>17.1</b> Summary</a></li>
<li class="chapter" data-level="17.2" data-path="figure-5.html"><a href="figure-5.html#details-of-plot_f5.r"><i class="fa fa-check"></i><b>17.2</b> Details of <code>plot_F5.R</code></a><ul>
<li class="chapter" data-level="17.2.1" data-path="figure-5.html"><a href="figure-5.html#config-4"><i class="fa fa-check"></i><b>17.2.1</b> Config</a></li>
<li class="chapter" data-level="17.2.2" data-path="figure-5.html"><a href="figure-5.html#data-import-2"><i class="fa fa-check"></i><b>17.2.2</b> Data import</a></li>
<li class="chapter" data-level="17.2.3" data-path="figure-5.html"><a href="figure-5.html#plotting-2"><i class="fa fa-check"></i><b>17.2.3</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="18" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html"><i class="fa fa-check"></i><b>18</b> Supplementary Figure 1</a><ul>
<li class="chapter" data-level="18.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#summary-15"><i class="fa fa-check"></i><b>18.1</b> Summary</a></li>
<li class="chapter" data-level="18.2" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#details-of-plot_sf1.r"><i class="fa fa-check"></i><b>18.2</b> Details of <code>plot_SF1.R</code></a><ul>
<li class="chapter" data-level="18.2.1" data-path="supplementary-figure-1.html"><a href="supplementary-figure-1.html#config-5"><i class="fa fa-check"></i><b>18.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="19" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html"><i class="fa fa-check"></i><b>19</b> Supplementary Figure 2</a><ul>
<li class="chapter" data-level="19.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#summary-16"><i class="fa fa-check"></i><b>19.1</b> Summary</a></li>
<li class="chapter" data-level="19.2" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#details-of-plot_sf2.r"><i class="fa fa-check"></i><b>19.2</b> Details of <code>plot_SF2.R</code></a><ul>
<li class="chapter" data-level="19.2.1" data-path="supplementary-figure-2.html"><a href="supplementary-figure-2.html#config-6"><i class="fa fa-check"></i><b>19.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="20" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html"><i class="fa fa-check"></i><b>20</b> Supplementary Figure 3</a><ul>
<li class="chapter" data-level="20.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#summary-17"><i class="fa fa-check"></i><b>20.1</b> Summary</a></li>
<li class="chapter" data-level="20.2" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#details-of-plot_sf3.r"><i class="fa fa-check"></i><b>20.2</b> Details of <code>plot_SF3.R</code></a><ul>
<li class="chapter" data-level="20.2.1" data-path="supplementary-figure-3.html"><a href="supplementary-figure-3.html#config-7"><i class="fa fa-check"></i><b>20.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="21" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html"><i class="fa fa-check"></i><b>21</b> Supplementary Figure 4</a><ul>
<li class="chapter" data-level="21.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#summary-18"><i class="fa fa-check"></i><b>21.1</b> Summary</a></li>
<li class="chapter" data-level="21.2" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#details-of-plot_sf4.r"><i class="fa fa-check"></i><b>21.2</b> Details of <code>plot_SF4.R</code></a><ul>
<li class="chapter" data-level="21.2.1" data-path="supplementary-figure-4.html"><a href="supplementary-figure-4.html#config-8"><i class="fa fa-check"></i><b>21.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="22" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html"><i class="fa fa-check"></i><b>22</b> Supplementary Figure 5</a><ul>
<li class="chapter" data-level="22.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#summary-19"><i class="fa fa-check"></i><b>22.1</b> Summary</a></li>
<li class="chapter" data-level="22.2" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#details-of-plot_sf5.r"><i class="fa fa-check"></i><b>22.2</b> Details of <code>plot_SF5.R</code></a><ul>
<li class="chapter" data-level="22.2.1" data-path="supplementary-figure-5.html"><a href="supplementary-figure-5.html#config-9"><i class="fa fa-check"></i><b>22.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="23" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html"><i class="fa fa-check"></i><b>23</b> Supplementary Figure 6</a><ul>
<li class="chapter" data-level="23.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#summary-20"><i class="fa fa-check"></i><b>23.1</b> Summary</a></li>
<li class="chapter" data-level="23.2" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#details-of-plot_sf6.r"><i class="fa fa-check"></i><b>23.2</b> Details of <code>plot_SF6.R</code></a><ul>
<li class="chapter" data-level="23.2.1" data-path="supplementary-figure-6.html"><a href="supplementary-figure-6.html#config-10"><i class="fa fa-check"></i><b>23.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="24" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html"><i class="fa fa-check"></i><b>24</b> Supplementary Figure 7</a><ul>
<li class="chapter" data-level="24.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#summary-21"><i class="fa fa-check"></i><b>24.1</b> Summary</a></li>
<li class="chapter" data-level="24.2" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#details-of-plot_sf7.r"><i class="fa fa-check"></i><b>24.2</b> Details of <code>plot_SF7.R</code></a><ul>
<li class="chapter" data-level="24.2.1" data-path="supplementary-figure-7.html"><a href="supplementary-figure-7.html#config-11"><i class="fa fa-check"></i><b>24.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="25" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html"><i class="fa fa-check"></i><b>25</b> Supplementary Figure 8</a><ul>
<li class="chapter" data-level="25.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#summary-22"><i class="fa fa-check"></i><b>25.1</b> Summary</a></li>
<li class="chapter" data-level="25.2" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#details-of-plot_sf8.r"><i class="fa fa-check"></i><b>25.2</b> Details of <code>plot_SF8.R</code></a><ul>
<li class="chapter" data-level="25.2.1" data-path="supplementary-figure-8.html"><a href="supplementary-figure-8.html#config-12"><i class="fa fa-check"></i><b>25.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="26" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html"><i class="fa fa-check"></i><b>26</b> Supplementary Figure 9</a><ul>
<li class="chapter" data-level="26.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#summary-23"><i class="fa fa-check"></i><b>26.1</b> Summary</a></li>
<li class="chapter" data-level="26.2" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#details-of-plot_sf9.r"><i class="fa fa-check"></i><b>26.2</b> Details of <code>plot_SF9.R</code></a><ul>
<li class="chapter" data-level="26.2.1" data-path="supplementary-figure-9.html"><a href="supplementary-figure-9.html#config-13"><i class="fa fa-check"></i><b>26.2.1</b> Config</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
<li> <image src="logo.svg" alt="Smiley face" style="width:100%; height:100px;"> </li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Script repository</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="supplementary-figure-7" class="section level1">
<h1><span class="header-section-number">24</span> Supplementary Figure 7</h1>
<div id="summary-21" class="section level2">
<h2><span class="header-section-number">24.1</span> Summary</h2>
<p>This is the accessory documentation of Supplementary Figure 7.
The Figure can be recreated by running the <strong>R</strong> script <code>plot_SF7.R</code>:</p>
<div class="sourceCode" id="cb180"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb180-1"><a href="supplementary-figure-7.html#cb180-1"></a><span class="bu">cd</span> <span class="va">$BASE_DIR</span></span>
<span id="cb180-2"><a href="supplementary-figure-7.html#cb180-2"></a></span>
<span id="cb180-3"><a href="supplementary-figure-7.html#cb180-3"></a><span class="ex">Rscript</span> --vanilla R/fig/plot_SF7.R 2_analysis/admixture/ metadata/phenotypes.sc</span></code></pre></div>
</div>
<div id="details-of-plot_sf7.r" class="section level2">
<h2><span class="header-section-number">24.2</span> Details of <code>plot_SF7.R</code></h2>
<p>In the following, the individual steps of the R script are documented.
It is an executable R script that depends on the accessory R package <a href="https://k-hench.github.io/GenomicOriginsScripts"><strong>GenomicOriginsScripts</strong></a>, as well as on the packages <a href="https://k-hench.github.io/hypoimg"><strong>hypoimg</strong></a>, <a href="https://wilkelab.org/ggtext/"><strong>ggtext</strong></a>, <a href="https://patchwork.data-imaginist.com/"><strong>patchwork</strong></a> and <a href="https://emilhvitfeldt.github.io/paletteer/"><strong>paletteer</strong></a>.</p>
<div id="config-11" class="section level3">
<h3><span class="header-section-number">24.2.1</span> Config</h3>
<p>The scripts start with a header that contains copy &amp; paste templates to execute or debug the script:</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="supplementary-figure-7.html#cb181-1"></a><span class="co">#!/usr/bin/env Rscript</span></span>
<span id="cb181-2"><a href="supplementary-figure-7.html#cb181-2"></a><span class="co"># run from terminal:</span></span>
<span id="cb181-3"><a href="supplementary-figure-7.html#cb181-3"></a><span class="co"># Rscript --vanilla R/fig/plot_SF7.R 2_analysis/admixture/ metadata/phenotypes.sc</span></span>
<span id="cb181-4"><a href="supplementary-figure-7.html#cb181-4"></a><span class="co"># ===============================================================</span></span>
<span id="cb181-5"><a href="supplementary-figure-7.html#cb181-5"></a><span class="co"># This script produces Suppl. Figure 7 of the study &quot;Ancestral variation,</span></span>
<span id="cb181-6"><a href="supplementary-figure-7.html#cb181-6"></a><span class="co"># hybridization and modularity fuel a marine radiation&quot;</span></span>
<span id="cb181-7"><a href="supplementary-figure-7.html#cb181-7"></a><span class="co"># by Hench, McMillan and Puebla</span></span>
<span id="cb181-8"><a href="supplementary-figure-7.html#cb181-8"></a><span class="co"># ---------------------------------------------------------------</span></span>
<span id="cb181-9"><a href="supplementary-figure-7.html#cb181-9"></a><span class="co"># ===============================================================</span></span>
<span id="cb181-10"><a href="supplementary-figure-7.html#cb181-10"></a><span class="co"># args &lt;- c( &quot;2_analysis/admixture/&quot;, &quot;metadata/phenotypes.sc&quot;)</span></span>
<span id="cb181-11"><a href="supplementary-figure-7.html#cb181-11"></a><span class="co"># script_name &lt;- &quot;R/fig/plot_SF7.R&quot;</span></span></code></pre></div>
<p>The next section processes the input from the command line.
It stores the arguments in the vector <code>args</code>.
The needed R packages are and the script name and the current working directory are stored inside variables (<code>script_name</code>, <code>plot_comment</code>).
This information will later be written into the meta data of the figure to help us tracing back the scripts that created the figures in the future.</p>
<p>Then we drop all the imported information besides the arguments following the script name and print the information to the terminal.</p>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="supplementary-figure-7.html#cb182-1"></a>args &lt;-<span class="st"> </span><span class="kw">commandArgs</span>(<span class="dt">trailingOnly=</span><span class="ot">FALSE</span>)</span>
<span id="cb182-2"><a href="supplementary-figure-7.html#cb182-2"></a><span class="co"># setup -----------------------</span></span>
<span id="cb182-3"><a href="supplementary-figure-7.html#cb182-3"></a><span class="kw">library</span>(paletteer)</span>
<span id="cb182-4"><a href="supplementary-figure-7.html#cb182-4"></a><span class="kw">library</span>(patchwork)</span>
<span id="cb182-5"><a href="supplementary-figure-7.html#cb182-5"></a><span class="kw">library</span>(GenomicOriginsScripts)</span>
<span id="cb182-6"><a href="supplementary-figure-7.html#cb182-6"></a><span class="kw">library</span>(hypoimg)</span>
<span id="cb182-7"><a href="supplementary-figure-7.html#cb182-7"></a><span class="kw">library</span>(ggtext)</span>
<span id="cb182-8"><a href="supplementary-figure-7.html#cb182-8"></a></span>
<span id="cb182-9"><a href="supplementary-figure-7.html#cb182-9"></a><span class="kw">cat</span>(<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>)</span>
<span id="cb182-10"><a href="supplementary-figure-7.html#cb182-10"></a>script_name &lt;-<span class="st"> </span>args[<span class="dv">5</span>] <span class="op">%&gt;%</span></span>
<span id="cb182-11"><a href="supplementary-figure-7.html#cb182-11"></a><span class="st">  </span><span class="kw">str_remove</span>(.,<span class="st">&#39;--file=&#39;</span>)</span>
<span id="cb182-12"><a href="supplementary-figure-7.html#cb182-12"></a></span>
<span id="cb182-13"><a href="supplementary-figure-7.html#cb182-13"></a>plot_comment &lt;-<span class="st"> </span>script_name <span class="op">%&gt;%</span></span>
<span id="cb182-14"><a href="supplementary-figure-7.html#cb182-14"></a><span class="st">  </span><span class="kw">str_c</span>(<span class="st">&#39;mother-script = &#39;</span>,<span class="kw">getwd</span>(),<span class="st">&#39;/&#39;</span>,.)</span>
<span id="cb182-15"><a href="supplementary-figure-7.html#cb182-15"></a></span>
<span id="cb182-16"><a href="supplementary-figure-7.html#cb182-16"></a>cli<span class="op">::</span><span class="kw">rule</span>( <span class="dt">left =</span> <span class="kw">str_c</span>(crayon<span class="op">::</span><span class="kw">bold</span>(<span class="st">&#39;Script: &#39;</span>),crayon<span class="op">::</span><span class="kw">red</span>(script_name)))</span>
<span id="cb182-17"><a href="supplementary-figure-7.html#cb182-17"></a>args =<span class="st"> </span>args[<span class="dv">7</span><span class="op">:</span><span class="kw">length</span>(args)]</span>
<span id="cb182-18"><a href="supplementary-figure-7.html#cb182-18"></a><span class="kw">cat</span>(<span class="st">&#39; &#39;</span>)</span>
<span id="cb182-19"><a href="supplementary-figure-7.html#cb182-19"></a><span class="kw">cat</span>(<span class="kw">str_c</span>(crayon<span class="op">::</span><span class="kw">green</span>(cli<span class="op">::</span>symbol<span class="op">$</span>star),<span class="st">&#39; &#39;</span>, <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(args),<span class="st">&#39;: &#39;</span>,crayon<span class="op">::</span><span class="kw">green</span>(args),<span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>))</span>
<span id="cb182-20"><a href="supplementary-figure-7.html#cb182-20"></a>cli<span class="op">::</span><span class="kw">rule</span>(<span class="dt">right =</span> <span class="kw">getwd</span>())</span></code></pre></div>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="supplementary-figure-7.html#cb183-1"></a><span class="co">#&gt; â”€â”€ Script: scripts/plot_SF7.R â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</span></span>
<span id="cb183-2"><a href="supplementary-figure-7.html#cb183-2"></a><span class="co">#&gt; Parameters read:</span></span>
<span id="cb183-3"><a href="supplementary-figure-7.html#cb183-3"></a><span class="co">#&gt; â˜… 1: 2_analysis/admixture/</span></span>
<span id="cb183-4"><a href="supplementary-figure-7.html#cb183-4"></a><span class="co">#&gt; â˜… 2: metadata/phenotypes.sc</span></span>
<span id="cb183-5"><a href="supplementary-figure-7.html#cb183-5"></a><span class="co">#&gt; â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€ /current/working/directory â”€â”€</span></span></code></pre></div>
<p>The path containing the admixture results and the phenotype data file are received and stored inside more descriptive variables.</p>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="supplementary-figure-7.html#cb184-1"></a><span class="co"># config -----------------------</span></span>
<span id="cb184-2"><a href="supplementary-figure-7.html#cb184-2"></a>admx_path &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">1</span>])</span>
<span id="cb184-3"><a href="supplementary-figure-7.html#cb184-3"></a>pheno_file &lt;-<span class="st"> </span><span class="kw">as.character</span>(args[<span class="dv">2</span>])</span></code></pre></div>
<p>Then, we extract the outlier region IDs from the admixture data file names for later use.</p>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="supplementary-figure-7.html#cb185-1"></a><span class="co"># load outlier window IDs (crop from admixture result file names)</span></span>
<span id="cb185-2"><a href="supplementary-figure-7.html#cb185-2"></a>gids &lt;-<span class="st"> </span><span class="kw">dir</span>(admx_path,</span>
<span id="cb185-3"><a href="supplementary-figure-7.html#cb185-3"></a>            <span class="dt">pattern =</span> <span class="st">&quot;pop.*15.txt&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb185-4"><a href="supplementary-figure-7.html#cb185-4"></a><span class="st">  </span><span class="kw">str_remove</span>(<span class="st">&quot;pop.&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb185-5"><a href="supplementary-figure-7.html#cb185-5"></a><span class="st">  </span><span class="kw">str_remove</span>(<span class="st">&quot;.15.txt&quot;</span>)</span></code></pre></div>
<p>We start the data import by loading the phenotype data.</p>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="supplementary-figure-7.html#cb186-1"></a><span class="co"># load phenotype data</span></span>
<span id="cb186-2"><a href="supplementary-figure-7.html#cb186-2"></a>pheno_data &lt;-<span class="st"> </span><span class="kw">read_sc</span>(pheno_file) <span class="op">%&gt;%</span></span>
<span id="cb186-3"><a href="supplementary-figure-7.html#cb186-3"></a><span class="st">  </span><span class="kw">select</span>(id, Bars, Peduncle, Snout) <span class="op">%&gt;%</span></span>
<span id="cb186-4"><a href="supplementary-figure-7.html#cb186-4"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(Bars))</span></code></pre></div>
<p>Then we load all the admixture results that used two clusters (<code>k = 2</code>).</p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="supplementary-figure-7.html#cb187-1"></a><span class="co"># load admixture data</span></span>
<span id="cb187-2"><a href="supplementary-figure-7.html#cb187-2"></a>data &lt;-<span class="st"> </span>gids <span class="op">%&gt;%</span></span>
<span id="cb187-3"><a href="supplementary-figure-7.html#cb187-3"></a><span class="st">  </span><span class="kw">map_dfr</span>(data_amdx, <span class="dt">admx_path =</span> admx_path,</span>
<span id="cb187-4"><a href="supplementary-figure-7.html#cb187-4"></a>          <span class="dt">k =</span> <span class="dv">2</span>)</span></code></pre></div>
<p>We want to indicate the phenotype of each trait under the outlier region that is expected to be most relevant for this trait.
Therefore we assign each trait to a specific outlier ID.</p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="supplementary-figure-7.html#cb188-1"></a><span class="co"># associate phenotypic trait with outlier region</span></span>
<span id="cb188-2"><a href="supplementary-figure-7.html#cb188-2"></a>pheno_facet &lt;-<span class="st"> </span><span class="kw">tibble</span>( <span class="dt">trait =</span> <span class="kw">c</span>(<span class="st">&quot;Snout&quot;</span>,<span class="st">&quot;Bars&quot;</span>,  <span class="st">&quot;Peduncle&quot;</span>),</span>
<span id="cb188-3"><a href="supplementary-figure-7.html#cb188-3"></a>                       <span class="dt">gid =</span> <span class="kw">c</span>(<span class="st">&quot;LG04_1&quot;</span>, <span class="st">&quot;LG12_3&quot;</span>, <span class="st">&quot;LG12_4&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb188-4"><a href="supplementary-figure-7.html#cb188-4"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">facet_label =</span> <span class="kw">str_c</span>(gid, <span class="st">&quot; / &quot;</span>, trait))</span></code></pre></div>
<p>Then we format the outlier ID labels for the final plot.</p>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="supplementary-figure-7.html#cb189-1"></a><span class="co"># set outlier region labels</span></span>
<span id="cb189-2"><a href="supplementary-figure-7.html#cb189-2"></a>gid_labels &lt;-<span class="st">  </span><span class="kw">c</span>(<span class="dt">LG04_1 =</span> <span class="st">&quot;LG04 (A)&quot;</span>,</span>
<span id="cb189-3"><a href="supplementary-figure-7.html#cb189-3"></a>                 <span class="dt">LG12_3 =</span> <span class="st">&quot;LG12 (B)&quot;</span>,</span>
<span id="cb189-4"><a href="supplementary-figure-7.html#cb189-4"></a>                 <span class="dt">LG12_4 =</span> <span class="st">&quot;LG12 (C)&quot;</span>)</span></code></pre></div>
<p>To due the historic development of this script, there is a redundant assignment of outlier ID and trait (both are needed within functions of <strong>GenomicOriginsScripts</strong> ðŸ¤·).</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="supplementary-figure-7.html#cb190-1"></a><span class="co"># set outlier region phenotypic traits</span></span>
<span id="cb190-2"><a href="supplementary-figure-7.html#cb190-2"></a>gid_traits &lt;-<span class="st">  </span><span class="kw">c</span>(<span class="dt">LG04_1 =</span> <span class="st">&quot;Snout&quot;</span>,</span>
<span id="cb190-3"><a href="supplementary-figure-7.html#cb190-3"></a>                 <span class="dt">LG12_3 =</span> <span class="st">&quot;Bars&quot;</span>,</span>
<span id="cb190-4"><a href="supplementary-figure-7.html#cb190-4"></a>                 <span class="dt">LG12_4 =</span> <span class="st">&quot;Peduncle&quot;</span>)</span></code></pre></div>
<p>Using the <code>markdown</code> functionality of <strong>ggtext</strong>, we store each trait image in the axis labels using <code>html</code> syntax.</p>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="supplementary-figure-7.html#cb191-1"></a><span class="co"># set path to trait images</span></span>
<span id="cb191-2"><a href="supplementary-figure-7.html#cb191-2"></a>trait_icons &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dt">LG04_1 =</span> <span class="st">&quot;&lt;img src=&#39;ressources/img/snout_c.png&#39; width=&#39;60&#39; /&gt;   &quot;</span>,</span>
<span id="cb191-3"><a href="supplementary-figure-7.html#cb191-3"></a>               <span class="dt">LG12_3 =</span> <span class="st">&quot;&lt;img src=&#39;ressources/img/bars_c.png&#39; width=&#39;60&#39; /&gt;    &quot;</span>,</span>
<span id="cb191-4"><a href="supplementary-figure-7.html#cb191-4"></a>               <span class="dt">LG12_4 =</span> <span class="st">&quot;&lt;img src=&#39;ressources/img/peduncle_c.png&#39; width=&#39;60&#39; /&gt;    &quot;</span>)</span></code></pre></div>
<p>Then we clean and format the phenotype data for plotting.</p>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="supplementary-figure-7.html#cb192-1"></a><span class="co"># format phenotype data</span></span>
<span id="cb192-2"><a href="supplementary-figure-7.html#cb192-2"></a>pheno_plot_data &lt;-<span class="st"> </span>data <span class="op">%&gt;%</span></span>
<span id="cb192-3"><a href="supplementary-figure-7.html#cb192-3"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">duplicated</span>(id)) <span class="op">%&gt;%</span></span>
<span id="cb192-4"><a href="supplementary-figure-7.html#cb192-4"></a><span class="st">  </span><span class="kw">select</span>(id<span class="op">:</span>id_order) <span class="op">%&gt;%</span></span>
<span id="cb192-5"><a href="supplementary-figure-7.html#cb192-5"></a><span class="st">  </span><span class="kw">left_join</span>(pheno_data,<span class="dt">by =</span> <span class="kw">c</span>( <span class="dt">id_nr =</span> <span class="st">&quot;id&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb192-6"><a href="supplementary-figure-7.html#cb192-6"></a><span class="st">  </span><span class="kw">arrange</span>(spec, Bars, Peduncle, Snout, id) <span class="op">%&gt;%</span></span>
<span id="cb192-7"><a href="supplementary-figure-7.html#cb192-7"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">ord_nr =</span> <span class="kw">row_number</span>()) <span class="op">%&gt;%</span></span>
<span id="cb192-8"><a href="supplementary-figure-7.html#cb192-8"></a><span class="st">  </span><span class="kw">pivot_longer</span>(<span class="dt">names_to =</span> <span class="st">&quot;trait&quot;</span>,</span>
<span id="cb192-9"><a href="supplementary-figure-7.html#cb192-9"></a>               <span class="dt">values_to =</span> <span class="st">&quot;phenotype&quot;</span>,</span>
<span id="cb192-10"><a href="supplementary-figure-7.html#cb192-10"></a>               <span class="dt">cols =</span> Bars<span class="op">:</span>Snout) <span class="op">%&gt;%</span></span>
<span id="cb192-11"><a href="supplementary-figure-7.html#cb192-11"></a><span class="st">  </span><span class="kw">left_join</span>(pheno_facet)</span></code></pre></div>
<p>The separate panels of the figure are actually individual plots.
To assure a consistent sample order, this is defined externally and stored within <code>sample_order</code>.</p>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="supplementary-figure-7.html#cb193-1"></a><span class="co"># helper for consistent sample order across all panels</span></span>
<span id="cb193-2"><a href="supplementary-figure-7.html#cb193-2"></a>sample_order &lt;-<span class="st"> </span>pheno_plot_data <span class="op">%&gt;%</span></span>
<span id="cb193-3"><a href="supplementary-figure-7.html#cb193-3"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">duplicated</span>(id)) <span class="op">%&gt;%</span></span>
<span id="cb193-4"><a href="supplementary-figure-7.html#cb193-4"></a><span class="st">  </span><span class="kw">select</span>(id, ord_nr)</span></code></pre></div>
<p>Then, the admixture panels (a - c) are created as three sepearted plots and stored within a list.</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="supplementary-figure-7.html#cb194-1"></a><span class="co"># create plot panels a-c</span></span>
<span id="cb194-2"><a href="supplementary-figure-7.html#cb194-2"></a>p_ad &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;LG04_1&quot;</span>, <span class="st">&quot;LG12_3&quot;</span>, <span class="st">&quot;LG12_4&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">map</span>(adm_plot)</span></code></pre></div>
<p>(Just as an example - here is the plot for panel c)</p>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="supplementary-figure-7.html#cb195-1"></a>p_ad[[<span class="dv">3</span>]] <span class="co"># not part of the actual script</span></span></code></pre></div>
<center>
<img src="plot_SF7_files/figure-html/unnamed-chunk-15-1.png" width="1382.4" />
</center>
<p>then we create a dummy plot to extract the legend for the phenotypes.</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="supplementary-figure-7.html#cb196-1"></a><span class="co"># create dummy plot for the phenotype legend</span></span>
<span id="cb196-2"><a href="supplementary-figure-7.html#cb196-2"></a>p_phno &lt;-<span class="st"> </span>pheno_plot_data <span class="op">%&gt;%</span></span>
<span id="cb196-3"><a href="supplementary-figure-7.html#cb196-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> ord_nr))<span class="op">+</span></span>
<span id="cb196-4"><a href="supplementary-figure-7.html#cb196-4"></a><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">y =</span> trait, <span class="dt">fill =</span> <span class="kw">factor</span>(phenotype)),<span class="dt">shape =</span> <span class="dv">21</span>)<span class="op">+</span></span>
<span id="cb196-5"><a href="supplementary-figure-7.html#cb196-5"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="st">&quot;Phenotype&lt;br&gt;&lt;img src=&#39;ressources/img/all_traits_c.png&#39; width=&#39;110&#39; /&gt;&quot;</span>,</span>
<span id="cb196-6"><a href="supplementary-figure-7.html#cb196-6"></a>                    <span class="dt">values =</span> <span class="kw">c</span>(<span class="st">`</span><span class="dt">0</span><span class="st">`</span> =<span class="st"> &quot;white&quot;</span>, <span class="st">`</span><span class="dt">1</span><span class="st">`</span> =<span class="st"> &quot;black&quot;</span>),</span>
<span id="cb196-7"><a href="supplementary-figure-7.html#cb196-7"></a>                    <span class="dt">na.value =</span> <span class="st">&quot;gray&quot;</span>,</span>
<span id="cb196-8"><a href="supplementary-figure-7.html#cb196-8"></a>                    <span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;absent&quot;</span>, <span class="st">&quot;present&quot;</span>, <span class="st">&quot;not scored&quot;</span>))<span class="op">+</span></span>
<span id="cb196-9"><a href="supplementary-figure-7.html#cb196-9"></a><span class="st">  </span><span class="kw">guides</span>(<span class="dt">fill =</span> <span class="kw">guide_legend</span>(<span class="dt">ncol =</span> <span class="dv">1</span>))<span class="op">+</span></span>
<span id="cb196-10"><a href="supplementary-figure-7.html#cb196-10"></a><span class="st">  </span><span class="kw">theme_minimal</span>()<span class="op">+</span></span>
<span id="cb196-11"><a href="supplementary-figure-7.html#cb196-11"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.title =</span> <span class="kw">element_markdown</span>(<span class="dt">hjust =</span> <span class="fl">.5</span>),</span>
<span id="cb196-12"><a href="supplementary-figure-7.html#cb196-12"></a>    <span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>)</span></code></pre></div>
<p>Below the panels a - c, the species of the samples are indicated.
To do this, we prepare a small table from the species affiliation of each sample.</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="supplementary-figure-7.html#cb197-1"></a><span class="co"># prepare table with fish annotations for the species indication</span></span>
<span id="cb197-2"><a href="supplementary-figure-7.html#cb197-2"></a>tib_drawing &lt;-<span class="st"> </span>pheno_plot_data <span class="op">%&gt;%</span></span>
<span id="cb197-3"><a href="supplementary-figure-7.html#cb197-3"></a><span class="st">  </span><span class="kw">group_by</span>(spec) <span class="op">%&gt;%</span></span>
<span id="cb197-4"><a href="supplementary-figure-7.html#cb197-4"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">pos =</span> (<span class="kw">min</span>(ord_nr)<span class="op">+</span><span class="kw">max</span>(ord_nr))<span class="op">*</span>.<span class="dv">5</span>) <span class="op">%&gt;%</span></span>
<span id="cb197-5"><a href="supplementary-figure-7.html#cb197-5"></a><span class="st">  </span><span class="kw">ungroup</span>()</span></code></pre></div>
<p>This table is then used to create a small plot.</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="supplementary-figure-7.html#cb198-1"></a><span class="co"># create sub-plot for species indication</span></span>
<span id="cb198-2"><a href="supplementary-figure-7.html#cb198-2"></a>p_spec &lt;-<span class="st"> </span>pheno_plot_data <span class="op">%&gt;%</span></span>
<span id="cb198-3"><a href="supplementary-figure-7.html#cb198-3"></a><span class="st">  </span><span class="kw">group_by</span>(spec) <span class="op">%&gt;%</span></span>
<span id="cb198-4"><a href="supplementary-figure-7.html#cb198-4"></a><span class="st">  </span><span class="kw">summarise</span>(<span class="dt">start =</span> <span class="kw">min</span>(ord_nr)<span class="op">-</span><span class="dv">1</span>,</span>
<span id="cb198-5"><a href="supplementary-figure-7.html#cb198-5"></a>            <span class="dt">end =</span> <span class="kw">max</span>(ord_nr)) <span class="op">%&gt;%</span></span>
<span id="cb198-6"><a href="supplementary-figure-7.html#cb198-6"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">xmin =</span> start, <span class="dt">xmax =</span> end,</span>
<span id="cb198-7"><a href="supplementary-figure-7.html#cb198-7"></a>             <span class="dt">ymin =</span> <span class="op">-</span><span class="ot">Inf</span>,</span>
<span id="cb198-8"><a href="supplementary-figure-7.html#cb198-8"></a>             <span class="dt">ymax =</span> <span class="ot">Inf</span>))<span class="op">+</span></span>
<span id="cb198-9"><a href="supplementary-figure-7.html#cb198-9"></a><span class="st">  </span><span class="co"># add colored backgroud boxes</span></span>
<span id="cb198-10"><a href="supplementary-figure-7.html#cb198-10"></a><span class="st">  </span><span class="kw">geom_rect</span>(<span class="kw">aes</span>(<span class="dt">fill =</span> spec), <span class="dt">color =</span> <span class="st">&quot;black&quot;</span>)<span class="op">+</span></span>
<span id="cb198-11"><a href="supplementary-figure-7.html#cb198-11"></a><span class="st">  </span><span class="co"># add fish images</span></span>
<span id="cb198-12"><a href="supplementary-figure-7.html#cb198-12"></a><span class="st">  </span>(tib_drawing <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pmap</span>(add_spec_drawing))<span class="op">+</span></span>
<span id="cb198-13"><a href="supplementary-figure-7.html#cb198-13"></a><span class="st">  </span><span class="co"># set axis layout</span></span>
<span id="cb198-14"><a href="supplementary-figure-7.html#cb198-14"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="fl">.5</span>, <span class="dt">labels =</span> <span class="kw">c</span>( <span class="st">&quot;Species&quot;</span>), <span class="dt">limits =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))<span class="op">+</span></span>
<span id="cb198-15"><a href="supplementary-figure-7.html#cb198-15"></a><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">breaks =</span> sample_order<span class="op">$</span>ord_nr,</span>
<span id="cb198-16"><a href="supplementary-figure-7.html#cb198-16"></a>                   <span class="dt">labels =</span> sample_order<span class="op">$</span>id,</span>
<span id="cb198-17"><a href="supplementary-figure-7.html#cb198-17"></a>                   <span class="dt">expand =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) <span class="op">+</span></span>
<span id="cb198-18"><a href="supplementary-figure-7.html#cb198-18"></a><span class="st">  </span><span class="co"># set species color scheme</span></span>
<span id="cb198-19"><a href="supplementary-figure-7.html#cb198-19"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="st">&quot;Species&quot;</span>, <span class="dt">values =</span> clr, <span class="dt">labels =</span> sp_labs)<span class="op">+</span></span>
<span id="cb198-20"><a href="supplementary-figure-7.html#cb198-20"></a><span class="st">  </span><span class="co"># set general plot layout</span></span>
<span id="cb198-21"><a href="supplementary-figure-7.html#cb198-21"></a><span class="st">  </span><span class="kw">theme_minimal</span>()<span class="op">+</span></span>
<span id="cb198-22"><a href="supplementary-figure-7.html#cb198-22"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">9</span>),</span>
<span id="cb198-23"><a href="supplementary-figure-7.html#cb198-23"></a>        <span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb198-24"><a href="supplementary-figure-7.html#cb198-24"></a>        <span class="dt">legend.text.align =</span> <span class="dv">0</span>,</span>
<span id="cb198-25"><a href="supplementary-figure-7.html#cb198-25"></a>        <span class="dt">axis.title =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb198-26"><a href="supplementary-figure-7.html#cb198-26"></a>        <span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb198-27"><a href="supplementary-figure-7.html#cb198-27"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>())</span></code></pre></div>
<center>
<img src="plot_SF7_files/figure-html/unnamed-chunk-19-1.png" width="1382.4" />
</center>
<p>Similarely to the species affiliation, we create a second plot to indicate the sampling location.</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="supplementary-figure-7.html#cb199-1"></a><span class="co"># create sub-plot for samplong location indication</span></span>
<span id="cb199-2"><a href="supplementary-figure-7.html#cb199-2"></a>p_loc &lt;-<span class="st"> </span>pheno_plot_data <span class="op">%&gt;%</span></span>
<span id="cb199-3"><a href="supplementary-figure-7.html#cb199-3"></a><span class="st">  </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> <span class="kw">factor</span>(ord_nr)))<span class="op">+</span></span>
<span id="cb199-4"><a href="supplementary-figure-7.html#cb199-4"></a><span class="st">  </span><span class="co"># add colored boxes</span></span>
<span id="cb199-5"><a href="supplementary-figure-7.html#cb199-5"></a><span class="st">  </span><span class="kw">geom_raster</span>(<span class="kw">aes</span>(<span class="dt">y =</span> <span class="dv">0</span>, <span class="dt">fill =</span> loc))<span class="op">+</span></span>
<span id="cb199-6"><a href="supplementary-figure-7.html#cb199-6"></a><span class="st">  </span><span class="co"># set axis layout</span></span>
<span id="cb199-7"><a href="supplementary-figure-7.html#cb199-7"></a><span class="st">  </span><span class="kw">scale_y_continuous</span>(<span class="dt">breaks =</span> <span class="kw">c</span>(<span class="dv">0</span>),<span class="dt">labels =</span> <span class="kw">c</span>(<span class="st">&quot;Location&quot;</span>))<span class="op">+</span></span>
<span id="cb199-8"><a href="supplementary-figure-7.html#cb199-8"></a><span class="st">  </span><span class="kw">scale_x_discrete</span>(<span class="dt">breaks =</span> sample_order<span class="op">$</span>ord_nr,</span>
<span id="cb199-9"><a href="supplementary-figure-7.html#cb199-9"></a>                   <span class="dt">labels =</span> sample_order<span class="op">$</span>id) <span class="op">+</span></span>
<span id="cb199-10"><a href="supplementary-figure-7.html#cb199-10"></a><span class="st">  </span><span class="co"># set location color scheme</span></span>
<span id="cb199-11"><a href="supplementary-figure-7.html#cb199-11"></a><span class="st">  </span><span class="kw">scale_fill_manual</span>(<span class="st">&quot;Location&quot;</span>, <span class="dt">values =</span>  clr_loc, loc_names)<span class="op">+</span></span>
<span id="cb199-12"><a href="supplementary-figure-7.html#cb199-12"></a><span class="st">  </span><span class="co"># set general plot layout</span></span>
<span id="cb199-13"><a href="supplementary-figure-7.html#cb199-13"></a><span class="st">  </span><span class="kw">theme_minimal</span>()<span class="op">+</span></span>
<span id="cb199-14"><a href="supplementary-figure-7.html#cb199-14"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">plot.title =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">9</span>),</span>
<span id="cb199-15"><a href="supplementary-figure-7.html#cb199-15"></a>        <span class="dt">legend.position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb199-16"><a href="supplementary-figure-7.html#cb199-16"></a>        <span class="dt">axis.title =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb199-17"><a href="supplementary-figure-7.html#cb199-17"></a>        <span class="dt">axis.ticks =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb199-18"><a href="supplementary-figure-7.html#cb199-18"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_blank</span>())</span></code></pre></div>
<center>
<img src="plot_SF7_files/figure-html/unnamed-chunk-21-1.png" width="1382.4" />
</center>
<p>Then, we compose the figre legend from individual sub-legends.</p>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="supplementary-figure-7.html#cb200-1"></a><span class="co"># compose legend from individual legend parts</span></span>
<span id="cb200-2"><a href="supplementary-figure-7.html#cb200-2"></a>p_l &lt;-<span class="st"> </span>(<span class="kw">get_legend</span>(p_phno) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggdraw</span>()) <span class="op">+</span></span>
<span id="cb200-3"><a href="supplementary-figure-7.html#cb200-3"></a><span class="st">  </span>(<span class="kw">get_legend</span>(p_spec) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggdraw</span>()) <span class="op">+</span></span>
<span id="cb200-4"><a href="supplementary-figure-7.html#cb200-4"></a><span class="st">  </span>(<span class="kw">get_legend</span>(p_loc) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggdraw</span>()) <span class="op">+</span></span>
<span id="cb200-5"><a href="supplementary-figure-7.html#cb200-5"></a><span class="st">  </span><span class="kw">plot_layout</span>(<span class="dt">nrow =</span> <span class="dv">1</span>)</span></code></pre></div>
<center>
<img src="plot_SF7_files/figure-html/unnamed-chunk-23-1.png" width="1382.4" />
</center>
<p>At this point, we can assemble the final figure.
Yet, due to the positioning of the trait images with <strong>ggtext</strong> we need to crop the left margin a little (hence we need <code>p_done2</code>).</p>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="supplementary-figure-7.html#cb201-1"></a><span class="co"># finalize figure</span></span>
<span id="cb201-2"><a href="supplementary-figure-7.html#cb201-2"></a>p_done &lt;-<span class="st"> </span>p_ad[[<span class="dv">1</span>]] <span class="op">+</span></span>
<span id="cb201-3"><a href="supplementary-figure-7.html#cb201-3"></a><span class="st">  </span>p_ad[[<span class="dv">2</span>]] <span class="op">+</span></span>
<span id="cb201-4"><a href="supplementary-figure-7.html#cb201-4"></a><span class="st">  </span>p_ad[[<span class="dv">3</span>]]<span class="op">+</span></span>
<span id="cb201-5"><a href="supplementary-figure-7.html#cb201-5"></a><span class="st">  </span>p_spec <span class="op">+</span><span class="st"> </span>p_loc <span class="op">+</span><span class="st"> </span>p_l <span class="op">+</span></span>
<span id="cb201-6"><a href="supplementary-figure-7.html#cb201-6"></a><span class="st">  </span><span class="kw">plot_layout</span>(<span class="dt">ncol =</span> <span class="dv">1</span>, <span class="dt">heights =</span> <span class="kw">c</span>(.<span class="dv">4</span>,.<span class="dv">4</span>,.<span class="dv">4</span>,.<span class="dv">08</span>,.<span class="dv">02</span>,.<span class="dv">1</span>)) <span class="op">&amp;</span></span>
<span id="cb201-7"><a href="supplementary-figure-7.html#cb201-7"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb201-8"><a href="supplementary-figure-7.html#cb201-8"></a>        <span class="dt">axis.text =</span> <span class="kw">element_text</span>(<span class="dt">size =</span> <span class="dv">12</span>))</span></code></pre></div>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="supplementary-figure-7.html#cb202-1"></a><span class="co"># crop final figure (remove whitespace on left margin)</span></span>
<span id="cb202-2"><a href="supplementary-figure-7.html#cb202-2"></a>p_done2 &lt;-<span class="st"> </span><span class="kw">ggdraw</span>(p_done, <span class="dt">xlim =</span> <span class="kw">c</span>(.<span class="dv">023</span>,<span class="dv">1</span>))</span></code></pre></div>
<center>
<img src="plot_SF7_files/figure-html/unnamed-chunk-26-1.png" width="1382.4" />
</center>
<p>Finally, we can export Supplementary Figure 7.</p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="supplementary-figure-7.html#cb203-1"></a><span class="co"># export final figure</span></span>
<span id="cb203-2"><a href="supplementary-figure-7.html#cb203-2"></a>scl &lt;-<span class="st"> </span><span class="fl">.9</span></span>
<span id="cb203-3"><a href="supplementary-figure-7.html#cb203-3"></a><span class="kw">ggsave</span>(<span class="st">&quot;figures/SF7.pdf&quot;</span>,</span>
<span id="cb203-4"><a href="supplementary-figure-7.html#cb203-4"></a>       <span class="dt">plot =</span> p_done2,</span>
<span id="cb203-5"><a href="supplementary-figure-7.html#cb203-5"></a>       <span class="dt">width =</span> <span class="dv">16</span><span class="op">*</span>scl,</span>
<span id="cb203-6"><a href="supplementary-figure-7.html#cb203-6"></a>       <span class="dt">height =</span> <span class="dv">10</span><span class="op">*</span>scl,</span>
<span id="cb203-7"><a href="supplementary-figure-7.html#cb203-7"></a>       <span class="dt">device =</span> cairo_pdf)</span></code></pre></div>
<hr />

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="supplementary-figure-6.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="supplementary-figure-8.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
